// autumn-js/dist/react/client/AutumnClientError.js
"use client";
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/react/client/AutumnClientError.ts
var AutumnClientError_exports = {};
__export(AutumnClientError_exports, {
  AutumnClientError: () => AutumnClientError
});
module.exports = __toCommonJS(AutumnClientError_exports);
var AutumnClientError = class extends Error {
  code;
  statusCode;
  details;
  constructor({
    message,
    code,
    statusCode,
    details
  }) {
    super(message);
    this.name = "AutumnClientError";
    this.code = code;
    this.statusCode = statusCode;
    this.details = details;
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AutumnClientError
});
