// autumn-js/dist/react/chunk-R7I3UPL4.mjs
"use client";
import {
  createAutumnClient
} from "./chunk-QEJLIFBC.mjs";
import {
  QueryClient,
  QueryClientProvider
} from "./chunk-XJZH5NMZ.mjs";
import {
  AutumnContext
} from "./chunk-SOA7GEAN.mjs";

// src/react/AutumnProvider.tsx
import { useMemo, useRef } from "react";
import { jsx } from "react/jsx-runtime";
var DEFAULT_PATH_PREFIX = "/api/autumn";
var BETTER_AUTH_PATH_PREFIX = "/api/auth/autumn";
var AutumnProvider = ({
  children,
  backendUrl,
  pathPrefix,
  useBetterAuth,
  includeCredentials
}) => {
  const queryClientRef = useRef(null);
  if (!queryClientRef.current) {
    queryClientRef.current = new QueryClient({
      defaultOptions: {
        queries: {
          staleTime: 1e3 * 60,
          retry: false,
          refetchOnWindowFocus: false,
          refetchOnReconnect: false
        }
      }
    });
  }
  const contextValue = useMemo(() => {
    const resolvedPathPrefix = pathPrefix ?? (useBetterAuth ? BETTER_AUTH_PATH_PREFIX : DEFAULT_PATH_PREFIX);
    const resolvedIncludeCredentials = includeCredentials ?? useBetterAuth;
    const client = createAutumnClient({
      backendUrl,
      pathPrefix: resolvedPathPrefix,
      includeCredentials: resolvedIncludeCredentials
    });
    return { client };
  }, [backendUrl, pathPrefix, useBetterAuth, includeCredentials]);
  return /* @__PURE__ */ jsx(QueryClientProvider, { client: queryClientRef.current, children: /* @__PURE__ */ jsx(AutumnContext.Provider, { value: contextValue, children }) });
};

export {
  AutumnProvider
};
