// autumn-js/dist/react/chunk-QEJLIFBC.mjs
"use client";
import {
  createHttpClient
} from "./chunk-M34U5F2H.mjs";

// src/react/client/AutumnClient.ts
var createAutumnClient = (config) => {
  const http = createHttpClient({
    backendUrl: config.backendUrl,
    pathPrefix: config.pathPrefix,
    includeCredentials: config.includeCredentials
  });
  return {
    getOrCreateCustomer: (params) => http.request({
      route: "getOrCreateCustomer",
      body: params
    }),
    attach: (params) => http.request({
      route: "attach",
      body: params
    }),
    previewAttach: (params) => http.request({
      route: "previewAttach",
      body: params
    }),
    updateSubscription: (params) => http.request({
      route: "updateSubscription",
      body: params
    }),
    previewUpdateSubscription: (params) => http.request({
      route: "previewUpdateSubscription",
      body: params
    }),
    multiAttach: (params) => http.request({
      route: "multiAttach",
      body: params
    }),
    previewMultiAttach: (params) => http.request({
      route: "previewMultiAttach",
      body: params
    }),
    setupPayment: (params) => http.request({
      route: "setupPayment",
      body: params
    }),
    openCustomerPortal: (params) => http.request({
      route: "openCustomerPortal",
      body: params
    }),
    createReferralCode: (params) => http.request({
      route: "createReferralCode",
      body: params
    }),
    redeemReferralCode: (params) => http.request({
      route: "redeemReferralCode",
      body: params
    }),
    listPlans: (params) => http.request({
      route: "listPlans",
      body: params ?? {}
    }),
    listEvents: (params) => http.request({
      route: "listEvents",
      body: params
    }),
    aggregateEvents: (params) => http.request({
      route: "aggregateEvents",
      body: params
    }),
    getEntity: (params) => http.request({
      route: "getEntity",
      body: params
    })
  };
};

export {
  createAutumnClient
};
