// autumn-js/dist/backend/core/handlers/errorTransformer.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/backend/core/handlers/errorTransformer.ts
var errorTransformer_exports = {};
__export(errorTransformer_exports, {
  transformSdkError: () => transformSdkError
});
module.exports = __toCommonJS(errorTransformer_exports);

// ../sdk/src/models/autumn-error.ts
var AutumnError = class extends Error {
  /** HTTP status code */
  statusCode;
  /** HTTP body */
  body;
  /** HTTP headers */
  headers;
  /** HTTP content type */
  contentType;
  /** Raw response */
  rawResponse;
  constructor(message, httpMeta) {
    super(message);
    this.statusCode = httpMeta.response.status;
    this.body = httpMeta.body;
    this.headers = httpMeta.response.headers;
    this.contentType = httpMeta.response.headers.get("content-type") || "";
    this.rawResponse = httpMeta.response;
    this.name = "AutumnError";
  }
};

// src/backend/core/utils/backendRes.ts
var backendError = ({
  message,
  code = "internal_server_error",
  statusCode = 500,
  details
}) => {
  return {
    statusCode,
    body: {
      message: message || "Internal server error",
      code,
      statusCode,
      ...details !== void 0 ? { details } : {}
    }
  };
};

// src/backend/core/handlers/errorTransformer.ts
var parseErrorBody = (body) => {
  if (!body || body.length === 0) return {};
  try {
    return JSON.parse(body);
  } catch {
    return {};
  }
};
var transformSdkError = (error) => {
  if (error instanceof AutumnError) {
    const parsed = parseErrorBody(error.body);
    return backendError({
      statusCode: error.statusCode,
      message: parsed.message ?? error.message ?? "Autumn API request failed",
      code: parsed.code ?? "autumn_api_error"
    });
  }
  if (error instanceof Error) {
    return backendError({
      message: error.message,
      code: "internal_error"
    });
  }
  return backendError({
    message: "Internal server error",
    code: "internal_error"
  });
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  transformSdkError
});
