// autumn-js/dist/backend/core/routes/routeConfigs.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/backend/core/routes/routeConfigs.ts
var routeConfigs_exports = {};
__export(routeConfigs_exports, {
  routeConfigs: () => routeConfigs
});
module.exports = __toCommonJS(routeConfigs_exports);
var import_v415 = require("zod/v4");

// src/generated/aggregateEventsSchemas.ts
var import_v4 = require("zod/v4");
var aggregateEventsGlobalsSchema = import_v4.z.object({
  xApiVersion: import_v4.z.union([import_v4.z.string(), import_v4.z.undefined()]).optional()
});
var aggregateEventsFeatureIdSchema = import_v4.z.union([
  import_v4.z.string(),
  import_v4.z.array(import_v4.z.string())
]);
var aggregateEventsCustomRangeSchema = import_v4.z.object({
  start: import_v4.z.number(),
  end: import_v4.z.number()
});
var aggregateEventsListSchema = import_v4.z.object({
  period: import_v4.z.number(),
  values: import_v4.z.record(import_v4.z.string(), import_v4.z.number()),
  groupedValues: import_v4.z.union([
    import_v4.z.record(import_v4.z.string(), import_v4.z.record(import_v4.z.string(), import_v4.z.number())),
    import_v4.z.undefined()
  ]).optional()
});
var totalSchema = import_v4.z.object({
  count: import_v4.z.number(),
  sum: import_v4.z.number()
});
var aggregateEventsResponseSchema = import_v4.z.object({
  list: import_v4.z.array(aggregateEventsListSchema),
  total: import_v4.z.record(import_v4.z.string(), totalSchema)
});
var aggregateEventsFeatureIdOutboundSchema = import_v4.z.union([
  import_v4.z.string(),
  import_v4.z.array(import_v4.z.string())
]);
var aggregateEventsCustomRangeOutboundSchema = import_v4.z.object({
  start: import_v4.z.number(),
  end: import_v4.z.number()
});
var eventsAggregateParamsOutboundSchema = import_v4.z.object({
  customer_id: import_v4.z.union([import_v4.z.string(), import_v4.z.undefined()]).optional(),
  entity_id: import_v4.z.union([import_v4.z.string(), import_v4.z.undefined()]).optional(),
  feature_id: import_v4.z.union([import_v4.z.string(), import_v4.z.array(import_v4.z.string())]),
  group_by: import_v4.z.union([import_v4.z.string(), import_v4.z.undefined()]).optional(),
  range: import_v4.z.union([import_v4.z.string(), import_v4.z.undefined()]).optional(),
  bin_size: import_v4.z.string(),
  custom_range: import_v4.z.union([aggregateEventsCustomRangeOutboundSchema, import_v4.z.undefined()]).optional(),
  max_groups: import_v4.z.union([import_v4.z.number(), import_v4.z.undefined()]).optional()
});
var closedEnumSchema = import_v4.z.any();
var rangeSchema = closedEnumSchema;
var binSizeSchema = closedEnumSchema;
var eventsAggregateParamsSchema = import_v4.z.object({
  customerId: import_v4.z.union([import_v4.z.string(), import_v4.z.undefined()]).optional(),
  entityId: import_v4.z.union([import_v4.z.string(), import_v4.z.undefined()]).optional(),
  featureId: import_v4.z.union([import_v4.z.string(), import_v4.z.array(import_v4.z.string())]),
  groupBy: import_v4.z.union([import_v4.z.string(), import_v4.z.undefined()]).optional(),
  range: import_v4.z.union([rangeSchema, import_v4.z.undefined()]).optional(),
  binSize: import_v4.z.union([binSizeSchema, import_v4.z.undefined()]).optional(),
  customRange: import_v4.z.union([aggregateEventsCustomRangeSchema, import_v4.z.undefined()]).optional(),
  maxGroups: import_v4.z.union([import_v4.z.number(), import_v4.z.undefined()]).optional()
});

// src/generated/attachSchemas.ts
var import_v42 = require("zod/v4");
var attachGlobalsSchema = import_v42.z.object({
  xApiVersion: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional()
});
var attachFeatureQuantitySchema = import_v42.z.object({
  featureId: import_v42.z.string(),
  quantity: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  adjustable: import_v42.z.union([import_v42.z.boolean(), import_v42.z.undefined()]).optional()
});
var attachToSchema = import_v42.z.union([import_v42.z.number(), import_v42.z.string()]);
var attachTierSchema = import_v42.z.object({
  to: import_v42.z.union([import_v42.z.number(), import_v42.z.string()]),
  amount: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  flatAmount: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachInvoiceModeSchema = import_v42.z.object({
  enabled: import_v42.z.boolean(),
  enablePlanImmediately: import_v42.z.union([import_v42.z.boolean(), import_v42.z.undefined()]).optional(),
  finalize: import_v42.z.union([import_v42.z.boolean(), import_v42.z.undefined()]).optional()
});
var attachAttachDiscountSchema = import_v42.z.object({
  rewardId: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  promotionCode: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional()
});
var attachCustomLineItemSchema = import_v42.z.object({
  amount: import_v42.z.number(),
  description: import_v42.z.string()
});
var attachCarryOverBalancesSchema = import_v42.z.object({
  enabled: import_v42.z.boolean(),
  featureIds: import_v42.z.union([import_v42.z.array(import_v42.z.string()), import_v42.z.undefined()]).optional()
});
var attachCarryOverUsagesSchema = import_v42.z.object({
  enabled: import_v42.z.boolean(),
  featureIds: import_v42.z.union([import_v42.z.array(import_v42.z.string()), import_v42.z.undefined()]).optional()
});
var attachInvoiceSchema = import_v42.z.object({
  status: import_v42.z.string().nullable(),
  stripeId: import_v42.z.string(),
  total: import_v42.z.number(),
  currency: import_v42.z.string(),
  hostedInvoiceUrl: import_v42.z.string().nullable()
});
var attachFeatureQuantityOutboundSchema = import_v42.z.object({
  feature_id: import_v42.z.string(),
  quantity: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  adjustable: import_v42.z.union([import_v42.z.boolean(), import_v42.z.undefined()]).optional()
});
var attachBasePriceOutboundSchema = import_v42.z.object({
  amount: import_v42.z.number(),
  interval: import_v42.z.string(),
  interval_count: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachResetOutboundSchema = import_v42.z.object({
  interval: import_v42.z.string(),
  interval_count: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachToOutboundSchema = import_v42.z.union([import_v42.z.number(), import_v42.z.string()]);
var attachTierOutboundSchema = import_v42.z.object({
  to: import_v42.z.union([import_v42.z.number(), import_v42.z.string()]),
  amount: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  flat_amount: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachPriceOutboundSchema = import_v42.z.object({
  amount: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  tiers: import_v42.z.union([import_v42.z.array(attachTierOutboundSchema), import_v42.z.undefined()]).optional(),
  tier_behavior: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  interval: import_v42.z.string(),
  interval_count: import_v42.z.number(),
  billing_units: import_v42.z.number(),
  billing_method: import_v42.z.string(),
  max_purchase: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachProrationOutboundSchema = import_v42.z.object({
  on_increase: import_v42.z.string(),
  on_decrease: import_v42.z.string()
});
var attachRolloverOutboundSchema = import_v42.z.object({
  max: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  expiry_duration_type: import_v42.z.string(),
  expiry_duration_length: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachPlanItemOutboundSchema = import_v42.z.object({
  feature_id: import_v42.z.string(),
  included: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  unlimited: import_v42.z.union([import_v42.z.boolean(), import_v42.z.undefined()]).optional(),
  reset: import_v42.z.union([attachResetOutboundSchema, import_v42.z.undefined()]).optional(),
  price: import_v42.z.union([attachPriceOutboundSchema, import_v42.z.undefined()]).optional(),
  proration: import_v42.z.union([attachProrationOutboundSchema, import_v42.z.undefined()]).optional(),
  rollover: import_v42.z.union([attachRolloverOutboundSchema, import_v42.z.undefined()]).optional()
});
var attachFreeTrialParamsOutboundSchema = import_v42.z.object({
  duration_length: import_v42.z.number(),
  duration_type: import_v42.z.string(),
  card_required: import_v42.z.boolean()
});
var attachCustomizeOutboundSchema = import_v42.z.object({
  price: import_v42.z.union([attachBasePriceOutboundSchema, import_v42.z.undefined()]).optional().nullable(),
  items: import_v42.z.union([import_v42.z.array(attachPlanItemOutboundSchema), import_v42.z.undefined()]).optional(),
  free_trial: import_v42.z.union([attachFreeTrialParamsOutboundSchema, import_v42.z.undefined()]).optional().nullable()
});
var attachInvoiceModeOutboundSchema = import_v42.z.object({
  enabled: import_v42.z.boolean(),
  enable_plan_immediately: import_v42.z.boolean(),
  finalize: import_v42.z.boolean()
});
var attachAttachDiscountOutboundSchema = import_v42.z.object({
  reward_id: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  promotion_code: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional()
});
var attachCustomLineItemOutboundSchema = import_v42.z.object({
  amount: import_v42.z.number(),
  description: import_v42.z.string()
});
var attachCarryOverBalancesOutboundSchema = import_v42.z.object({
  enabled: import_v42.z.boolean(),
  feature_ids: import_v42.z.union([import_v42.z.array(import_v42.z.string()), import_v42.z.undefined()]).optional()
});
var attachCarryOverUsagesOutboundSchema = import_v42.z.object({
  enabled: import_v42.z.boolean(),
  feature_ids: import_v42.z.union([import_v42.z.array(import_v42.z.string()), import_v42.z.undefined()]).optional()
});
var attachParamsOutboundSchema = import_v42.z.object({
  customer_id: import_v42.z.string(),
  entity_id: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  plan_id: import_v42.z.string(),
  feature_quantities: import_v42.z.union([import_v42.z.array(attachFeatureQuantityOutboundSchema), import_v42.z.undefined()]).optional(),
  version: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  customize: import_v42.z.union([attachCustomizeOutboundSchema, import_v42.z.undefined()]).optional(),
  invoice_mode: import_v42.z.union([attachInvoiceModeOutboundSchema, import_v42.z.undefined()]).optional(),
  proration_behavior: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  redirect_mode: import_v42.z.string(),
  subscription_id: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  discounts: import_v42.z.union([import_v42.z.array(attachAttachDiscountOutboundSchema), import_v42.z.undefined()]).optional(),
  success_url: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  new_billing_subscription: import_v42.z.union([import_v42.z.boolean(), import_v42.z.undefined()]).optional(),
  plan_schedule: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  checkout_session_params: import_v42.z.union([import_v42.z.record(import_v42.z.string(), import_v42.z.any()), import_v42.z.undefined()]).optional(),
  custom_line_items: import_v42.z.union([import_v42.z.array(attachCustomLineItemOutboundSchema), import_v42.z.undefined()]).optional(),
  processor_subscription_id: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  carry_over_balances: import_v42.z.union([attachCarryOverBalancesOutboundSchema, import_v42.z.undefined()]).optional(),
  carry_over_usages: import_v42.z.union([attachCarryOverUsagesOutboundSchema, import_v42.z.undefined()]).optional(),
  metadata: import_v42.z.union([import_v42.z.record(import_v42.z.string(), import_v42.z.string()), import_v42.z.undefined()]).optional()
});
var closedEnumSchema2 = import_v42.z.any();
var openEnumSchema = import_v42.z.any();
var attachPriceIntervalSchema = closedEnumSchema2;
var attachBasePriceSchema = import_v42.z.object({
  amount: import_v42.z.number(),
  interval: attachPriceIntervalSchema,
  intervalCount: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachResetIntervalSchema = closedEnumSchema2;
var attachResetSchema = import_v42.z.object({
  interval: attachResetIntervalSchema,
  intervalCount: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachTierBehaviorSchema = closedEnumSchema2;
var attachItemPriceIntervalSchema = closedEnumSchema2;
var attachBillingMethodSchema = closedEnumSchema2;
var attachPriceSchema = import_v42.z.object({
  amount: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  tiers: import_v42.z.union([import_v42.z.array(attachTierSchema), import_v42.z.undefined()]).optional(),
  tierBehavior: import_v42.z.union([attachTierBehaviorSchema, import_v42.z.undefined()]).optional(),
  interval: attachItemPriceIntervalSchema,
  intervalCount: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  billingUnits: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  billingMethod: attachBillingMethodSchema,
  maxPurchase: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachOnIncreaseSchema = closedEnumSchema2;
var attachOnDecreaseSchema = closedEnumSchema2;
var attachProrationSchema = import_v42.z.object({
  onIncrease: attachOnIncreaseSchema,
  onDecrease: attachOnDecreaseSchema
});
var attachExpiryDurationTypeSchema = closedEnumSchema2;
var attachRolloverSchema = import_v42.z.object({
  max: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  expiryDurationType: attachExpiryDurationTypeSchema,
  expiryDurationLength: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional()
});
var attachPlanItemSchema = import_v42.z.object({
  featureId: import_v42.z.string(),
  included: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  unlimited: import_v42.z.union([import_v42.z.boolean(), import_v42.z.undefined()]).optional(),
  reset: import_v42.z.union([attachResetSchema, import_v42.z.undefined()]).optional(),
  price: import_v42.z.union([attachPriceSchema, import_v42.z.undefined()]).optional(),
  proration: import_v42.z.union([attachProrationSchema, import_v42.z.undefined()]).optional(),
  rollover: import_v42.z.union([attachRolloverSchema, import_v42.z.undefined()]).optional()
});
var attachDurationTypeSchema = closedEnumSchema2;
var attachFreeTrialParamsSchema = import_v42.z.object({
  durationLength: import_v42.z.number(),
  durationType: import_v42.z.union([attachDurationTypeSchema, import_v42.z.undefined()]).optional(),
  cardRequired: import_v42.z.union([import_v42.z.boolean(), import_v42.z.undefined()]).optional()
});
var attachCustomizeSchema = import_v42.z.object({
  price: import_v42.z.union([attachBasePriceSchema, import_v42.z.undefined()]).optional().nullable(),
  items: import_v42.z.union([import_v42.z.array(attachPlanItemSchema), import_v42.z.undefined()]).optional(),
  freeTrial: import_v42.z.union([attachFreeTrialParamsSchema, import_v42.z.undefined()]).optional().nullable()
});
var attachProrationBehaviorSchema = closedEnumSchema2;
var attachRedirectModeSchema = closedEnumSchema2;
var attachPlanScheduleSchema = closedEnumSchema2;
var attachParamsSchema = import_v42.z.object({
  customerId: import_v42.z.string(),
  entityId: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  planId: import_v42.z.string(),
  featureQuantities: import_v42.z.union([import_v42.z.array(attachFeatureQuantitySchema), import_v42.z.undefined()]).optional(),
  version: import_v42.z.union([import_v42.z.number(), import_v42.z.undefined()]).optional(),
  customize: import_v42.z.union([attachCustomizeSchema, import_v42.z.undefined()]).optional(),
  invoiceMode: import_v42.z.union([attachInvoiceModeSchema, import_v42.z.undefined()]).optional(),
  prorationBehavior: import_v42.z.union([attachProrationBehaviorSchema, import_v42.z.undefined()]).optional(),
  redirectMode: import_v42.z.union([attachRedirectModeSchema, import_v42.z.undefined()]).optional(),
  subscriptionId: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  discounts: import_v42.z.union([import_v42.z.array(attachAttachDiscountSchema), import_v42.z.undefined()]).optional(),
  successUrl: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  newBillingSubscription: import_v42.z.union([import_v42.z.boolean(), import_v42.z.undefined()]).optional(),
  planSchedule: import_v42.z.union([attachPlanScheduleSchema, import_v42.z.undefined()]).optional(),
  checkoutSessionParams: import_v42.z.union([import_v42.z.record(import_v42.z.string(), import_v42.z.any()), import_v42.z.undefined()]).optional(),
  customLineItems: import_v42.z.union([import_v42.z.array(attachCustomLineItemSchema), import_v42.z.undefined()]).optional(),
  processorSubscriptionId: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  carryOverBalances: import_v42.z.union([attachCarryOverBalancesSchema, import_v42.z.undefined()]).optional(),
  carryOverUsages: import_v42.z.union([attachCarryOverUsagesSchema, import_v42.z.undefined()]).optional(),
  metadata: import_v42.z.union([import_v42.z.record(import_v42.z.string(), import_v42.z.string()), import_v42.z.undefined()]).optional()
});
var attachCodeSchema = openEnumSchema;
var attachRequiredActionSchema = import_v42.z.object({
  code: attachCodeSchema,
  reason: import_v42.z.string()
});
var attachResponseSchema = import_v42.z.object({
  customerId: import_v42.z.string(),
  entityId: import_v42.z.union([import_v42.z.string(), import_v42.z.undefined()]).optional(),
  invoice: import_v42.z.union([attachInvoiceSchema, import_v42.z.undefined()]).optional(),
  paymentUrl: import_v42.z.string().nullable(),
  requiredAction: import_v42.z.union([attachRequiredActionSchema, import_v42.z.undefined()]).optional()
});

// src/generated/createReferralCodeSchemas.ts
var import_v43 = require("zod/v4");
var createReferralCodeGlobalsSchema = import_v43.z.object({
  xApiVersion: import_v43.z.union([import_v43.z.string(), import_v43.z.undefined()]).optional()
});
var createReferralCodeParamsSchema = import_v43.z.object({
  customerId: import_v43.z.string(),
  programId: import_v43.z.string()
});
var createReferralCodeResponseSchema = import_v43.z.object({
  code: import_v43.z.string(),
  customerId: import_v43.z.string(),
  createdAt: import_v43.z.number()
});
var createReferralCodeParamsOutboundSchema = import_v43.z.object({
  customer_id: import_v43.z.string(),
  program_id: import_v43.z.string()
});

// src/generated/getOrCreateCustomerSchemas.ts
var import_v44 = require("zod/v4");
var getOrCreateCustomerGlobalsSchema = import_v44.z.object({
  xApiVersion: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional()
});
var getOrCreateCustomerSpendLimitSchema = import_v44.z.object({
  featureId: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional(),
  enabled: import_v44.z.union([import_v44.z.boolean(), import_v44.z.undefined()]).optional(),
  overageLimit: import_v44.z.union([import_v44.z.number(), import_v44.z.undefined()]).optional()
});
var getOrCreateCustomerPurchaseLimitOutboundSchema = import_v44.z.object({
  interval: import_v44.z.string(),
  interval_count: import_v44.z.number(),
  limit: import_v44.z.number()
});
var getOrCreateCustomerAutoTopupOutboundSchema = import_v44.z.object({
  feature_id: import_v44.z.string(),
  enabled: import_v44.z.boolean(),
  threshold: import_v44.z.number(),
  quantity: import_v44.z.number(),
  purchase_limit: import_v44.z.union([getOrCreateCustomerPurchaseLimitOutboundSchema, import_v44.z.undefined()]).optional()
});
var getOrCreateCustomerSpendLimitOutboundSchema = import_v44.z.object({
  feature_id: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional(),
  enabled: import_v44.z.boolean(),
  overage_limit: import_v44.z.union([import_v44.z.number(), import_v44.z.undefined()]).optional()
});
var getOrCreateCustomerUsageAlertOutboundSchema = import_v44.z.object({
  feature_id: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional(),
  enabled: import_v44.z.boolean(),
  threshold: import_v44.z.number(),
  threshold_type: import_v44.z.string(),
  name: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional()
});
var getOrCreateCustomerBillingControlsOutboundSchema = import_v44.z.object({
  auto_topups: import_v44.z.union([import_v44.z.array(getOrCreateCustomerAutoTopupOutboundSchema), import_v44.z.undefined()]).optional(),
  spend_limits: import_v44.z.union([
    import_v44.z.array(getOrCreateCustomerSpendLimitOutboundSchema),
    import_v44.z.undefined()
  ]).optional(),
  usage_alerts: import_v44.z.union([
    import_v44.z.array(getOrCreateCustomerUsageAlertOutboundSchema),
    import_v44.z.undefined()
  ]).optional()
});
var getOrCreateCustomerParamsOutboundSchema = import_v44.z.object({
  customer_id: import_v44.z.string().nullable(),
  name: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional().nullable(),
  email: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional().nullable(),
  fingerprint: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional().nullable(),
  metadata: import_v44.z.union([import_v44.z.record(import_v44.z.string(), import_v44.z.any()), import_v44.z.undefined()]).optional().nullable(),
  stripe_id: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional().nullable(),
  create_in_stripe: import_v44.z.union([import_v44.z.boolean(), import_v44.z.undefined()]).optional(),
  auto_enable_plan_id: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional(),
  send_email_receipts: import_v44.z.union([import_v44.z.boolean(), import_v44.z.undefined()]).optional(),
  billing_controls: import_v44.z.union([getOrCreateCustomerBillingControlsOutboundSchema, import_v44.z.undefined()]).optional(),
  expand: import_v44.z.union([import_v44.z.array(import_v44.z.string()), import_v44.z.undefined()]).optional()
});
var closedEnumSchema3 = import_v44.z.any();
var getOrCreateCustomerIntervalSchema = closedEnumSchema3;
var getOrCreateCustomerPurchaseLimitSchema = import_v44.z.object({
  interval: getOrCreateCustomerIntervalSchema,
  intervalCount: import_v44.z.union([import_v44.z.number(), import_v44.z.undefined()]).optional(),
  limit: import_v44.z.number()
});
var getOrCreateCustomerAutoTopupSchema = import_v44.z.object({
  featureId: import_v44.z.string(),
  enabled: import_v44.z.union([import_v44.z.boolean(), import_v44.z.undefined()]).optional(),
  threshold: import_v44.z.number(),
  quantity: import_v44.z.number(),
  purchaseLimit: import_v44.z.union([getOrCreateCustomerPurchaseLimitSchema, import_v44.z.undefined()]).optional()
});
var getOrCreateCustomerThresholdTypeSchema = closedEnumSchema3;
var getOrCreateCustomerUsageAlertSchema = import_v44.z.object({
  featureId: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional(),
  enabled: import_v44.z.union([import_v44.z.boolean(), import_v44.z.undefined()]).optional(),
  threshold: import_v44.z.number(),
  thresholdType: getOrCreateCustomerThresholdTypeSchema,
  name: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional()
});
var getOrCreateCustomerBillingControlsSchema = import_v44.z.object({
  autoTopups: import_v44.z.union([import_v44.z.array(getOrCreateCustomerAutoTopupSchema), import_v44.z.undefined()]).optional(),
  spendLimits: import_v44.z.union([import_v44.z.array(getOrCreateCustomerSpendLimitSchema), import_v44.z.undefined()]).optional(),
  usageAlerts: import_v44.z.union([import_v44.z.array(getOrCreateCustomerUsageAlertSchema), import_v44.z.undefined()]).optional()
});
var getOrCreateCustomerParamsSchema = import_v44.z.object({
  customerId: import_v44.z.string().nullable(),
  name: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional().nullable(),
  email: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional().nullable(),
  fingerprint: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional().nullable(),
  metadata: import_v44.z.union([import_v44.z.record(import_v44.z.string(), import_v44.z.any()), import_v44.z.undefined()]).optional().nullable(),
  stripeId: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional().nullable(),
  createInStripe: import_v44.z.union([import_v44.z.boolean(), import_v44.z.undefined()]).optional(),
  autoEnablePlanId: import_v44.z.union([import_v44.z.string(), import_v44.z.undefined()]).optional(),
  sendEmailReceipts: import_v44.z.union([import_v44.z.boolean(), import_v44.z.undefined()]).optional(),
  billingControls: import_v44.z.union([getOrCreateCustomerBillingControlsSchema, import_v44.z.undefined()]).optional(),
  expand: import_v44.z.union([import_v44.z.array(import_v44.z.string()), import_v44.z.undefined()]).optional()
});

// src/generated/listEventsSchemas.ts
var import_v45 = require("zod/v4");
var listEventsGlobalsSchema = import_v45.z.object({
  xApiVersion: import_v45.z.union([import_v45.z.string(), import_v45.z.undefined()]).optional()
});
var listEventsFeatureIdSchema = import_v45.z.union([
  import_v45.z.string(),
  import_v45.z.array(import_v45.z.string())
]);
var listEventsCustomRangeSchema = import_v45.z.object({
  start: import_v45.z.union([import_v45.z.number(), import_v45.z.undefined()]).optional(),
  end: import_v45.z.union([import_v45.z.number(), import_v45.z.undefined()]).optional()
});
var eventsListParamsSchema = import_v45.z.object({
  offset: import_v45.z.union([import_v45.z.number(), import_v45.z.undefined()]).optional(),
  limit: import_v45.z.union([import_v45.z.number(), import_v45.z.undefined()]).optional(),
  customerId: import_v45.z.union([import_v45.z.string(), import_v45.z.undefined()]).optional(),
  entityId: import_v45.z.union([import_v45.z.string(), import_v45.z.undefined()]).optional(),
  featureId: import_v45.z.union([import_v45.z.string(), import_v45.z.array(import_v45.z.string()), import_v45.z.undefined()]).optional(),
  customRange: import_v45.z.union([listEventsCustomRangeSchema, import_v45.z.undefined()]).optional()
});
var listEventsPropertiesSchema = import_v45.z.object({});
var listEventsListSchema = import_v45.z.object({
  id: import_v45.z.string(),
  timestamp: import_v45.z.number(),
  featureId: import_v45.z.string(),
  customerId: import_v45.z.string(),
  value: import_v45.z.number(),
  properties: listEventsPropertiesSchema
});
var listEventsResponseSchema = import_v45.z.object({
  list: import_v45.z.array(listEventsListSchema),
  hasMore: import_v45.z.boolean(),
  offset: import_v45.z.number(),
  limit: import_v45.z.number(),
  total: import_v45.z.number()
});
var listEventsFeatureIdOutboundSchema = import_v45.z.union([
  import_v45.z.string(),
  import_v45.z.array(import_v45.z.string())
]);
var listEventsCustomRangeOutboundSchema = import_v45.z.object({
  start: import_v45.z.union([import_v45.z.number(), import_v45.z.undefined()]).optional(),
  end: import_v45.z.union([import_v45.z.number(), import_v45.z.undefined()]).optional()
});
var eventsListParamsOutboundSchema = import_v45.z.object({
  offset: import_v45.z.number(),
  limit: import_v45.z.number(),
  customer_id: import_v45.z.union([import_v45.z.string(), import_v45.z.undefined()]).optional(),
  entity_id: import_v45.z.union([import_v45.z.string(), import_v45.z.undefined()]).optional(),
  feature_id: import_v45.z.union([import_v45.z.string(), import_v45.z.array(import_v45.z.string()), import_v45.z.undefined()]).optional(),
  custom_range: import_v45.z.union([listEventsCustomRangeOutboundSchema, import_v45.z.undefined()]).optional()
});

// src/generated/listPlansSchemas.ts
var import_v46 = require("zod/v4");
var listPlansGlobalsSchema = import_v46.z.object({
  xApiVersion: import_v46.z.union([import_v46.z.string(), import_v46.z.undefined()]).optional()
});
var listPlansParamsSchema = import_v46.z.object({
  customerId: import_v46.z.union([import_v46.z.string(), import_v46.z.undefined()]).optional(),
  entityId: import_v46.z.union([import_v46.z.string(), import_v46.z.undefined()]).optional(),
  includeArchived: import_v46.z.union([import_v46.z.boolean(), import_v46.z.undefined()]).optional()
});
var listPlansPriceDisplaySchema = import_v46.z.object({
  primaryText: import_v46.z.string(),
  secondaryText: import_v46.z.union([import_v46.z.string(), import_v46.z.undefined()]).optional()
});
var listPlansFeatureDisplaySchema = import_v46.z.object({
  singular: import_v46.z.string(),
  plural: import_v46.z.string()
});
var listPlansCreditSchemaSchema = import_v46.z.object({
  meteredFeatureId: import_v46.z.string(),
  creditCost: import_v46.z.number()
});
var listPlansItemDisplaySchema = import_v46.z.object({
  primaryText: import_v46.z.string(),
  secondaryText: import_v46.z.union([import_v46.z.string(), import_v46.z.undefined()]).optional()
});
var listPlansParamsOutboundSchema = import_v46.z.object({
  customer_id: import_v46.z.union([import_v46.z.string(), import_v46.z.undefined()]).optional(),
  entity_id: import_v46.z.union([import_v46.z.string(), import_v46.z.undefined()]).optional(),
  include_archived: import_v46.z.union([import_v46.z.boolean(), import_v46.z.undefined()]).optional()
});
var openEnumSchema2 = import_v46.z.any();
var listPlansPriceIntervalSchema = openEnumSchema2;
var listPlansPriceSchema = import_v46.z.object({
  amount: import_v46.z.number(),
  interval: listPlansPriceIntervalSchema,
  intervalCount: import_v46.z.union([import_v46.z.number(), import_v46.z.undefined()]).optional(),
  display: import_v46.z.union([listPlansPriceDisplaySchema, import_v46.z.undefined()]).optional()
});
var listPlansTypeSchema = openEnumSchema2;
var listPlansFeatureSchema = import_v46.z.object({
  id: import_v46.z.string(),
  name: import_v46.z.union([import_v46.z.string(), import_v46.z.undefined()]).optional().nullable(),
  type: listPlansTypeSchema,
  display: import_v46.z.union([listPlansFeatureDisplaySchema, import_v46.z.undefined()]).optional().nullable(),
  creditSchema: import_v46.z.union([import_v46.z.array(listPlansCreditSchemaSchema), import_v46.z.undefined()]).optional().nullable(),
  archived: import_v46.z.union([import_v46.z.boolean(), import_v46.z.undefined()]).optional().nullable()
});
var listPlansResetIntervalSchema = openEnumSchema2;
var listPlansResetSchema = import_v46.z.object({
  interval: listPlansResetIntervalSchema,
  intervalCount: import_v46.z.union([import_v46.z.number(), import_v46.z.undefined()]).optional()
});
var listPlansTierBehaviorSchema = openEnumSchema2;
var listPlansPriceItemIntervalSchema = openEnumSchema2;
var listPlansBillingMethodSchema = openEnumSchema2;
var listPlansItemPriceSchema = import_v46.z.object({
  amount: import_v46.z.union([import_v46.z.number(), import_v46.z.undefined()]).optional(),
  tiers: import_v46.z.union([import_v46.z.array(import_v46.z.any().nullable()), import_v46.z.undefined()]).optional(),
  tierBehavior: import_v46.z.union([listPlansTierBehaviorSchema, import_v46.z.undefined()]).optional(),
  interval: listPlansPriceItemIntervalSchema,
  intervalCount: import_v46.z.union([import_v46.z.number(), import_v46.z.undefined()]).optional(),
  billingUnits: import_v46.z.number(),
  billingMethod: listPlansBillingMethodSchema,
  maxPurchase: import_v46.z.number().nullable()
});
var listPlansExpiryDurationTypeSchema = openEnumSchema2;
var listPlansRolloverSchema = import_v46.z.object({
  max: import_v46.z.number().nullable(),
  expiryDurationType: listPlansExpiryDurationTypeSchema,
  expiryDurationLength: import_v46.z.union([import_v46.z.number(), import_v46.z.undefined()]).optional()
});
var listPlansItemSchema = import_v46.z.object({
  featureId: import_v46.z.string(),
  feature: import_v46.z.union([listPlansFeatureSchema, import_v46.z.undefined()]).optional(),
  included: import_v46.z.number(),
  unlimited: import_v46.z.boolean(),
  reset: listPlansResetSchema.nullable(),
  price: listPlansItemPriceSchema.nullable(),
  display: import_v46.z.union([listPlansItemDisplaySchema, import_v46.z.undefined()]).optional(),
  rollover: import_v46.z.union([listPlansRolloverSchema, import_v46.z.undefined()]).optional()
});
var listPlansDurationTypeSchema = openEnumSchema2;
var listPlansFreeTrialSchema = import_v46.z.object({
  durationLength: import_v46.z.number(),
  durationType: listPlansDurationTypeSchema,
  cardRequired: import_v46.z.boolean()
});
var listPlansEnvSchema = openEnumSchema2;
var listPlansStatusSchema = openEnumSchema2;
var listPlansAttachActionSchema = openEnumSchema2;
var listPlansCustomerEligibilitySchema = import_v46.z.object({
  trialAvailable: import_v46.z.union([import_v46.z.boolean(), import_v46.z.undefined()]).optional(),
  status: import_v46.z.union([listPlansStatusSchema, import_v46.z.undefined()]).optional(),
  canceling: import_v46.z.union([import_v46.z.boolean(), import_v46.z.undefined()]).optional(),
  trialing: import_v46.z.union([import_v46.z.boolean(), import_v46.z.undefined()]).optional(),
  attachAction: listPlansAttachActionSchema
});
var listPlansListSchema = import_v46.z.object({
  id: import_v46.z.string(),
  name: import_v46.z.string(),
  description: import_v46.z.string().nullable(),
  group: import_v46.z.string().nullable(),
  version: import_v46.z.number(),
  addOn: import_v46.z.boolean(),
  autoEnable: import_v46.z.boolean(),
  price: listPlansPriceSchema.nullable(),
  items: import_v46.z.array(listPlansItemSchema),
  freeTrial: import_v46.z.union([listPlansFreeTrialSchema, import_v46.z.undefined()]).optional(),
  createdAt: import_v46.z.number(),
  env: listPlansEnvSchema,
  archived: import_v46.z.boolean(),
  baseVariantId: import_v46.z.string().nullable(),
  customerEligibility: import_v46.z.union([listPlansCustomerEligibilitySchema, import_v46.z.undefined()]).optional()
});
var listPlansResponseSchema = import_v46.z.object({
  list: import_v46.z.array(listPlansListSchema)
});

// src/generated/multiAttachSchemas.ts
var import_v47 = require("zod/v4");
var multiAttachGlobalsSchema = import_v47.z.object({
  xApiVersion: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional()
});
var multiAttachToSchema = import_v47.z.union([import_v47.z.number(), import_v47.z.string()]);
var multiAttachTierSchema = import_v47.z.object({
  to: import_v47.z.union([import_v47.z.number(), import_v47.z.string()]),
  amount: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  flatAmount: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachFeatureQuantitySchema = import_v47.z.object({
  featureId: import_v47.z.string(),
  quantity: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  adjustable: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional()
});
var multiAttachInvoiceModeSchema = import_v47.z.object({
  enabled: import_v47.z.boolean(),
  enablePlanImmediately: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional(),
  finalize: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional()
});
var multiAttachAttachDiscountSchema = import_v47.z.object({
  rewardId: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  promotionCode: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional()
});
var multiAttachSpendLimitSchema = import_v47.z.object({
  featureId: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  enabled: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional(),
  overageLimit: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachInvoiceSchema = import_v47.z.object({
  status: import_v47.z.string().nullable(),
  stripeId: import_v47.z.string(),
  total: import_v47.z.number(),
  currency: import_v47.z.string(),
  hostedInvoiceUrl: import_v47.z.string().nullable()
});
var multiAttachBasePriceOutboundSchema = import_v47.z.object({
  amount: import_v47.z.number(),
  interval: import_v47.z.string(),
  interval_count: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachResetOutboundSchema = import_v47.z.object({
  interval: import_v47.z.string(),
  interval_count: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachToOutboundSchema = import_v47.z.union([import_v47.z.number(), import_v47.z.string()]);
var multiAttachTierOutboundSchema = import_v47.z.object({
  to: import_v47.z.union([import_v47.z.number(), import_v47.z.string()]),
  amount: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  flat_amount: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachPriceOutboundSchema = import_v47.z.object({
  amount: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  tiers: import_v47.z.union([import_v47.z.array(multiAttachTierOutboundSchema), import_v47.z.undefined()]).optional(),
  tier_behavior: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  interval: import_v47.z.string(),
  interval_count: import_v47.z.number(),
  billing_units: import_v47.z.number(),
  billing_method: import_v47.z.string(),
  max_purchase: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachProrationOutboundSchema = import_v47.z.object({
  on_increase: import_v47.z.string(),
  on_decrease: import_v47.z.string()
});
var multiAttachRolloverOutboundSchema = import_v47.z.object({
  max: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  expiry_duration_type: import_v47.z.string(),
  expiry_duration_length: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachPlanItemOutboundSchema = import_v47.z.object({
  feature_id: import_v47.z.string(),
  included: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  unlimited: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional(),
  reset: import_v47.z.union([multiAttachResetOutboundSchema, import_v47.z.undefined()]).optional(),
  price: import_v47.z.union([multiAttachPriceOutboundSchema, import_v47.z.undefined()]).optional(),
  proration: import_v47.z.union([multiAttachProrationOutboundSchema, import_v47.z.undefined()]).optional(),
  rollover: import_v47.z.union([multiAttachRolloverOutboundSchema, import_v47.z.undefined()]).optional()
});
var multiAttachCustomizeOutboundSchema = import_v47.z.object({
  price: import_v47.z.union([multiAttachBasePriceOutboundSchema, import_v47.z.undefined()]).optional().nullable(),
  items: import_v47.z.union([import_v47.z.array(multiAttachPlanItemOutboundSchema), import_v47.z.undefined()]).optional()
});
var multiAttachFeatureQuantityOutboundSchema = import_v47.z.object({
  feature_id: import_v47.z.string(),
  quantity: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  adjustable: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional()
});
var multiAttachPlanOutboundSchema = import_v47.z.object({
  plan_id: import_v47.z.string(),
  customize: import_v47.z.union([multiAttachCustomizeOutboundSchema, import_v47.z.undefined()]).optional(),
  feature_quantities: import_v47.z.union([import_v47.z.array(multiAttachFeatureQuantityOutboundSchema), import_v47.z.undefined()]).optional(),
  version: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  subscription_id: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional()
});
var multiAttachFreeTrialParamsOutboundSchema = import_v47.z.object({
  duration_length: import_v47.z.number(),
  duration_type: import_v47.z.string(),
  card_required: import_v47.z.boolean()
});
var multiAttachInvoiceModeOutboundSchema = import_v47.z.object({
  enabled: import_v47.z.boolean(),
  enable_plan_immediately: import_v47.z.boolean(),
  finalize: import_v47.z.boolean()
});
var multiAttachAttachDiscountOutboundSchema = import_v47.z.object({
  reward_id: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  promotion_code: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional()
});
var multiAttachSpendLimitOutboundSchema = import_v47.z.object({
  feature_id: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  enabled: import_v47.z.boolean(),
  overage_limit: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachUsageAlertOutboundSchema = import_v47.z.object({
  feature_id: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  enabled: import_v47.z.boolean(),
  threshold: import_v47.z.number(),
  threshold_type: import_v47.z.string(),
  name: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional()
});
var multiAttachBillingControlsOutboundSchema = import_v47.z.object({
  spend_limits: import_v47.z.union([import_v47.z.array(multiAttachSpendLimitOutboundSchema), import_v47.z.undefined()]).optional(),
  usage_alerts: import_v47.z.union([import_v47.z.array(multiAttachUsageAlertOutboundSchema), import_v47.z.undefined()]).optional()
});
var multiAttachEntityDataOutboundSchema = import_v47.z.object({
  feature_id: import_v47.z.string(),
  name: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  billing_controls: import_v47.z.union([multiAttachBillingControlsOutboundSchema, import_v47.z.undefined()]).optional()
});
var closedEnumSchema4 = import_v47.z.any();
var customerDataSchema = import_v47.z.any();
var openEnumSchema3 = import_v47.z.any();
var customerDataOutboundSchema = import_v47.z.any();
var multiAttachPriceIntervalSchema = closedEnumSchema4;
var multiAttachBasePriceSchema = import_v47.z.object({
  amount: import_v47.z.number(),
  interval: multiAttachPriceIntervalSchema,
  intervalCount: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachResetIntervalSchema = closedEnumSchema4;
var multiAttachResetSchema = import_v47.z.object({
  interval: multiAttachResetIntervalSchema,
  intervalCount: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachTierBehaviorSchema = closedEnumSchema4;
var multiAttachItemPriceIntervalSchema = closedEnumSchema4;
var multiAttachBillingMethodSchema = closedEnumSchema4;
var multiAttachPriceSchema = import_v47.z.object({
  amount: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  tiers: import_v47.z.union([import_v47.z.array(multiAttachTierSchema), import_v47.z.undefined()]).optional(),
  tierBehavior: import_v47.z.union([multiAttachTierBehaviorSchema, import_v47.z.undefined()]).optional(),
  interval: multiAttachItemPriceIntervalSchema,
  intervalCount: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  billingUnits: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  billingMethod: multiAttachBillingMethodSchema,
  maxPurchase: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachOnIncreaseSchema = closedEnumSchema4;
var multiAttachOnDecreaseSchema = closedEnumSchema4;
var multiAttachProrationSchema = import_v47.z.object({
  onIncrease: multiAttachOnIncreaseSchema,
  onDecrease: multiAttachOnDecreaseSchema
});
var multiAttachExpiryDurationTypeSchema = closedEnumSchema4;
var multiAttachRolloverSchema = import_v47.z.object({
  max: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  expiryDurationType: multiAttachExpiryDurationTypeSchema,
  expiryDurationLength: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional()
});
var multiAttachPlanItemSchema = import_v47.z.object({
  featureId: import_v47.z.string(),
  included: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  unlimited: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional(),
  reset: import_v47.z.union([multiAttachResetSchema, import_v47.z.undefined()]).optional(),
  price: import_v47.z.union([multiAttachPriceSchema, import_v47.z.undefined()]).optional(),
  proration: import_v47.z.union([multiAttachProrationSchema, import_v47.z.undefined()]).optional(),
  rollover: import_v47.z.union([multiAttachRolloverSchema, import_v47.z.undefined()]).optional()
});
var multiAttachCustomizeSchema = import_v47.z.object({
  price: import_v47.z.union([multiAttachBasePriceSchema, import_v47.z.undefined()]).optional().nullable(),
  items: import_v47.z.union([import_v47.z.array(multiAttachPlanItemSchema), import_v47.z.undefined()]).optional()
});
var multiAttachPlanSchema = import_v47.z.object({
  planId: import_v47.z.string(),
  customize: import_v47.z.union([multiAttachCustomizeSchema, import_v47.z.undefined()]).optional(),
  featureQuantities: import_v47.z.union([import_v47.z.array(multiAttachFeatureQuantitySchema), import_v47.z.undefined()]).optional(),
  version: import_v47.z.union([import_v47.z.number(), import_v47.z.undefined()]).optional(),
  subscriptionId: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional()
});
var multiAttachDurationTypeSchema = closedEnumSchema4;
var multiAttachFreeTrialParamsSchema = import_v47.z.object({
  durationLength: import_v47.z.number(),
  durationType: import_v47.z.union([multiAttachDurationTypeSchema, import_v47.z.undefined()]).optional(),
  cardRequired: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional()
});
var multiAttachRedirectModeSchema = closedEnumSchema4;
var multiAttachThresholdTypeSchema = closedEnumSchema4;
var multiAttachUsageAlertSchema = import_v47.z.object({
  featureId: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  enabled: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional(),
  threshold: import_v47.z.number(),
  thresholdType: multiAttachThresholdTypeSchema,
  name: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional()
});
var multiAttachBillingControlsSchema = import_v47.z.object({
  spendLimits: import_v47.z.union([import_v47.z.array(multiAttachSpendLimitSchema), import_v47.z.undefined()]).optional(),
  usageAlerts: import_v47.z.union([import_v47.z.array(multiAttachUsageAlertSchema), import_v47.z.undefined()]).optional()
});
var multiAttachEntityDataSchema = import_v47.z.object({
  featureId: import_v47.z.string(),
  name: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  billingControls: import_v47.z.union([multiAttachBillingControlsSchema, import_v47.z.undefined()]).optional()
});
var multiAttachParamsSchema = import_v47.z.object({
  customerId: import_v47.z.string(),
  entityId: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  plans: import_v47.z.array(multiAttachPlanSchema),
  freeTrial: import_v47.z.union([multiAttachFreeTrialParamsSchema, import_v47.z.undefined()]).optional().nullable(),
  invoiceMode: import_v47.z.union([multiAttachInvoiceModeSchema, import_v47.z.undefined()]).optional(),
  discounts: import_v47.z.union([import_v47.z.array(multiAttachAttachDiscountSchema), import_v47.z.undefined()]).optional(),
  successUrl: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  checkoutSessionParams: import_v47.z.union([import_v47.z.record(import_v47.z.string(), import_v47.z.any()), import_v47.z.undefined()]).optional(),
  redirectMode: import_v47.z.union([multiAttachRedirectModeSchema, import_v47.z.undefined()]).optional(),
  newBillingSubscription: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional(),
  customerData: import_v47.z.union([customerDataSchema, import_v47.z.undefined()]).optional(),
  entityData: import_v47.z.union([multiAttachEntityDataSchema, import_v47.z.undefined()]).optional()
});
var multiAttachCodeSchema = openEnumSchema3;
var multiAttachRequiredActionSchema = import_v47.z.object({
  code: multiAttachCodeSchema,
  reason: import_v47.z.string()
});
var multiAttachResponseSchema = import_v47.z.object({
  customerId: import_v47.z.string(),
  entityId: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  invoice: import_v47.z.union([multiAttachInvoiceSchema, import_v47.z.undefined()]).optional(),
  paymentUrl: import_v47.z.string().nullable(),
  requiredAction: import_v47.z.union([multiAttachRequiredActionSchema, import_v47.z.undefined()]).optional()
});
var multiAttachParamsOutboundSchema = import_v47.z.object({
  customer_id: import_v47.z.string(),
  entity_id: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  plans: import_v47.z.array(multiAttachPlanOutboundSchema),
  free_trial: import_v47.z.union([multiAttachFreeTrialParamsOutboundSchema, import_v47.z.undefined()]).optional().nullable(),
  invoice_mode: import_v47.z.union([multiAttachInvoiceModeOutboundSchema, import_v47.z.undefined()]).optional(),
  discounts: import_v47.z.union([import_v47.z.array(multiAttachAttachDiscountOutboundSchema), import_v47.z.undefined()]).optional(),
  success_url: import_v47.z.union([import_v47.z.string(), import_v47.z.undefined()]).optional(),
  checkout_session_params: import_v47.z.union([import_v47.z.record(import_v47.z.string(), import_v47.z.any()), import_v47.z.undefined()]).optional(),
  redirect_mode: import_v47.z.string(),
  new_billing_subscription: import_v47.z.union([import_v47.z.boolean(), import_v47.z.undefined()]).optional(),
  customer_data: import_v47.z.union([customerDataOutboundSchema, import_v47.z.undefined()]).optional(),
  entity_data: import_v47.z.union([multiAttachEntityDataOutboundSchema, import_v47.z.undefined()]).optional()
});

// src/generated/openCustomerPortalSchemas.ts
var import_v48 = require("zod/v4");
var openCustomerPortalGlobalsSchema = import_v48.z.object({
  xApiVersion: import_v48.z.union([import_v48.z.string(), import_v48.z.undefined()]).optional()
});
var openCustomerPortalParamsSchema = import_v48.z.object({
  customerId: import_v48.z.string(),
  configurationId: import_v48.z.union([import_v48.z.string(), import_v48.z.undefined()]).optional(),
  returnUrl: import_v48.z.union([import_v48.z.string(), import_v48.z.undefined()]).optional()
});
var openCustomerPortalResponseSchema = import_v48.z.object({
  customerId: import_v48.z.string(),
  url: import_v48.z.string()
});
var openCustomerPortalParamsOutboundSchema = import_v48.z.object({
  customer_id: import_v48.z.string(),
  configuration_id: import_v48.z.union([import_v48.z.string(), import_v48.z.undefined()]).optional(),
  return_url: import_v48.z.union([import_v48.z.string(), import_v48.z.undefined()]).optional()
});

// src/generated/previewAttachSchemas.ts
var import_v49 = require("zod/v4");
var previewAttachGlobalsSchema = import_v49.z.object({
  xApiVersion: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional()
});
var previewAttachFeatureQuantityRequestSchema = import_v49.z.object({
  featureId: import_v49.z.string(),
  quantity: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  adjustable: import_v49.z.union([import_v49.z.boolean(), import_v49.z.undefined()]).optional()
});
var previewAttachToSchema = import_v49.z.union([import_v49.z.number(), import_v49.z.string()]);
var previewAttachTierSchema = import_v49.z.object({
  to: import_v49.z.union([import_v49.z.number(), import_v49.z.string()]),
  amount: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  flatAmount: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachInvoiceModeSchema = import_v49.z.object({
  enabled: import_v49.z.boolean(),
  enablePlanImmediately: import_v49.z.union([import_v49.z.boolean(), import_v49.z.undefined()]).optional(),
  finalize: import_v49.z.union([import_v49.z.boolean(), import_v49.z.undefined()]).optional()
});
var previewAttachAttachDiscountSchema = import_v49.z.object({
  rewardId: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  promotionCode: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional()
});
var previewAttachCustomLineItemSchema = import_v49.z.object({
  amount: import_v49.z.number(),
  description: import_v49.z.string()
});
var previewAttachCarryOverBalancesSchema = import_v49.z.object({
  enabled: import_v49.z.boolean(),
  featureIds: import_v49.z.union([import_v49.z.array(import_v49.z.string()), import_v49.z.undefined()]).optional()
});
var previewAttachCarryOverUsagesSchema = import_v49.z.object({
  enabled: import_v49.z.boolean(),
  featureIds: import_v49.z.union([import_v49.z.array(import_v49.z.string()), import_v49.z.undefined()]).optional()
});
var previewAttachDiscountSchema = import_v49.z.object({
  amountOff: import_v49.z.number(),
  percentOff: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  rewardId: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  rewardName: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional()
});
var previewAttachLineItemPeriodSchema = import_v49.z.object({
  start: import_v49.z.number(),
  end: import_v49.z.number()
});
var previewAttachLineItemSchema = import_v49.z.object({
  displayName: import_v49.z.string(),
  description: import_v49.z.string(),
  subtotal: import_v49.z.number(),
  total: import_v49.z.number(),
  discounts: import_v49.z.union([import_v49.z.array(previewAttachDiscountSchema), import_v49.z.undefined()]).optional(),
  planId: import_v49.z.string(),
  featureId: import_v49.z.string().nullable(),
  period: import_v49.z.union([previewAttachLineItemPeriodSchema, import_v49.z.undefined()]).optional(),
  quantity: import_v49.z.number()
});
var previewAttachNextCycleDiscountSchema = import_v49.z.object({
  amountOff: import_v49.z.number(),
  percentOff: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  rewardId: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  rewardName: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional()
});
var previewAttachNextCycleLineItemPeriodSchema = import_v49.z.object({
  start: import_v49.z.number(),
  end: import_v49.z.number()
});
var previewAttachNextCycleLineItemSchema = import_v49.z.object({
  displayName: import_v49.z.string(),
  description: import_v49.z.string(),
  subtotal: import_v49.z.number(),
  total: import_v49.z.number(),
  discounts: import_v49.z.union([import_v49.z.array(previewAttachNextCycleDiscountSchema), import_v49.z.undefined()]).optional(),
  planId: import_v49.z.string(),
  featureId: import_v49.z.string().nullable(),
  period: import_v49.z.union([previewAttachNextCycleLineItemPeriodSchema, import_v49.z.undefined()]).optional(),
  quantity: import_v49.z.number()
});
var previewAttachUsageLineItemPeriodSchema = import_v49.z.object({
  start: import_v49.z.number(),
  end: import_v49.z.number()
});
var previewAttachUsageLineItemSchema = import_v49.z.object({
  displayName: import_v49.z.string(),
  planId: import_v49.z.string(),
  featureId: import_v49.z.string().nullable(),
  period: import_v49.z.union([previewAttachUsageLineItemPeriodSchema, import_v49.z.undefined()]).optional()
});
var previewAttachNextCycleSchema = import_v49.z.object({
  startsAt: import_v49.z.number(),
  subtotal: import_v49.z.number(),
  total: import_v49.z.number(),
  lineItems: import_v49.z.array(previewAttachNextCycleLineItemSchema),
  usageLineItems: import_v49.z.array(previewAttachUsageLineItemSchema)
});
var previewAttachIncomingFeatureQuantitySchema = import_v49.z.object({
  featureId: import_v49.z.string(),
  quantity: import_v49.z.number()
});
var previewAttachOutgoingFeatureQuantitySchema = import_v49.z.object({
  featureId: import_v49.z.string(),
  quantity: import_v49.z.number()
});
var previewAttachFeatureQuantityRequestOutboundSchema = import_v49.z.object({
  feature_id: import_v49.z.string(),
  quantity: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  adjustable: import_v49.z.union([import_v49.z.boolean(), import_v49.z.undefined()]).optional()
});
var previewAttachBasePriceOutboundSchema = import_v49.z.object({
  amount: import_v49.z.number(),
  interval: import_v49.z.string(),
  interval_count: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachResetOutboundSchema = import_v49.z.object({
  interval: import_v49.z.string(),
  interval_count: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachToOutboundSchema = import_v49.z.union([import_v49.z.number(), import_v49.z.string()]);
var previewAttachTierOutboundSchema = import_v49.z.object({
  to: import_v49.z.union([import_v49.z.number(), import_v49.z.string()]),
  amount: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  flat_amount: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachPriceOutboundSchema = import_v49.z.object({
  amount: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  tiers: import_v49.z.union([import_v49.z.array(previewAttachTierOutboundSchema), import_v49.z.undefined()]).optional(),
  tier_behavior: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  interval: import_v49.z.string(),
  interval_count: import_v49.z.number(),
  billing_units: import_v49.z.number(),
  billing_method: import_v49.z.string(),
  max_purchase: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachProrationOutboundSchema = import_v49.z.object({
  on_increase: import_v49.z.string(),
  on_decrease: import_v49.z.string()
});
var previewAttachRolloverOutboundSchema = import_v49.z.object({
  max: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  expiry_duration_type: import_v49.z.string(),
  expiry_duration_length: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachPlanItemOutboundSchema = import_v49.z.object({
  feature_id: import_v49.z.string(),
  included: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  unlimited: import_v49.z.union([import_v49.z.boolean(), import_v49.z.undefined()]).optional(),
  reset: import_v49.z.union([previewAttachResetOutboundSchema, import_v49.z.undefined()]).optional(),
  price: import_v49.z.union([previewAttachPriceOutboundSchema, import_v49.z.undefined()]).optional(),
  proration: import_v49.z.union([previewAttachProrationOutboundSchema, import_v49.z.undefined()]).optional(),
  rollover: import_v49.z.union([previewAttachRolloverOutboundSchema, import_v49.z.undefined()]).optional()
});
var previewAttachFreeTrialParamsOutboundSchema = import_v49.z.object({
  duration_length: import_v49.z.number(),
  duration_type: import_v49.z.string(),
  card_required: import_v49.z.boolean()
});
var previewAttachCustomizeOutboundSchema = import_v49.z.object({
  price: import_v49.z.union([previewAttachBasePriceOutboundSchema, import_v49.z.undefined()]).optional().nullable(),
  items: import_v49.z.union([import_v49.z.array(previewAttachPlanItemOutboundSchema), import_v49.z.undefined()]).optional(),
  free_trial: import_v49.z.union([previewAttachFreeTrialParamsOutboundSchema, import_v49.z.undefined()]).optional().nullable()
});
var previewAttachInvoiceModeOutboundSchema = import_v49.z.object({
  enabled: import_v49.z.boolean(),
  enable_plan_immediately: import_v49.z.boolean(),
  finalize: import_v49.z.boolean()
});
var previewAttachAttachDiscountOutboundSchema = import_v49.z.object({
  reward_id: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  promotion_code: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional()
});
var previewAttachCustomLineItemOutboundSchema = import_v49.z.object({
  amount: import_v49.z.number(),
  description: import_v49.z.string()
});
var previewAttachCarryOverBalancesOutboundSchema = import_v49.z.object({
  enabled: import_v49.z.boolean(),
  feature_ids: import_v49.z.union([import_v49.z.array(import_v49.z.string()), import_v49.z.undefined()]).optional()
});
var previewAttachCarryOverUsagesOutboundSchema = import_v49.z.object({
  enabled: import_v49.z.boolean(),
  feature_ids: import_v49.z.union([import_v49.z.array(import_v49.z.string()), import_v49.z.undefined()]).optional()
});
var previewAttachParamsOutboundSchema = import_v49.z.object({
  customer_id: import_v49.z.string(),
  entity_id: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  plan_id: import_v49.z.string(),
  feature_quantities: import_v49.z.union([
    import_v49.z.array(previewAttachFeatureQuantityRequestOutboundSchema),
    import_v49.z.undefined()
  ]).optional(),
  version: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  customize: import_v49.z.union([previewAttachCustomizeOutboundSchema, import_v49.z.undefined()]).optional(),
  invoice_mode: import_v49.z.union([previewAttachInvoiceModeOutboundSchema, import_v49.z.undefined()]).optional(),
  proration_behavior: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  redirect_mode: import_v49.z.string(),
  subscription_id: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  discounts: import_v49.z.union([import_v49.z.array(previewAttachAttachDiscountOutboundSchema), import_v49.z.undefined()]).optional(),
  success_url: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  new_billing_subscription: import_v49.z.union([import_v49.z.boolean(), import_v49.z.undefined()]).optional(),
  plan_schedule: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  checkout_session_params: import_v49.z.union([import_v49.z.record(import_v49.z.string(), import_v49.z.any()), import_v49.z.undefined()]).optional(),
  custom_line_items: import_v49.z.union([import_v49.z.array(previewAttachCustomLineItemOutboundSchema), import_v49.z.undefined()]).optional(),
  processor_subscription_id: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  carry_over_balances: import_v49.z.union([previewAttachCarryOverBalancesOutboundSchema, import_v49.z.undefined()]).optional(),
  carry_over_usages: import_v49.z.union([previewAttachCarryOverUsagesOutboundSchema, import_v49.z.undefined()]).optional(),
  metadata: import_v49.z.union([import_v49.z.record(import_v49.z.string(), import_v49.z.string()), import_v49.z.undefined()]).optional()
});
var closedEnumSchema5 = import_v49.z.any();
var planSchema = import_v49.z.any();
var openEnumSchema4 = import_v49.z.any();
var previewAttachPriceIntervalSchema = closedEnumSchema5;
var previewAttachBasePriceSchema = import_v49.z.object({
  amount: import_v49.z.number(),
  interval: previewAttachPriceIntervalSchema,
  intervalCount: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachResetIntervalSchema = closedEnumSchema5;
var previewAttachResetSchema = import_v49.z.object({
  interval: previewAttachResetIntervalSchema,
  intervalCount: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachTierBehaviorSchema = closedEnumSchema5;
var previewAttachItemPriceIntervalSchema = closedEnumSchema5;
var previewAttachBillingMethodSchema = closedEnumSchema5;
var previewAttachPriceSchema = import_v49.z.object({
  amount: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  tiers: import_v49.z.union([import_v49.z.array(previewAttachTierSchema), import_v49.z.undefined()]).optional(),
  tierBehavior: import_v49.z.union([previewAttachTierBehaviorSchema, import_v49.z.undefined()]).optional(),
  interval: previewAttachItemPriceIntervalSchema,
  intervalCount: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  billingUnits: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  billingMethod: previewAttachBillingMethodSchema,
  maxPurchase: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachOnIncreaseSchema = closedEnumSchema5;
var previewAttachOnDecreaseSchema = closedEnumSchema5;
var previewAttachProrationSchema = import_v49.z.object({
  onIncrease: previewAttachOnIncreaseSchema,
  onDecrease: previewAttachOnDecreaseSchema
});
var previewAttachExpiryDurationTypeSchema = closedEnumSchema5;
var previewAttachRolloverSchema = import_v49.z.object({
  max: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  expiryDurationType: previewAttachExpiryDurationTypeSchema,
  expiryDurationLength: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional()
});
var previewAttachPlanItemSchema = import_v49.z.object({
  featureId: import_v49.z.string(),
  included: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  unlimited: import_v49.z.union([import_v49.z.boolean(), import_v49.z.undefined()]).optional(),
  reset: import_v49.z.union([previewAttachResetSchema, import_v49.z.undefined()]).optional(),
  price: import_v49.z.union([previewAttachPriceSchema, import_v49.z.undefined()]).optional(),
  proration: import_v49.z.union([previewAttachProrationSchema, import_v49.z.undefined()]).optional(),
  rollover: import_v49.z.union([previewAttachRolloverSchema, import_v49.z.undefined()]).optional()
});
var previewAttachDurationTypeSchema = closedEnumSchema5;
var previewAttachFreeTrialParamsSchema = import_v49.z.object({
  durationLength: import_v49.z.number(),
  durationType: import_v49.z.union([previewAttachDurationTypeSchema, import_v49.z.undefined()]).optional(),
  cardRequired: import_v49.z.union([import_v49.z.boolean(), import_v49.z.undefined()]).optional()
});
var previewAttachCustomizeSchema = import_v49.z.object({
  price: import_v49.z.union([previewAttachBasePriceSchema, import_v49.z.undefined()]).optional().nullable(),
  items: import_v49.z.union([import_v49.z.array(previewAttachPlanItemSchema), import_v49.z.undefined()]).optional(),
  freeTrial: import_v49.z.union([previewAttachFreeTrialParamsSchema, import_v49.z.undefined()]).optional().nullable()
});
var previewAttachProrationBehaviorSchema = closedEnumSchema5;
var previewAttachRedirectModeSchema = closedEnumSchema5;
var previewAttachPlanScheduleSchema = closedEnumSchema5;
var previewAttachParamsSchema = import_v49.z.object({
  customerId: import_v49.z.string(),
  entityId: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  planId: import_v49.z.string(),
  featureQuantities: import_v49.z.union([import_v49.z.array(previewAttachFeatureQuantityRequestSchema), import_v49.z.undefined()]).optional(),
  version: import_v49.z.union([import_v49.z.number(), import_v49.z.undefined()]).optional(),
  customize: import_v49.z.union([previewAttachCustomizeSchema, import_v49.z.undefined()]).optional(),
  invoiceMode: import_v49.z.union([previewAttachInvoiceModeSchema, import_v49.z.undefined()]).optional(),
  prorationBehavior: import_v49.z.union([previewAttachProrationBehaviorSchema, import_v49.z.undefined()]).optional(),
  redirectMode: import_v49.z.union([previewAttachRedirectModeSchema, import_v49.z.undefined()]).optional(),
  subscriptionId: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  discounts: import_v49.z.union([import_v49.z.array(previewAttachAttachDiscountSchema), import_v49.z.undefined()]).optional(),
  successUrl: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  newBillingSubscription: import_v49.z.union([import_v49.z.boolean(), import_v49.z.undefined()]).optional(),
  planSchedule: import_v49.z.union([previewAttachPlanScheduleSchema, import_v49.z.undefined()]).optional(),
  checkoutSessionParams: import_v49.z.union([import_v49.z.record(import_v49.z.string(), import_v49.z.any()), import_v49.z.undefined()]).optional(),
  customLineItems: import_v49.z.union([import_v49.z.array(previewAttachCustomLineItemSchema), import_v49.z.undefined()]).optional(),
  processorSubscriptionId: import_v49.z.union([import_v49.z.string(), import_v49.z.undefined()]).optional(),
  carryOverBalances: import_v49.z.union([previewAttachCarryOverBalancesSchema, import_v49.z.undefined()]).optional(),
  carryOverUsages: import_v49.z.union([previewAttachCarryOverUsagesSchema, import_v49.z.undefined()]).optional(),
  metadata: import_v49.z.union([import_v49.z.record(import_v49.z.string(), import_v49.z.string()), import_v49.z.undefined()]).optional()
});
var previewAttachIncomingSchema = import_v49.z.object({
  planId: import_v49.z.string(),
  plan: import_v49.z.union([planSchema, import_v49.z.undefined()]).optional(),
  featureQuantities: import_v49.z.array(previewAttachIncomingFeatureQuantitySchema),
  effectiveAt: import_v49.z.number().nullable(),
  canceledAt: import_v49.z.number().nullable(),
  expiresAt: import_v49.z.number().nullable()
});
var previewAttachOutgoingSchema = import_v49.z.object({
  planId: import_v49.z.string(),
  plan: import_v49.z.union([planSchema, import_v49.z.undefined()]).optional(),
  featureQuantities: import_v49.z.array(previewAttachOutgoingFeatureQuantitySchema),
  effectiveAt: import_v49.z.number().nullable(),
  canceledAt: import_v49.z.number().nullable(),
  expiresAt: import_v49.z.number().nullable()
});
var previewAttachCheckoutTypeSchema = openEnumSchema4;
var previewAttachResponseSchema = import_v49.z.object({
  customerId: import_v49.z.string(),
  lineItems: import_v49.z.array(previewAttachLineItemSchema),
  subtotal: import_v49.z.number(),
  total: import_v49.z.number(),
  currency: import_v49.z.string(),
  nextCycle: import_v49.z.union([previewAttachNextCycleSchema, import_v49.z.undefined()]).optional(),
  expand: import_v49.z.union([import_v49.z.array(import_v49.z.string()), import_v49.z.undefined()]).optional(),
  incoming: import_v49.z.array(previewAttachIncomingSchema),
  outgoing: import_v49.z.array(previewAttachOutgoingSchema),
  redirectToCheckout: import_v49.z.boolean(),
  checkoutType: previewAttachCheckoutTypeSchema.nullable()
});

// src/generated/previewMultiAttachSchemas.ts
var import_v410 = require("zod/v4");
var previewMultiAttachGlobalsSchema = import_v410.z.object({
  xApiVersion: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachToSchema = import_v410.z.union([import_v410.z.number(), import_v410.z.string()]);
var previewMultiAttachTierSchema = import_v410.z.object({
  to: import_v410.z.union([import_v410.z.number(), import_v410.z.string()]),
  amount: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  flatAmount: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachPlanFeatureQuantitySchema = import_v410.z.object({
  featureId: import_v410.z.string(),
  quantity: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  adjustable: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachInvoiceModeSchema = import_v410.z.object({
  enabled: import_v410.z.boolean(),
  enablePlanImmediately: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional(),
  finalize: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachAttachDiscountSchema = import_v410.z.object({
  rewardId: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  promotionCode: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachSpendLimitSchema = import_v410.z.object({
  featureId: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  enabled: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional(),
  overageLimit: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachDiscountSchema = import_v410.z.object({
  amountOff: import_v410.z.number(),
  percentOff: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  rewardId: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  rewardName: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachLineItemPeriodSchema = import_v410.z.object({
  start: import_v410.z.number(),
  end: import_v410.z.number()
});
var previewMultiAttachLineItemSchema = import_v410.z.object({
  displayName: import_v410.z.string(),
  description: import_v410.z.string(),
  subtotal: import_v410.z.number(),
  total: import_v410.z.number(),
  discounts: import_v410.z.union([import_v410.z.array(previewMultiAttachDiscountSchema), import_v410.z.undefined()]).optional(),
  planId: import_v410.z.string(),
  featureId: import_v410.z.string().nullable(),
  period: import_v410.z.union([previewMultiAttachLineItemPeriodSchema, import_v410.z.undefined()]).optional(),
  quantity: import_v410.z.number()
});
var previewMultiAttachNextCycleDiscountSchema = import_v410.z.object({
  amountOff: import_v410.z.number(),
  percentOff: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  rewardId: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  rewardName: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachNextCycleLineItemPeriodSchema = import_v410.z.object({
  start: import_v410.z.number(),
  end: import_v410.z.number()
});
var previewMultiAttachNextCycleLineItemSchema = import_v410.z.object({
  displayName: import_v410.z.string(),
  description: import_v410.z.string(),
  subtotal: import_v410.z.number(),
  total: import_v410.z.number(),
  discounts: import_v410.z.union([import_v410.z.array(previewMultiAttachNextCycleDiscountSchema), import_v410.z.undefined()]).optional(),
  planId: import_v410.z.string(),
  featureId: import_v410.z.string().nullable(),
  period: import_v410.z.union([previewMultiAttachNextCycleLineItemPeriodSchema, import_v410.z.undefined()]).optional(),
  quantity: import_v410.z.number()
});
var previewMultiAttachUsageLineItemPeriodSchema = import_v410.z.object({
  start: import_v410.z.number(),
  end: import_v410.z.number()
});
var previewMultiAttachUsageLineItemSchema = import_v410.z.object({
  displayName: import_v410.z.string(),
  planId: import_v410.z.string(),
  featureId: import_v410.z.string().nullable(),
  period: import_v410.z.union([previewMultiAttachUsageLineItemPeriodSchema, import_v410.z.undefined()]).optional()
});
var previewMultiAttachNextCycleSchema = import_v410.z.object({
  startsAt: import_v410.z.number(),
  subtotal: import_v410.z.number(),
  total: import_v410.z.number(),
  lineItems: import_v410.z.array(previewMultiAttachNextCycleLineItemSchema),
  usageLineItems: import_v410.z.array(previewMultiAttachUsageLineItemSchema)
});
var previewMultiAttachIncomingFeatureQuantitySchema = import_v410.z.object({
  featureId: import_v410.z.string(),
  quantity: import_v410.z.number()
});
var previewMultiAttachOutgoingFeatureQuantitySchema = import_v410.z.object({
  featureId: import_v410.z.string(),
  quantity: import_v410.z.number()
});
var previewMultiAttachBasePriceOutboundSchema = import_v410.z.object({
  amount: import_v410.z.number(),
  interval: import_v410.z.string(),
  interval_count: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachResetOutboundSchema = import_v410.z.object({
  interval: import_v410.z.string(),
  interval_count: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachToOutboundSchema = import_v410.z.union([
  import_v410.z.number(),
  import_v410.z.string()
]);
var previewMultiAttachTierOutboundSchema = import_v410.z.object({
  to: import_v410.z.union([import_v410.z.number(), import_v410.z.string()]),
  amount: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  flat_amount: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachPriceOutboundSchema = import_v410.z.object({
  amount: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  tiers: import_v410.z.union([import_v410.z.array(previewMultiAttachTierOutboundSchema), import_v410.z.undefined()]).optional(),
  tier_behavior: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  interval: import_v410.z.string(),
  interval_count: import_v410.z.number(),
  billing_units: import_v410.z.number(),
  billing_method: import_v410.z.string(),
  max_purchase: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachProrationOutboundSchema = import_v410.z.object({
  on_increase: import_v410.z.string(),
  on_decrease: import_v410.z.string()
});
var previewMultiAttachRolloverOutboundSchema = import_v410.z.object({
  max: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  expiry_duration_type: import_v410.z.string(),
  expiry_duration_length: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachPlanItemOutboundSchema = import_v410.z.object({
  feature_id: import_v410.z.string(),
  included: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  unlimited: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional(),
  reset: import_v410.z.union([previewMultiAttachResetOutboundSchema, import_v410.z.undefined()]).optional(),
  price: import_v410.z.union([previewMultiAttachPriceOutboundSchema, import_v410.z.undefined()]).optional(),
  proration: import_v410.z.union([previewMultiAttachProrationOutboundSchema, import_v410.z.undefined()]).optional(),
  rollover: import_v410.z.union([previewMultiAttachRolloverOutboundSchema, import_v410.z.undefined()]).optional()
});
var previewMultiAttachCustomizeOutboundSchema = import_v410.z.object({
  price: import_v410.z.union([previewMultiAttachBasePriceOutboundSchema, import_v410.z.undefined()]).optional().nullable(),
  items: import_v410.z.union([import_v410.z.array(previewMultiAttachPlanItemOutboundSchema), import_v410.z.undefined()]).optional()
});
var previewMultiAttachPlanFeatureQuantityOutboundSchema = import_v410.z.object({
  feature_id: import_v410.z.string(),
  quantity: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  adjustable: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachPlanOutboundSchema = import_v410.z.object({
  plan_id: import_v410.z.string(),
  customize: import_v410.z.union([previewMultiAttachCustomizeOutboundSchema, import_v410.z.undefined()]).optional(),
  feature_quantities: import_v410.z.union([
    import_v410.z.array(previewMultiAttachPlanFeatureQuantityOutboundSchema),
    import_v410.z.undefined()
  ]).optional(),
  version: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  subscription_id: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachFreeTrialParamsOutboundSchema = import_v410.z.object({
  duration_length: import_v410.z.number(),
  duration_type: import_v410.z.string(),
  card_required: import_v410.z.boolean()
});
var previewMultiAttachInvoiceModeOutboundSchema = import_v410.z.object({
  enabled: import_v410.z.boolean(),
  enable_plan_immediately: import_v410.z.boolean(),
  finalize: import_v410.z.boolean()
});
var previewMultiAttachAttachDiscountOutboundSchema = import_v410.z.object({
  reward_id: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  promotion_code: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachSpendLimitOutboundSchema = import_v410.z.object({
  feature_id: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  enabled: import_v410.z.boolean(),
  overage_limit: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachUsageAlertOutboundSchema = import_v410.z.object({
  feature_id: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  enabled: import_v410.z.boolean(),
  threshold: import_v410.z.number(),
  threshold_type: import_v410.z.string(),
  name: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachBillingControlsOutboundSchema = import_v410.z.object({
  spend_limits: import_v410.z.union([import_v410.z.array(previewMultiAttachSpendLimitOutboundSchema), import_v410.z.undefined()]).optional(),
  usage_alerts: import_v410.z.union([import_v410.z.array(previewMultiAttachUsageAlertOutboundSchema), import_v410.z.undefined()]).optional()
});
var previewMultiAttachEntityDataOutboundSchema = import_v410.z.object({
  feature_id: import_v410.z.string(),
  name: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  billing_controls: import_v410.z.union([previewMultiAttachBillingControlsOutboundSchema, import_v410.z.undefined()]).optional()
});
var closedEnumSchema6 = import_v410.z.any();
var customerDataSchema2 = import_v410.z.any();
var planSchema2 = import_v410.z.any();
var openEnumSchema5 = import_v410.z.any();
var customerDataOutboundSchema2 = import_v410.z.any();
var previewMultiAttachPriceIntervalSchema = closedEnumSchema6;
var previewMultiAttachBasePriceSchema = import_v410.z.object({
  amount: import_v410.z.number(),
  interval: previewMultiAttachPriceIntervalSchema,
  intervalCount: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachResetIntervalSchema = closedEnumSchema6;
var previewMultiAttachResetSchema = import_v410.z.object({
  interval: previewMultiAttachResetIntervalSchema,
  intervalCount: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachTierBehaviorSchema = closedEnumSchema6;
var previewMultiAttachItemPriceIntervalSchema = closedEnumSchema6;
var previewMultiAttachBillingMethodSchema = closedEnumSchema6;
var previewMultiAttachPriceSchema = import_v410.z.object({
  amount: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  tiers: import_v410.z.union([import_v410.z.array(previewMultiAttachTierSchema), import_v410.z.undefined()]).optional(),
  tierBehavior: import_v410.z.union([previewMultiAttachTierBehaviorSchema, import_v410.z.undefined()]).optional(),
  interval: previewMultiAttachItemPriceIntervalSchema,
  intervalCount: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  billingUnits: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  billingMethod: previewMultiAttachBillingMethodSchema,
  maxPurchase: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachOnIncreaseSchema = closedEnumSchema6;
var previewMultiAttachOnDecreaseSchema = closedEnumSchema6;
var previewMultiAttachProrationSchema = import_v410.z.object({
  onIncrease: previewMultiAttachOnIncreaseSchema,
  onDecrease: previewMultiAttachOnDecreaseSchema
});
var previewMultiAttachExpiryDurationTypeSchema = closedEnumSchema6;
var previewMultiAttachRolloverSchema = import_v410.z.object({
  max: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  expiryDurationType: previewMultiAttachExpiryDurationTypeSchema,
  expiryDurationLength: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachPlanItemSchema = import_v410.z.object({
  featureId: import_v410.z.string(),
  included: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  unlimited: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional(),
  reset: import_v410.z.union([previewMultiAttachResetSchema, import_v410.z.undefined()]).optional(),
  price: import_v410.z.union([previewMultiAttachPriceSchema, import_v410.z.undefined()]).optional(),
  proration: import_v410.z.union([previewMultiAttachProrationSchema, import_v410.z.undefined()]).optional(),
  rollover: import_v410.z.union([previewMultiAttachRolloverSchema, import_v410.z.undefined()]).optional()
});
var previewMultiAttachCustomizeSchema = import_v410.z.object({
  price: import_v410.z.union([previewMultiAttachBasePriceSchema, import_v410.z.undefined()]).optional().nullable(),
  items: import_v410.z.union([import_v410.z.array(previewMultiAttachPlanItemSchema), import_v410.z.undefined()]).optional()
});
var previewMultiAttachPlanSchema = import_v410.z.object({
  planId: import_v410.z.string(),
  customize: import_v410.z.union([previewMultiAttachCustomizeSchema, import_v410.z.undefined()]).optional(),
  featureQuantities: import_v410.z.union([
    import_v410.z.array(previewMultiAttachPlanFeatureQuantitySchema),
    import_v410.z.undefined()
  ]).optional(),
  version: import_v410.z.union([import_v410.z.number(), import_v410.z.undefined()]).optional(),
  subscriptionId: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachDurationTypeSchema = closedEnumSchema6;
var previewMultiAttachFreeTrialParamsSchema = import_v410.z.object({
  durationLength: import_v410.z.number(),
  durationType: import_v410.z.union([previewMultiAttachDurationTypeSchema, import_v410.z.undefined()]).optional(),
  cardRequired: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachRedirectModeSchema = closedEnumSchema6;
var previewMultiAttachThresholdTypeSchema = closedEnumSchema6;
var previewMultiAttachUsageAlertSchema = import_v410.z.object({
  featureId: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  enabled: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional(),
  threshold: import_v410.z.number(),
  thresholdType: previewMultiAttachThresholdTypeSchema,
  name: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional()
});
var previewMultiAttachBillingControlsSchema = import_v410.z.object({
  spendLimits: import_v410.z.union([import_v410.z.array(previewMultiAttachSpendLimitSchema), import_v410.z.undefined()]).optional(),
  usageAlerts: import_v410.z.union([import_v410.z.array(previewMultiAttachUsageAlertSchema), import_v410.z.undefined()]).optional()
});
var previewMultiAttachEntityDataSchema = import_v410.z.object({
  featureId: import_v410.z.string(),
  name: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  billingControls: import_v410.z.union([previewMultiAttachBillingControlsSchema, import_v410.z.undefined()]).optional()
});
var previewMultiAttachParamsSchema = import_v410.z.object({
  customerId: import_v410.z.string(),
  entityId: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  plans: import_v410.z.array(previewMultiAttachPlanSchema),
  freeTrial: import_v410.z.union([previewMultiAttachFreeTrialParamsSchema, import_v410.z.undefined()]).optional().nullable(),
  invoiceMode: import_v410.z.union([previewMultiAttachInvoiceModeSchema, import_v410.z.undefined()]).optional(),
  discounts: import_v410.z.union([import_v410.z.array(previewMultiAttachAttachDiscountSchema), import_v410.z.undefined()]).optional(),
  successUrl: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  checkoutSessionParams: import_v410.z.union([import_v410.z.record(import_v410.z.string(), import_v410.z.any()), import_v410.z.undefined()]).optional(),
  redirectMode: import_v410.z.union([previewMultiAttachRedirectModeSchema, import_v410.z.undefined()]).optional(),
  newBillingSubscription: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional(),
  customerData: import_v410.z.union([customerDataSchema2, import_v410.z.undefined()]).optional(),
  entityData: import_v410.z.union([previewMultiAttachEntityDataSchema, import_v410.z.undefined()]).optional()
});
var previewMultiAttachIncomingSchema = import_v410.z.object({
  planId: import_v410.z.string(),
  plan: import_v410.z.union([planSchema2, import_v410.z.undefined()]).optional(),
  featureQuantities: import_v410.z.array(previewMultiAttachIncomingFeatureQuantitySchema),
  effectiveAt: import_v410.z.number().nullable(),
  canceledAt: import_v410.z.number().nullable(),
  expiresAt: import_v410.z.number().nullable()
});
var previewMultiAttachOutgoingSchema = import_v410.z.object({
  planId: import_v410.z.string(),
  plan: import_v410.z.union([planSchema2, import_v410.z.undefined()]).optional(),
  featureQuantities: import_v410.z.array(previewMultiAttachOutgoingFeatureQuantitySchema),
  effectiveAt: import_v410.z.number().nullable(),
  canceledAt: import_v410.z.number().nullable(),
  expiresAt: import_v410.z.number().nullable()
});
var previewMultiAttachCheckoutTypeSchema = openEnumSchema5;
var previewMultiAttachResponseSchema = import_v410.z.object({
  customerId: import_v410.z.string(),
  lineItems: import_v410.z.array(previewMultiAttachLineItemSchema),
  subtotal: import_v410.z.number(),
  total: import_v410.z.number(),
  currency: import_v410.z.string(),
  nextCycle: import_v410.z.union([previewMultiAttachNextCycleSchema, import_v410.z.undefined()]).optional(),
  expand: import_v410.z.union([import_v410.z.array(import_v410.z.string()), import_v410.z.undefined()]).optional(),
  incoming: import_v410.z.array(previewMultiAttachIncomingSchema),
  outgoing: import_v410.z.array(previewMultiAttachOutgoingSchema),
  redirectToCheckout: import_v410.z.boolean(),
  checkoutType: previewMultiAttachCheckoutTypeSchema.nullable()
});
var previewMultiAttachParamsOutboundSchema = import_v410.z.object({
  customer_id: import_v410.z.string(),
  entity_id: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  plans: import_v410.z.array(previewMultiAttachPlanOutboundSchema),
  free_trial: import_v410.z.union([previewMultiAttachFreeTrialParamsOutboundSchema, import_v410.z.undefined()]).optional().nullable(),
  invoice_mode: import_v410.z.union([previewMultiAttachInvoiceModeOutboundSchema, import_v410.z.undefined()]).optional(),
  discounts: import_v410.z.union([
    import_v410.z.array(previewMultiAttachAttachDiscountOutboundSchema),
    import_v410.z.undefined()
  ]).optional(),
  success_url: import_v410.z.union([import_v410.z.string(), import_v410.z.undefined()]).optional(),
  checkout_session_params: import_v410.z.union([import_v410.z.record(import_v410.z.string(), import_v410.z.any()), import_v410.z.undefined()]).optional(),
  redirect_mode: import_v410.z.string(),
  new_billing_subscription: import_v410.z.union([import_v410.z.boolean(), import_v410.z.undefined()]).optional(),
  customer_data: import_v410.z.union([customerDataOutboundSchema2, import_v410.z.undefined()]).optional(),
  entity_data: import_v410.z.union([previewMultiAttachEntityDataOutboundSchema, import_v410.z.undefined()]).optional()
});

// src/generated/previewUpdateSubscriptionSchemas.ts
var import_v411 = require("zod/v4");
var previewUpdateGlobalsSchema = import_v411.z.object({
  xApiVersion: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional()
});
var previewUpdateFeatureQuantityRequestSchema = import_v411.z.object({
  featureId: import_v411.z.string(),
  quantity: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  adjustable: import_v411.z.union([import_v411.z.boolean(), import_v411.z.undefined()]).optional()
});
var previewUpdateToSchema = import_v411.z.union([import_v411.z.number(), import_v411.z.string()]);
var previewUpdateTierSchema = import_v411.z.object({
  to: import_v411.z.union([import_v411.z.number(), import_v411.z.string()]),
  amount: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  flatAmount: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdateInvoiceModeSchema = import_v411.z.object({
  enabled: import_v411.z.boolean(),
  enablePlanImmediately: import_v411.z.union([import_v411.z.boolean(), import_v411.z.undefined()]).optional(),
  finalize: import_v411.z.union([import_v411.z.boolean(), import_v411.z.undefined()]).optional()
});
var previewUpdateRecalculateBalancesSchema = import_v411.z.object({
  enabled: import_v411.z.boolean()
});
var previewUpdateDiscountSchema = import_v411.z.object({
  amountOff: import_v411.z.number(),
  percentOff: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  rewardId: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  rewardName: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional()
});
var previewUpdateLineItemPeriodSchema = import_v411.z.object({
  start: import_v411.z.number(),
  end: import_v411.z.number()
});
var previewUpdateLineItemSchema = import_v411.z.object({
  displayName: import_v411.z.string(),
  description: import_v411.z.string(),
  subtotal: import_v411.z.number(),
  total: import_v411.z.number(),
  discounts: import_v411.z.union([import_v411.z.array(previewUpdateDiscountSchema), import_v411.z.undefined()]).optional(),
  planId: import_v411.z.string(),
  featureId: import_v411.z.string().nullable(),
  period: import_v411.z.union([previewUpdateLineItemPeriodSchema, import_v411.z.undefined()]).optional(),
  quantity: import_v411.z.number()
});
var previewUpdateNextCycleDiscountSchema = import_v411.z.object({
  amountOff: import_v411.z.number(),
  percentOff: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  rewardId: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  rewardName: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional()
});
var previewUpdateNextCycleLineItemPeriodSchema = import_v411.z.object({
  start: import_v411.z.number(),
  end: import_v411.z.number()
});
var previewUpdateNextCycleLineItemSchema = import_v411.z.object({
  displayName: import_v411.z.string(),
  description: import_v411.z.string(),
  subtotal: import_v411.z.number(),
  total: import_v411.z.number(),
  discounts: import_v411.z.union([import_v411.z.array(previewUpdateNextCycleDiscountSchema), import_v411.z.undefined()]).optional(),
  planId: import_v411.z.string(),
  featureId: import_v411.z.string().nullable(),
  period: import_v411.z.union([previewUpdateNextCycleLineItemPeriodSchema, import_v411.z.undefined()]).optional(),
  quantity: import_v411.z.number()
});
var previewUpdateUsageLineItemPeriodSchema = import_v411.z.object({
  start: import_v411.z.number(),
  end: import_v411.z.number()
});
var previewUpdateUsageLineItemSchema = import_v411.z.object({
  displayName: import_v411.z.string(),
  planId: import_v411.z.string(),
  featureId: import_v411.z.string().nullable(),
  period: import_v411.z.union([previewUpdateUsageLineItemPeriodSchema, import_v411.z.undefined()]).optional()
});
var previewUpdateNextCycleSchema = import_v411.z.object({
  startsAt: import_v411.z.number(),
  subtotal: import_v411.z.number(),
  total: import_v411.z.number(),
  lineItems: import_v411.z.array(previewUpdateNextCycleLineItemSchema),
  usageLineItems: import_v411.z.array(previewUpdateUsageLineItemSchema)
});
var previewUpdateIncomingFeatureQuantitySchema = import_v411.z.object({
  featureId: import_v411.z.string(),
  quantity: import_v411.z.number()
});
var previewUpdateOutgoingFeatureQuantitySchema = import_v411.z.object({
  featureId: import_v411.z.string(),
  quantity: import_v411.z.number()
});
var previewUpdateFeatureQuantityRequestOutboundSchema = import_v411.z.object({
  feature_id: import_v411.z.string(),
  quantity: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  adjustable: import_v411.z.union([import_v411.z.boolean(), import_v411.z.undefined()]).optional()
});
var previewUpdateBasePriceOutboundSchema = import_v411.z.object({
  amount: import_v411.z.number(),
  interval: import_v411.z.string(),
  interval_count: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdateResetOutboundSchema = import_v411.z.object({
  interval: import_v411.z.string(),
  interval_count: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdateToOutboundSchema = import_v411.z.union([import_v411.z.number(), import_v411.z.string()]);
var previewUpdateTierOutboundSchema = import_v411.z.object({
  to: import_v411.z.union([import_v411.z.number(), import_v411.z.string()]),
  amount: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  flat_amount: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdatePriceOutboundSchema = import_v411.z.object({
  amount: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  tiers: import_v411.z.union([import_v411.z.array(previewUpdateTierOutboundSchema), import_v411.z.undefined()]).optional(),
  tier_behavior: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  interval: import_v411.z.string(),
  interval_count: import_v411.z.number(),
  billing_units: import_v411.z.number(),
  billing_method: import_v411.z.string(),
  max_purchase: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdateProrationOutboundSchema = import_v411.z.object({
  on_increase: import_v411.z.string(),
  on_decrease: import_v411.z.string()
});
var previewUpdateRolloverOutboundSchema = import_v411.z.object({
  max: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  expiry_duration_type: import_v411.z.string(),
  expiry_duration_length: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdatePlanItemOutboundSchema = import_v411.z.object({
  feature_id: import_v411.z.string(),
  included: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  unlimited: import_v411.z.union([import_v411.z.boolean(), import_v411.z.undefined()]).optional(),
  reset: import_v411.z.union([previewUpdateResetOutboundSchema, import_v411.z.undefined()]).optional(),
  price: import_v411.z.union([previewUpdatePriceOutboundSchema, import_v411.z.undefined()]).optional(),
  proration: import_v411.z.union([previewUpdateProrationOutboundSchema, import_v411.z.undefined()]).optional(),
  rollover: import_v411.z.union([previewUpdateRolloverOutboundSchema, import_v411.z.undefined()]).optional()
});
var previewUpdateFreeTrialParamsOutboundSchema = import_v411.z.object({
  duration_length: import_v411.z.number(),
  duration_type: import_v411.z.string(),
  card_required: import_v411.z.boolean()
});
var previewUpdateCustomizeOutboundSchema = import_v411.z.object({
  price: import_v411.z.union([previewUpdateBasePriceOutboundSchema, import_v411.z.undefined()]).optional().nullable(),
  items: import_v411.z.union([import_v411.z.array(previewUpdatePlanItemOutboundSchema), import_v411.z.undefined()]).optional(),
  free_trial: import_v411.z.union([previewUpdateFreeTrialParamsOutboundSchema, import_v411.z.undefined()]).optional().nullable()
});
var previewUpdateInvoiceModeOutboundSchema = import_v411.z.object({
  enabled: import_v411.z.boolean(),
  enable_plan_immediately: import_v411.z.boolean(),
  finalize: import_v411.z.boolean()
});
var previewUpdateRecalculateBalancesOutboundSchema = import_v411.z.object({
  enabled: import_v411.z.boolean()
});
var previewUpdateParamsOutboundSchema = import_v411.z.object({
  customer_id: import_v411.z.string(),
  entity_id: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  plan_id: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  feature_quantities: import_v411.z.union([
    import_v411.z.array(previewUpdateFeatureQuantityRequestOutboundSchema),
    import_v411.z.undefined()
  ]).optional(),
  version: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  customize: import_v411.z.union([previewUpdateCustomizeOutboundSchema, import_v411.z.undefined()]).optional(),
  invoice_mode: import_v411.z.union([previewUpdateInvoiceModeOutboundSchema, import_v411.z.undefined()]).optional(),
  proration_behavior: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  redirect_mode: import_v411.z.string(),
  subscription_id: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  cancel_action: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  no_billing_changes: import_v411.z.union([import_v411.z.boolean(), import_v411.z.undefined()]).optional(),
  recalculate_balances: import_v411.z.union([previewUpdateRecalculateBalancesOutboundSchema, import_v411.z.undefined()]).optional()
});
var closedEnumSchema7 = import_v411.z.any();
var planSchema3 = import_v411.z.any();
var openEnumSchema6 = import_v411.z.any();
var previewUpdatePriceIntervalSchema = closedEnumSchema7;
var previewUpdateBasePriceSchema = import_v411.z.object({
  amount: import_v411.z.number(),
  interval: previewUpdatePriceIntervalSchema,
  intervalCount: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdateResetIntervalSchema = closedEnumSchema7;
var previewUpdateResetSchema = import_v411.z.object({
  interval: previewUpdateResetIntervalSchema,
  intervalCount: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdateTierBehaviorSchema = closedEnumSchema7;
var previewUpdateItemPriceIntervalSchema = closedEnumSchema7;
var previewUpdateBillingMethodSchema = closedEnumSchema7;
var previewUpdatePriceSchema = import_v411.z.object({
  amount: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  tiers: import_v411.z.union([import_v411.z.array(previewUpdateTierSchema), import_v411.z.undefined()]).optional(),
  tierBehavior: import_v411.z.union([previewUpdateTierBehaviorSchema, import_v411.z.undefined()]).optional(),
  interval: previewUpdateItemPriceIntervalSchema,
  intervalCount: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  billingUnits: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  billingMethod: previewUpdateBillingMethodSchema,
  maxPurchase: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdateOnIncreaseSchema = closedEnumSchema7;
var previewUpdateOnDecreaseSchema = closedEnumSchema7;
var previewUpdateProrationSchema = import_v411.z.object({
  onIncrease: previewUpdateOnIncreaseSchema,
  onDecrease: previewUpdateOnDecreaseSchema
});
var previewUpdateExpiryDurationTypeSchema = closedEnumSchema7;
var previewUpdateRolloverSchema = import_v411.z.object({
  max: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  expiryDurationType: previewUpdateExpiryDurationTypeSchema,
  expiryDurationLength: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional()
});
var previewUpdatePlanItemSchema = import_v411.z.object({
  featureId: import_v411.z.string(),
  included: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  unlimited: import_v411.z.union([import_v411.z.boolean(), import_v411.z.undefined()]).optional(),
  reset: import_v411.z.union([previewUpdateResetSchema, import_v411.z.undefined()]).optional(),
  price: import_v411.z.union([previewUpdatePriceSchema, import_v411.z.undefined()]).optional(),
  proration: import_v411.z.union([previewUpdateProrationSchema, import_v411.z.undefined()]).optional(),
  rollover: import_v411.z.union([previewUpdateRolloverSchema, import_v411.z.undefined()]).optional()
});
var previewUpdateDurationTypeSchema = closedEnumSchema7;
var previewUpdateFreeTrialParamsSchema = import_v411.z.object({
  durationLength: import_v411.z.number(),
  durationType: import_v411.z.union([previewUpdateDurationTypeSchema, import_v411.z.undefined()]).optional(),
  cardRequired: import_v411.z.union([import_v411.z.boolean(), import_v411.z.undefined()]).optional()
});
var previewUpdateCustomizeSchema = import_v411.z.object({
  price: import_v411.z.union([previewUpdateBasePriceSchema, import_v411.z.undefined()]).optional().nullable(),
  items: import_v411.z.union([import_v411.z.array(previewUpdatePlanItemSchema), import_v411.z.undefined()]).optional(),
  freeTrial: import_v411.z.union([previewUpdateFreeTrialParamsSchema, import_v411.z.undefined()]).optional().nullable()
});
var previewUpdateProrationBehaviorSchema = closedEnumSchema7;
var previewUpdateRedirectModeSchema = closedEnumSchema7;
var previewUpdateCancelActionSchema = closedEnumSchema7;
var previewUpdateParamsSchema = import_v411.z.object({
  customerId: import_v411.z.string(),
  entityId: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  planId: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  featureQuantities: import_v411.z.union([import_v411.z.array(previewUpdateFeatureQuantityRequestSchema), import_v411.z.undefined()]).optional(),
  version: import_v411.z.union([import_v411.z.number(), import_v411.z.undefined()]).optional(),
  customize: import_v411.z.union([previewUpdateCustomizeSchema, import_v411.z.undefined()]).optional(),
  invoiceMode: import_v411.z.union([previewUpdateInvoiceModeSchema, import_v411.z.undefined()]).optional(),
  prorationBehavior: import_v411.z.union([previewUpdateProrationBehaviorSchema, import_v411.z.undefined()]).optional(),
  redirectMode: import_v411.z.union([previewUpdateRedirectModeSchema, import_v411.z.undefined()]).optional(),
  subscriptionId: import_v411.z.union([import_v411.z.string(), import_v411.z.undefined()]).optional(),
  cancelAction: import_v411.z.union([previewUpdateCancelActionSchema, import_v411.z.undefined()]).optional(),
  noBillingChanges: import_v411.z.union([import_v411.z.boolean(), import_v411.z.undefined()]).optional(),
  recalculateBalances: import_v411.z.union([previewUpdateRecalculateBalancesSchema, import_v411.z.undefined()]).optional()
});
var previewUpdateIncomingSchema = import_v411.z.object({
  planId: import_v411.z.string(),
  plan: import_v411.z.union([planSchema3, import_v411.z.undefined()]).optional(),
  featureQuantities: import_v411.z.array(previewUpdateIncomingFeatureQuantitySchema),
  effectiveAt: import_v411.z.number().nullable(),
  canceledAt: import_v411.z.number().nullable(),
  expiresAt: import_v411.z.number().nullable()
});
var previewUpdateOutgoingSchema = import_v411.z.object({
  planId: import_v411.z.string(),
  plan: import_v411.z.union([planSchema3, import_v411.z.undefined()]).optional(),
  featureQuantities: import_v411.z.array(previewUpdateOutgoingFeatureQuantitySchema),
  effectiveAt: import_v411.z.number().nullable(),
  canceledAt: import_v411.z.number().nullable(),
  expiresAt: import_v411.z.number().nullable()
});
var intentSchema = openEnumSchema6;
var previewUpdateResponseSchema = import_v411.z.object({
  customerId: import_v411.z.string(),
  lineItems: import_v411.z.array(previewUpdateLineItemSchema),
  subtotal: import_v411.z.number(),
  total: import_v411.z.number(),
  currency: import_v411.z.string(),
  nextCycle: import_v411.z.union([previewUpdateNextCycleSchema, import_v411.z.undefined()]).optional(),
  expand: import_v411.z.union([import_v411.z.array(import_v411.z.string()), import_v411.z.undefined()]).optional(),
  incoming: import_v411.z.array(previewUpdateIncomingSchema),
  outgoing: import_v411.z.array(previewUpdateOutgoingSchema),
  intent: intentSchema
});

// src/generated/redeemReferralCodeSchemas.ts
var import_v412 = require("zod/v4");
var redeemReferralCodeGlobalsSchema = import_v412.z.object({
  xApiVersion: import_v412.z.union([import_v412.z.string(), import_v412.z.undefined()]).optional()
});
var redeemReferralCodeParamsSchema = import_v412.z.object({
  code: import_v412.z.string(),
  customerId: import_v412.z.string()
});
var redeemReferralCodeResponseSchema = import_v412.z.object({
  id: import_v412.z.string(),
  customerId: import_v412.z.string(),
  rewardId: import_v412.z.string()
});
var redeemReferralCodeParamsOutboundSchema = import_v412.z.object({
  code: import_v412.z.string(),
  customer_id: import_v412.z.string()
});

// src/generated/setupPaymentSchemas.ts
var import_v413 = require("zod/v4");
var setupPaymentGlobalsSchema = import_v413.z.object({
  xApiVersion: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional()
});
var setupPaymentFeatureQuantitySchema = import_v413.z.object({
  featureId: import_v413.z.string(),
  quantity: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  adjustable: import_v413.z.union([import_v413.z.boolean(), import_v413.z.undefined()]).optional()
});
var setupPaymentToSchema = import_v413.z.union([import_v413.z.number(), import_v413.z.string()]);
var setupPaymentTierSchema = import_v413.z.object({
  to: import_v413.z.union([import_v413.z.number(), import_v413.z.string()]),
  amount: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  flatAmount: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentAttachDiscountSchema = import_v413.z.object({
  rewardId: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  promotionCode: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional()
});
var setupPaymentCustomLineItemSchema = import_v413.z.object({
  amount: import_v413.z.number(),
  description: import_v413.z.string()
});
var setupPaymentCarryOverBalancesSchema = import_v413.z.object({
  enabled: import_v413.z.boolean(),
  featureIds: import_v413.z.union([import_v413.z.array(import_v413.z.string()), import_v413.z.undefined()]).optional()
});
var setupPaymentCarryOverUsagesSchema = import_v413.z.object({
  enabled: import_v413.z.boolean(),
  featureIds: import_v413.z.union([import_v413.z.array(import_v413.z.string()), import_v413.z.undefined()]).optional()
});
var setupPaymentResponseSchema = import_v413.z.object({
  customerId: import_v413.z.string(),
  entityId: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  url: import_v413.z.string()
});
var setupPaymentFeatureQuantityOutboundSchema = import_v413.z.object({
  feature_id: import_v413.z.string(),
  quantity: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  adjustable: import_v413.z.union([import_v413.z.boolean(), import_v413.z.undefined()]).optional()
});
var setupPaymentBasePriceOutboundSchema = import_v413.z.object({
  amount: import_v413.z.number(),
  interval: import_v413.z.string(),
  interval_count: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentResetOutboundSchema = import_v413.z.object({
  interval: import_v413.z.string(),
  interval_count: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentToOutboundSchema = import_v413.z.union([import_v413.z.number(), import_v413.z.string()]);
var setupPaymentTierOutboundSchema = import_v413.z.object({
  to: import_v413.z.union([import_v413.z.number(), import_v413.z.string()]),
  amount: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  flat_amount: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentPriceOutboundSchema = import_v413.z.object({
  amount: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  tiers: import_v413.z.union([import_v413.z.array(setupPaymentTierOutboundSchema), import_v413.z.undefined()]).optional(),
  tier_behavior: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  interval: import_v413.z.string(),
  interval_count: import_v413.z.number(),
  billing_units: import_v413.z.number(),
  billing_method: import_v413.z.string(),
  max_purchase: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentProrationOutboundSchema = import_v413.z.object({
  on_increase: import_v413.z.string(),
  on_decrease: import_v413.z.string()
});
var setupPaymentRolloverOutboundSchema = import_v413.z.object({
  max: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  expiry_duration_type: import_v413.z.string(),
  expiry_duration_length: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentPlanItemOutboundSchema = import_v413.z.object({
  feature_id: import_v413.z.string(),
  included: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  unlimited: import_v413.z.union([import_v413.z.boolean(), import_v413.z.undefined()]).optional(),
  reset: import_v413.z.union([setupPaymentResetOutboundSchema, import_v413.z.undefined()]).optional(),
  price: import_v413.z.union([setupPaymentPriceOutboundSchema, import_v413.z.undefined()]).optional(),
  proration: import_v413.z.union([setupPaymentProrationOutboundSchema, import_v413.z.undefined()]).optional(),
  rollover: import_v413.z.union([setupPaymentRolloverOutboundSchema, import_v413.z.undefined()]).optional()
});
var setupPaymentFreeTrialParamsOutboundSchema = import_v413.z.object({
  duration_length: import_v413.z.number(),
  duration_type: import_v413.z.string(),
  card_required: import_v413.z.boolean()
});
var setupPaymentCustomizeOutboundSchema = import_v413.z.object({
  price: import_v413.z.union([setupPaymentBasePriceOutboundSchema, import_v413.z.undefined()]).optional().nullable(),
  items: import_v413.z.union([import_v413.z.array(setupPaymentPlanItemOutboundSchema), import_v413.z.undefined()]).optional(),
  free_trial: import_v413.z.union([setupPaymentFreeTrialParamsOutboundSchema, import_v413.z.undefined()]).optional().nullable()
});
var setupPaymentAttachDiscountOutboundSchema = import_v413.z.object({
  reward_id: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  promotion_code: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional()
});
var setupPaymentCustomLineItemOutboundSchema = import_v413.z.object({
  amount: import_v413.z.number(),
  description: import_v413.z.string()
});
var setupPaymentCarryOverBalancesOutboundSchema = import_v413.z.object({
  enabled: import_v413.z.boolean(),
  feature_ids: import_v413.z.union([import_v413.z.array(import_v413.z.string()), import_v413.z.undefined()]).optional()
});
var setupPaymentCarryOverUsagesOutboundSchema = import_v413.z.object({
  enabled: import_v413.z.boolean(),
  feature_ids: import_v413.z.union([import_v413.z.array(import_v413.z.string()), import_v413.z.undefined()]).optional()
});
var setupPaymentParamsOutboundSchema = import_v413.z.object({
  customer_id: import_v413.z.string(),
  entity_id: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  plan_id: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  feature_quantities: import_v413.z.union([import_v413.z.array(setupPaymentFeatureQuantityOutboundSchema), import_v413.z.undefined()]).optional(),
  version: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  customize: import_v413.z.union([setupPaymentCustomizeOutboundSchema, import_v413.z.undefined()]).optional(),
  proration_behavior: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  subscription_id: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  discounts: import_v413.z.union([import_v413.z.array(setupPaymentAttachDiscountOutboundSchema), import_v413.z.undefined()]).optional(),
  success_url: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  checkout_session_params: import_v413.z.union([import_v413.z.record(import_v413.z.string(), import_v413.z.any()), import_v413.z.undefined()]).optional(),
  custom_line_items: import_v413.z.union([import_v413.z.array(setupPaymentCustomLineItemOutboundSchema), import_v413.z.undefined()]).optional(),
  processor_subscription_id: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  carry_over_balances: import_v413.z.union([setupPaymentCarryOverBalancesOutboundSchema, import_v413.z.undefined()]).optional(),
  carry_over_usages: import_v413.z.union([setupPaymentCarryOverUsagesOutboundSchema, import_v413.z.undefined()]).optional(),
  metadata: import_v413.z.union([import_v413.z.record(import_v413.z.string(), import_v413.z.string()), import_v413.z.undefined()]).optional()
});
var closedEnumSchema8 = import_v413.z.any();
var setupPaymentPriceIntervalSchema = closedEnumSchema8;
var setupPaymentBasePriceSchema = import_v413.z.object({
  amount: import_v413.z.number(),
  interval: setupPaymentPriceIntervalSchema,
  intervalCount: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentResetIntervalSchema = closedEnumSchema8;
var setupPaymentResetSchema = import_v413.z.object({
  interval: setupPaymentResetIntervalSchema,
  intervalCount: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentTierBehaviorSchema = closedEnumSchema8;
var setupPaymentItemPriceIntervalSchema = closedEnumSchema8;
var setupPaymentBillingMethodSchema = closedEnumSchema8;
var setupPaymentPriceSchema = import_v413.z.object({
  amount: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  tiers: import_v413.z.union([import_v413.z.array(setupPaymentTierSchema), import_v413.z.undefined()]).optional(),
  tierBehavior: import_v413.z.union([setupPaymentTierBehaviorSchema, import_v413.z.undefined()]).optional(),
  interval: setupPaymentItemPriceIntervalSchema,
  intervalCount: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  billingUnits: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  billingMethod: setupPaymentBillingMethodSchema,
  maxPurchase: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentOnIncreaseSchema = closedEnumSchema8;
var setupPaymentOnDecreaseSchema = closedEnumSchema8;
var setupPaymentProrationSchema = import_v413.z.object({
  onIncrease: setupPaymentOnIncreaseSchema,
  onDecrease: setupPaymentOnDecreaseSchema
});
var setupPaymentExpiryDurationTypeSchema = closedEnumSchema8;
var setupPaymentRolloverSchema = import_v413.z.object({
  max: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  expiryDurationType: setupPaymentExpiryDurationTypeSchema,
  expiryDurationLength: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional()
});
var setupPaymentPlanItemSchema = import_v413.z.object({
  featureId: import_v413.z.string(),
  included: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  unlimited: import_v413.z.union([import_v413.z.boolean(), import_v413.z.undefined()]).optional(),
  reset: import_v413.z.union([setupPaymentResetSchema, import_v413.z.undefined()]).optional(),
  price: import_v413.z.union([setupPaymentPriceSchema, import_v413.z.undefined()]).optional(),
  proration: import_v413.z.union([setupPaymentProrationSchema, import_v413.z.undefined()]).optional(),
  rollover: import_v413.z.union([setupPaymentRolloverSchema, import_v413.z.undefined()]).optional()
});
var setupPaymentDurationTypeSchema = closedEnumSchema8;
var setupPaymentFreeTrialParamsSchema = import_v413.z.object({
  durationLength: import_v413.z.number(),
  durationType: import_v413.z.union([setupPaymentDurationTypeSchema, import_v413.z.undefined()]).optional(),
  cardRequired: import_v413.z.union([import_v413.z.boolean(), import_v413.z.undefined()]).optional()
});
var setupPaymentCustomizeSchema = import_v413.z.object({
  price: import_v413.z.union([setupPaymentBasePriceSchema, import_v413.z.undefined()]).optional().nullable(),
  items: import_v413.z.union([import_v413.z.array(setupPaymentPlanItemSchema), import_v413.z.undefined()]).optional(),
  freeTrial: import_v413.z.union([setupPaymentFreeTrialParamsSchema, import_v413.z.undefined()]).optional().nullable()
});
var setupPaymentProrationBehaviorSchema = closedEnumSchema8;
var setupPaymentParamsSchema = import_v413.z.object({
  customerId: import_v413.z.string(),
  entityId: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  planId: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  featureQuantities: import_v413.z.union([import_v413.z.array(setupPaymentFeatureQuantitySchema), import_v413.z.undefined()]).optional(),
  version: import_v413.z.union([import_v413.z.number(), import_v413.z.undefined()]).optional(),
  customize: import_v413.z.union([setupPaymentCustomizeSchema, import_v413.z.undefined()]).optional(),
  prorationBehavior: import_v413.z.union([setupPaymentProrationBehaviorSchema, import_v413.z.undefined()]).optional(),
  subscriptionId: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  discounts: import_v413.z.union([import_v413.z.array(setupPaymentAttachDiscountSchema), import_v413.z.undefined()]).optional(),
  successUrl: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  checkoutSessionParams: import_v413.z.union([import_v413.z.record(import_v413.z.string(), import_v413.z.any()), import_v413.z.undefined()]).optional(),
  customLineItems: import_v413.z.union([import_v413.z.array(setupPaymentCustomLineItemSchema), import_v413.z.undefined()]).optional(),
  processorSubscriptionId: import_v413.z.union([import_v413.z.string(), import_v413.z.undefined()]).optional(),
  carryOverBalances: import_v413.z.union([setupPaymentCarryOverBalancesSchema, import_v413.z.undefined()]).optional(),
  carryOverUsages: import_v413.z.union([setupPaymentCarryOverUsagesSchema, import_v413.z.undefined()]).optional(),
  metadata: import_v413.z.union([import_v413.z.record(import_v413.z.string(), import_v413.z.string()), import_v413.z.undefined()]).optional()
});

// src/generated/updateSubscriptionSchemas.ts
var import_v414 = require("zod/v4");
var billingUpdateGlobalsSchema = import_v414.z.object({
  xApiVersion: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional()
});
var billingUpdateFeatureQuantitySchema = import_v414.z.object({
  featureId: import_v414.z.string(),
  quantity: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  adjustable: import_v414.z.union([import_v414.z.boolean(), import_v414.z.undefined()]).optional()
});
var billingUpdateToSchema = import_v414.z.union([import_v414.z.number(), import_v414.z.string()]);
var billingUpdateTierSchema = import_v414.z.object({
  to: import_v414.z.union([import_v414.z.number(), import_v414.z.string()]),
  amount: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  flatAmount: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdateInvoiceModeSchema = import_v414.z.object({
  enabled: import_v414.z.boolean(),
  enablePlanImmediately: import_v414.z.union([import_v414.z.boolean(), import_v414.z.undefined()]).optional(),
  finalize: import_v414.z.union([import_v414.z.boolean(), import_v414.z.undefined()]).optional()
});
var billingUpdateRecalculateBalancesSchema = import_v414.z.object({
  enabled: import_v414.z.boolean()
});
var billingUpdateInvoiceSchema = import_v414.z.object({
  status: import_v414.z.string().nullable(),
  stripeId: import_v414.z.string(),
  total: import_v414.z.number(),
  currency: import_v414.z.string(),
  hostedInvoiceUrl: import_v414.z.string().nullable()
});
var billingUpdateFeatureQuantityOutboundSchema = import_v414.z.object({
  feature_id: import_v414.z.string(),
  quantity: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  adjustable: import_v414.z.union([import_v414.z.boolean(), import_v414.z.undefined()]).optional()
});
var billingUpdateBasePriceOutboundSchema = import_v414.z.object({
  amount: import_v414.z.number(),
  interval: import_v414.z.string(),
  interval_count: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdateResetOutboundSchema = import_v414.z.object({
  interval: import_v414.z.string(),
  interval_count: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdateToOutboundSchema = import_v414.z.union([import_v414.z.number(), import_v414.z.string()]);
var billingUpdateTierOutboundSchema = import_v414.z.object({
  to: import_v414.z.union([import_v414.z.number(), import_v414.z.string()]),
  amount: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  flat_amount: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdatePriceOutboundSchema = import_v414.z.object({
  amount: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  tiers: import_v414.z.union([import_v414.z.array(billingUpdateTierOutboundSchema), import_v414.z.undefined()]).optional(),
  tier_behavior: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  interval: import_v414.z.string(),
  interval_count: import_v414.z.number(),
  billing_units: import_v414.z.number(),
  billing_method: import_v414.z.string(),
  max_purchase: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdateProrationOutboundSchema = import_v414.z.object({
  on_increase: import_v414.z.string(),
  on_decrease: import_v414.z.string()
});
var billingUpdateRolloverOutboundSchema = import_v414.z.object({
  max: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  expiry_duration_type: import_v414.z.string(),
  expiry_duration_length: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdatePlanItemOutboundSchema = import_v414.z.object({
  feature_id: import_v414.z.string(),
  included: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  unlimited: import_v414.z.union([import_v414.z.boolean(), import_v414.z.undefined()]).optional(),
  reset: import_v414.z.union([billingUpdateResetOutboundSchema, import_v414.z.undefined()]).optional(),
  price: import_v414.z.union([billingUpdatePriceOutboundSchema, import_v414.z.undefined()]).optional(),
  proration: import_v414.z.union([billingUpdateProrationOutboundSchema, import_v414.z.undefined()]).optional(),
  rollover: import_v414.z.union([billingUpdateRolloverOutboundSchema, import_v414.z.undefined()]).optional()
});
var billingUpdateFreeTrialParamsOutboundSchema = import_v414.z.object({
  duration_length: import_v414.z.number(),
  duration_type: import_v414.z.string(),
  card_required: import_v414.z.boolean()
});
var billingUpdateCustomizeOutboundSchema = import_v414.z.object({
  price: import_v414.z.union([billingUpdateBasePriceOutboundSchema, import_v414.z.undefined()]).optional().nullable(),
  items: import_v414.z.union([import_v414.z.array(billingUpdatePlanItemOutboundSchema), import_v414.z.undefined()]).optional(),
  free_trial: import_v414.z.union([billingUpdateFreeTrialParamsOutboundSchema, import_v414.z.undefined()]).optional().nullable()
});
var billingUpdateInvoiceModeOutboundSchema = import_v414.z.object({
  enabled: import_v414.z.boolean(),
  enable_plan_immediately: import_v414.z.boolean(),
  finalize: import_v414.z.boolean()
});
var billingUpdateRecalculateBalancesOutboundSchema = import_v414.z.object({
  enabled: import_v414.z.boolean()
});
var updateSubscriptionParamsOutboundSchema = import_v414.z.object({
  customer_id: import_v414.z.string(),
  entity_id: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  plan_id: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  feature_quantities: import_v414.z.union([import_v414.z.array(billingUpdateFeatureQuantityOutboundSchema), import_v414.z.undefined()]).optional(),
  version: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  customize: import_v414.z.union([billingUpdateCustomizeOutboundSchema, import_v414.z.undefined()]).optional(),
  invoice_mode: import_v414.z.union([billingUpdateInvoiceModeOutboundSchema, import_v414.z.undefined()]).optional(),
  proration_behavior: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  redirect_mode: import_v414.z.string(),
  subscription_id: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  cancel_action: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  no_billing_changes: import_v414.z.union([import_v414.z.boolean(), import_v414.z.undefined()]).optional(),
  recalculate_balances: import_v414.z.union([billingUpdateRecalculateBalancesOutboundSchema, import_v414.z.undefined()]).optional()
});
var closedEnumSchema9 = import_v414.z.any();
var openEnumSchema7 = import_v414.z.any();
var billingUpdatePriceIntervalSchema = closedEnumSchema9;
var billingUpdateBasePriceSchema = import_v414.z.object({
  amount: import_v414.z.number(),
  interval: billingUpdatePriceIntervalSchema,
  intervalCount: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdateResetIntervalSchema = closedEnumSchema9;
var billingUpdateResetSchema = import_v414.z.object({
  interval: billingUpdateResetIntervalSchema,
  intervalCount: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdateTierBehaviorSchema = closedEnumSchema9;
var billingUpdateItemPriceIntervalSchema = closedEnumSchema9;
var billingUpdateBillingMethodSchema = closedEnumSchema9;
var billingUpdatePriceSchema = import_v414.z.object({
  amount: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  tiers: import_v414.z.union([import_v414.z.array(billingUpdateTierSchema), import_v414.z.undefined()]).optional(),
  tierBehavior: import_v414.z.union([billingUpdateTierBehaviorSchema, import_v414.z.undefined()]).optional(),
  interval: billingUpdateItemPriceIntervalSchema,
  intervalCount: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  billingUnits: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  billingMethod: billingUpdateBillingMethodSchema,
  maxPurchase: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdateOnIncreaseSchema = closedEnumSchema9;
var billingUpdateOnDecreaseSchema = closedEnumSchema9;
var billingUpdateProrationSchema = import_v414.z.object({
  onIncrease: billingUpdateOnIncreaseSchema,
  onDecrease: billingUpdateOnDecreaseSchema
});
var billingUpdateExpiryDurationTypeSchema = closedEnumSchema9;
var billingUpdateRolloverSchema = import_v414.z.object({
  max: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  expiryDurationType: billingUpdateExpiryDurationTypeSchema,
  expiryDurationLength: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional()
});
var billingUpdatePlanItemSchema = import_v414.z.object({
  featureId: import_v414.z.string(),
  included: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  unlimited: import_v414.z.union([import_v414.z.boolean(), import_v414.z.undefined()]).optional(),
  reset: import_v414.z.union([billingUpdateResetSchema, import_v414.z.undefined()]).optional(),
  price: import_v414.z.union([billingUpdatePriceSchema, import_v414.z.undefined()]).optional(),
  proration: import_v414.z.union([billingUpdateProrationSchema, import_v414.z.undefined()]).optional(),
  rollover: import_v414.z.union([billingUpdateRolloverSchema, import_v414.z.undefined()]).optional()
});
var billingUpdateDurationTypeSchema = closedEnumSchema9;
var billingUpdateFreeTrialParamsSchema = import_v414.z.object({
  durationLength: import_v414.z.number(),
  durationType: import_v414.z.union([billingUpdateDurationTypeSchema, import_v414.z.undefined()]).optional(),
  cardRequired: import_v414.z.union([import_v414.z.boolean(), import_v414.z.undefined()]).optional()
});
var billingUpdateCustomizeSchema = import_v414.z.object({
  price: import_v414.z.union([billingUpdateBasePriceSchema, import_v414.z.undefined()]).optional().nullable(),
  items: import_v414.z.union([import_v414.z.array(billingUpdatePlanItemSchema), import_v414.z.undefined()]).optional(),
  freeTrial: import_v414.z.union([billingUpdateFreeTrialParamsSchema, import_v414.z.undefined()]).optional().nullable()
});
var billingUpdateProrationBehaviorSchema = closedEnumSchema9;
var billingUpdateRedirectModeSchema = closedEnumSchema9;
var billingUpdateCancelActionSchema = closedEnumSchema9;
var updateSubscriptionParamsSchema = import_v414.z.object({
  customerId: import_v414.z.string(),
  entityId: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  planId: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  featureQuantities: import_v414.z.union([import_v414.z.array(billingUpdateFeatureQuantitySchema), import_v414.z.undefined()]).optional(),
  version: import_v414.z.union([import_v414.z.number(), import_v414.z.undefined()]).optional(),
  customize: import_v414.z.union([billingUpdateCustomizeSchema, import_v414.z.undefined()]).optional(),
  invoiceMode: import_v414.z.union([billingUpdateInvoiceModeSchema, import_v414.z.undefined()]).optional(),
  prorationBehavior: import_v414.z.union([billingUpdateProrationBehaviorSchema, import_v414.z.undefined()]).optional(),
  redirectMode: import_v414.z.union([billingUpdateRedirectModeSchema, import_v414.z.undefined()]).optional(),
  subscriptionId: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  cancelAction: import_v414.z.union([billingUpdateCancelActionSchema, import_v414.z.undefined()]).optional(),
  noBillingChanges: import_v414.z.union([import_v414.z.boolean(), import_v414.z.undefined()]).optional(),
  recalculateBalances: import_v414.z.union([billingUpdateRecalculateBalancesSchema, import_v414.z.undefined()]).optional()
});
var billingUpdateCodeSchema = openEnumSchema7;
var billingUpdateRequiredActionSchema = import_v414.z.object({
  code: billingUpdateCodeSchema,
  reason: import_v414.z.string()
});
var billingUpdateResponseSchema = import_v414.z.object({
  customerId: import_v414.z.string(),
  entityId: import_v414.z.union([import_v414.z.string(), import_v414.z.undefined()]).optional(),
  invoice: import_v414.z.union([billingUpdateInvoiceSchema, import_v414.z.undefined()]).optional(),
  paymentUrl: import_v414.z.string().nullable(),
  requiredAction: import_v414.z.union([billingUpdateRequiredActionSchema, import_v414.z.undefined()]).optional()
});

// src/backend/core/utils/backendRes.ts
var backendSuccess = ({
  body,
  statusCode = 200
}) => {
  return { statusCode, body };
};
var backendError = ({
  message,
  code = "internal_server_error",
  statusCode = 500,
  details
}) => {
  return {
    statusCode,
    body: {
      message: message || "Internal server error",
      code,
      statusCode,
      ...details !== void 0 ? { details } : {}
    }
  };
};

// src/backend/core/utils/sanitizeBody.ts
var PROTECTED_FIELDS = [
  "customerId",
  "name",
  "email",
  "metadata",
  "stripeId"
];
var sanitizeBody = (body) => {
  const rawBody = body || {};
  const sanitized = {};
  for (const [key, value] of Object.entries(rawBody)) {
    if (!PROTECTED_FIELDS.includes(key)) {
      sanitized[key] = value;
    }
  }
  return sanitized;
};

// src/backend/core/routes/routeConfigs.ts
var getEntityBodySchema = import_v415.z.object({
  entityId: import_v415.z.string()
});
var routeConfigs = [
  {
    route: "getOrCreateCustomer",
    sdkMethod: (autumn, args) => autumn.customers.getOrCreate(args),
    requireCustomer: false,
    // customHandler handles auth logic for errorOnNotFound
    bodySchema: import_v415.z.object({
      errorOnNotFound: import_v415.z.boolean().optional().default(true),
      // expand: z.array(z.enum(CustomerExpand)).optional(),
      expand: import_v415.z.array(import_v415.z.string()).optional()
    }),
    customHandler: async ({ autumn, identity, body }) => {
      const sanitizedBody = sanitizeBody(body);
      if (!identity?.customerId && sanitizedBody.errorOnNotFound === false) {
        return backendSuccess({ statusCode: 204, body: null });
      }
      if (!identity?.customerId) {
        return backendError({
          message: "customerId not found",
          code: "no_customer_id",
          statusCode: 401
        });
      }
      const existingExpand = Array.isArray(sanitizedBody.expand) ? sanitizedBody.expand : [];
      const args = {
        customerId: identity.customerId,
        ...identity.customerData,
        ...sanitizedBody,
        expand: [...existingExpand, "balances.feature"]
      };
      return autumn.customers.getOrCreate(args);
    }
  },
  {
    route: "getEntity",
    sdkMethod: (autumn, args) => autumn.entities.get(args),
    bodySchema: getEntityBodySchema
  },
  {
    route: "attach",
    sdkMethod: (autumn, args) => autumn.billing.attach(args),
    bodySchema: attachParamsSchema
  },
  {
    route: "previewAttach",
    sdkMethod: (autumn, args) => autumn.billing.previewAttach(args),
    bodySchema: previewAttachParamsSchema
  },
  {
    route: "updateSubscription",
    sdkMethod: (autumn, args) => autumn.billing.update(args),
    bodySchema: updateSubscriptionParamsSchema
  },
  {
    route: "previewUpdateSubscription",
    sdkMethod: (autumn, args) => autumn.billing.previewUpdate(args),
    bodySchema: previewUpdateParamsSchema
  },
  {
    route: "openCustomerPortal",
    sdkMethod: (autumn, args) => autumn.billing.openCustomerPortal(args),
    bodySchema: openCustomerPortalParamsSchema
  },
  {
    route: "createReferralCode",
    sdkMethod: (autumn, args) => autumn.referrals.createCode(args),
    bodySchema: createReferralCodeParamsSchema
  },
  {
    route: "redeemReferralCode",
    sdkMethod: (autumn, args) => autumn.referrals.redeemCode(args),
    bodySchema: redeemReferralCodeParamsSchema
  },
  {
    route: "multiAttach",
    sdkMethod: (autumn, args) => autumn.billing.multiAttach(args),
    bodySchema: multiAttachParamsSchema
  },
  {
    route: "previewMultiAttach",
    sdkMethod: (autumn, args) => autumn.billing.previewMultiAttach(args),
    bodySchema: previewMultiAttachParamsSchema
  },
  {
    route: "setupPayment",
    sdkMethod: (autumn, args) => autumn.billing.setupPayment(args),
    bodySchema: setupPaymentParamsSchema
  },
  {
    route: "listPlans",
    sdkMethod: (autumn, args) => autumn.plans.list(args),
    requireCustomer: false,
    bodySchema: listPlansParamsSchema.optional()
  },
  {
    route: "listEvents",
    sdkMethod: (autumn, args) => autumn.events.list(args),
    bodySchema: eventsListParamsSchema.optional()
  },
  {
    route: "aggregateEvents",
    sdkMethod: (autumn, args) => autumn.events.aggregate(args),
    bodySchema: eventsAggregateParamsSchema
  }
];
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  routeConfigs
});
