// autumn-js/dist/backend/chunk-BSY23UM3.mjs
import {
  sanitizeBody
} from "./chunk-Z6QP3YEA.mjs";
import {
  backendError,
  backendSuccess
} from "./chunk-N7NFYY54.mjs";

// src/backend/core/routes/routeConfigs.ts
import { z as z15 } from "zod/v4";

// src/generated/aggregateEventsSchemas.ts
import { z } from "zod/v4";
var aggregateEventsGlobalsSchema = z.object({
  xApiVersion: z.union([z.string(), z.undefined()]).optional()
});
var aggregateEventsFeatureIdSchema = z.union([
  z.string(),
  z.array(z.string())
]);
var aggregateEventsCustomRangeSchema = z.object({
  start: z.number(),
  end: z.number()
});
var aggregateEventsListSchema = z.object({
  period: z.number(),
  values: z.record(z.string(), z.number()),
  groupedValues: z.union([
    z.record(z.string(), z.record(z.string(), z.number())),
    z.undefined()
  ]).optional()
});
var totalSchema = z.object({
  count: z.number(),
  sum: z.number()
});
var aggregateEventsResponseSchema = z.object({
  list: z.array(aggregateEventsListSchema),
  total: z.record(z.string(), totalSchema)
});
var aggregateEventsFeatureIdOutboundSchema = z.union([
  z.string(),
  z.array(z.string())
]);
var aggregateEventsCustomRangeOutboundSchema = z.object({
  start: z.number(),
  end: z.number()
});
var eventsAggregateParamsOutboundSchema = z.object({
  customer_id: z.union([z.string(), z.undefined()]).optional(),
  entity_id: z.union([z.string(), z.undefined()]).optional(),
  feature_id: z.union([z.string(), z.array(z.string())]),
  group_by: z.union([z.string(), z.undefined()]).optional(),
  range: z.union([z.string(), z.undefined()]).optional(),
  bin_size: z.string(),
  custom_range: z.union([aggregateEventsCustomRangeOutboundSchema, z.undefined()]).optional(),
  max_groups: z.union([z.number(), z.undefined()]).optional()
});
var closedEnumSchema = z.any();
var rangeSchema = closedEnumSchema;
var binSizeSchema = closedEnumSchema;
var eventsAggregateParamsSchema = z.object({
  customerId: z.union([z.string(), z.undefined()]).optional(),
  entityId: z.union([z.string(), z.undefined()]).optional(),
  featureId: z.union([z.string(), z.array(z.string())]),
  groupBy: z.union([z.string(), z.undefined()]).optional(),
  range: z.union([rangeSchema, z.undefined()]).optional(),
  binSize: z.union([binSizeSchema, z.undefined()]).optional(),
  customRange: z.union([aggregateEventsCustomRangeSchema, z.undefined()]).optional(),
  maxGroups: z.union([z.number(), z.undefined()]).optional()
});

// src/generated/attachSchemas.ts
import { z as z2 } from "zod/v4";
var attachGlobalsSchema = z2.object({
  xApiVersion: z2.union([z2.string(), z2.undefined()]).optional()
});
var attachFeatureQuantitySchema = z2.object({
  featureId: z2.string(),
  quantity: z2.union([z2.number(), z2.undefined()]).optional(),
  adjustable: z2.union([z2.boolean(), z2.undefined()]).optional()
});
var attachToSchema = z2.union([z2.number(), z2.string()]);
var attachTierSchema = z2.object({
  to: z2.union([z2.number(), z2.string()]),
  amount: z2.union([z2.number(), z2.undefined()]).optional(),
  flatAmount: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachInvoiceModeSchema = z2.object({
  enabled: z2.boolean(),
  enablePlanImmediately: z2.union([z2.boolean(), z2.undefined()]).optional(),
  finalize: z2.union([z2.boolean(), z2.undefined()]).optional()
});
var attachAttachDiscountSchema = z2.object({
  rewardId: z2.union([z2.string(), z2.undefined()]).optional(),
  promotionCode: z2.union([z2.string(), z2.undefined()]).optional()
});
var attachCustomLineItemSchema = z2.object({
  amount: z2.number(),
  description: z2.string()
});
var attachCarryOverBalancesSchema = z2.object({
  enabled: z2.boolean(),
  featureIds: z2.union([z2.array(z2.string()), z2.undefined()]).optional()
});
var attachCarryOverUsagesSchema = z2.object({
  enabled: z2.boolean(),
  featureIds: z2.union([z2.array(z2.string()), z2.undefined()]).optional()
});
var attachInvoiceSchema = z2.object({
  status: z2.string().nullable(),
  stripeId: z2.string(),
  total: z2.number(),
  currency: z2.string(),
  hostedInvoiceUrl: z2.string().nullable()
});
var attachFeatureQuantityOutboundSchema = z2.object({
  feature_id: z2.string(),
  quantity: z2.union([z2.number(), z2.undefined()]).optional(),
  adjustable: z2.union([z2.boolean(), z2.undefined()]).optional()
});
var attachBasePriceOutboundSchema = z2.object({
  amount: z2.number(),
  interval: z2.string(),
  interval_count: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachResetOutboundSchema = z2.object({
  interval: z2.string(),
  interval_count: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachToOutboundSchema = z2.union([z2.number(), z2.string()]);
var attachTierOutboundSchema = z2.object({
  to: z2.union([z2.number(), z2.string()]),
  amount: z2.union([z2.number(), z2.undefined()]).optional(),
  flat_amount: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachPriceOutboundSchema = z2.object({
  amount: z2.union([z2.number(), z2.undefined()]).optional(),
  tiers: z2.union([z2.array(attachTierOutboundSchema), z2.undefined()]).optional(),
  tier_behavior: z2.union([z2.string(), z2.undefined()]).optional(),
  interval: z2.string(),
  interval_count: z2.number(),
  billing_units: z2.number(),
  billing_method: z2.string(),
  max_purchase: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachProrationOutboundSchema = z2.object({
  on_increase: z2.string(),
  on_decrease: z2.string()
});
var attachRolloverOutboundSchema = z2.object({
  max: z2.union([z2.number(), z2.undefined()]).optional(),
  expiry_duration_type: z2.string(),
  expiry_duration_length: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachPlanItemOutboundSchema = z2.object({
  feature_id: z2.string(),
  included: z2.union([z2.number(), z2.undefined()]).optional(),
  unlimited: z2.union([z2.boolean(), z2.undefined()]).optional(),
  reset: z2.union([attachResetOutboundSchema, z2.undefined()]).optional(),
  price: z2.union([attachPriceOutboundSchema, z2.undefined()]).optional(),
  proration: z2.union([attachProrationOutboundSchema, z2.undefined()]).optional(),
  rollover: z2.union([attachRolloverOutboundSchema, z2.undefined()]).optional()
});
var attachFreeTrialParamsOutboundSchema = z2.object({
  duration_length: z2.number(),
  duration_type: z2.string(),
  card_required: z2.boolean()
});
var attachCustomizeOutboundSchema = z2.object({
  price: z2.union([attachBasePriceOutboundSchema, z2.undefined()]).optional().nullable(),
  items: z2.union([z2.array(attachPlanItemOutboundSchema), z2.undefined()]).optional(),
  free_trial: z2.union([attachFreeTrialParamsOutboundSchema, z2.undefined()]).optional().nullable()
});
var attachInvoiceModeOutboundSchema = z2.object({
  enabled: z2.boolean(),
  enable_plan_immediately: z2.boolean(),
  finalize: z2.boolean()
});
var attachAttachDiscountOutboundSchema = z2.object({
  reward_id: z2.union([z2.string(), z2.undefined()]).optional(),
  promotion_code: z2.union([z2.string(), z2.undefined()]).optional()
});
var attachCustomLineItemOutboundSchema = z2.object({
  amount: z2.number(),
  description: z2.string()
});
var attachCarryOverBalancesOutboundSchema = z2.object({
  enabled: z2.boolean(),
  feature_ids: z2.union([z2.array(z2.string()), z2.undefined()]).optional()
});
var attachCarryOverUsagesOutboundSchema = z2.object({
  enabled: z2.boolean(),
  feature_ids: z2.union([z2.array(z2.string()), z2.undefined()]).optional()
});
var attachParamsOutboundSchema = z2.object({
  customer_id: z2.string(),
  entity_id: z2.union([z2.string(), z2.undefined()]).optional(),
  plan_id: z2.string(),
  feature_quantities: z2.union([z2.array(attachFeatureQuantityOutboundSchema), z2.undefined()]).optional(),
  version: z2.union([z2.number(), z2.undefined()]).optional(),
  customize: z2.union([attachCustomizeOutboundSchema, z2.undefined()]).optional(),
  invoice_mode: z2.union([attachInvoiceModeOutboundSchema, z2.undefined()]).optional(),
  proration_behavior: z2.union([z2.string(), z2.undefined()]).optional(),
  redirect_mode: z2.string(),
  subscription_id: z2.union([z2.string(), z2.undefined()]).optional(),
  discounts: z2.union([z2.array(attachAttachDiscountOutboundSchema), z2.undefined()]).optional(),
  success_url: z2.union([z2.string(), z2.undefined()]).optional(),
  new_billing_subscription: z2.union([z2.boolean(), z2.undefined()]).optional(),
  plan_schedule: z2.union([z2.string(), z2.undefined()]).optional(),
  checkout_session_params: z2.union([z2.record(z2.string(), z2.any()), z2.undefined()]).optional(),
  custom_line_items: z2.union([z2.array(attachCustomLineItemOutboundSchema), z2.undefined()]).optional(),
  processor_subscription_id: z2.union([z2.string(), z2.undefined()]).optional(),
  carry_over_balances: z2.union([attachCarryOverBalancesOutboundSchema, z2.undefined()]).optional(),
  carry_over_usages: z2.union([attachCarryOverUsagesOutboundSchema, z2.undefined()]).optional(),
  metadata: z2.union([z2.record(z2.string(), z2.string()), z2.undefined()]).optional()
});
var closedEnumSchema2 = z2.any();
var openEnumSchema = z2.any();
var attachPriceIntervalSchema = closedEnumSchema2;
var attachBasePriceSchema = z2.object({
  amount: z2.number(),
  interval: attachPriceIntervalSchema,
  intervalCount: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachResetIntervalSchema = closedEnumSchema2;
var attachResetSchema = z2.object({
  interval: attachResetIntervalSchema,
  intervalCount: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachTierBehaviorSchema = closedEnumSchema2;
var attachItemPriceIntervalSchema = closedEnumSchema2;
var attachBillingMethodSchema = closedEnumSchema2;
var attachPriceSchema = z2.object({
  amount: z2.union([z2.number(), z2.undefined()]).optional(),
  tiers: z2.union([z2.array(attachTierSchema), z2.undefined()]).optional(),
  tierBehavior: z2.union([attachTierBehaviorSchema, z2.undefined()]).optional(),
  interval: attachItemPriceIntervalSchema,
  intervalCount: z2.union([z2.number(), z2.undefined()]).optional(),
  billingUnits: z2.union([z2.number(), z2.undefined()]).optional(),
  billingMethod: attachBillingMethodSchema,
  maxPurchase: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachOnIncreaseSchema = closedEnumSchema2;
var attachOnDecreaseSchema = closedEnumSchema2;
var attachProrationSchema = z2.object({
  onIncrease: attachOnIncreaseSchema,
  onDecrease: attachOnDecreaseSchema
});
var attachExpiryDurationTypeSchema = closedEnumSchema2;
var attachRolloverSchema = z2.object({
  max: z2.union([z2.number(), z2.undefined()]).optional(),
  expiryDurationType: attachExpiryDurationTypeSchema,
  expiryDurationLength: z2.union([z2.number(), z2.undefined()]).optional()
});
var attachPlanItemSchema = z2.object({
  featureId: z2.string(),
  included: z2.union([z2.number(), z2.undefined()]).optional(),
  unlimited: z2.union([z2.boolean(), z2.undefined()]).optional(),
  reset: z2.union([attachResetSchema, z2.undefined()]).optional(),
  price: z2.union([attachPriceSchema, z2.undefined()]).optional(),
  proration: z2.union([attachProrationSchema, z2.undefined()]).optional(),
  rollover: z2.union([attachRolloverSchema, z2.undefined()]).optional()
});
var attachDurationTypeSchema = closedEnumSchema2;
var attachFreeTrialParamsSchema = z2.object({
  durationLength: z2.number(),
  durationType: z2.union([attachDurationTypeSchema, z2.undefined()]).optional(),
  cardRequired: z2.union([z2.boolean(), z2.undefined()]).optional()
});
var attachCustomizeSchema = z2.object({
  price: z2.union([attachBasePriceSchema, z2.undefined()]).optional().nullable(),
  items: z2.union([z2.array(attachPlanItemSchema), z2.undefined()]).optional(),
  freeTrial: z2.union([attachFreeTrialParamsSchema, z2.undefined()]).optional().nullable()
});
var attachProrationBehaviorSchema = closedEnumSchema2;
var attachRedirectModeSchema = closedEnumSchema2;
var attachPlanScheduleSchema = closedEnumSchema2;
var attachParamsSchema = z2.object({
  customerId: z2.string(),
  entityId: z2.union([z2.string(), z2.undefined()]).optional(),
  planId: z2.string(),
  featureQuantities: z2.union([z2.array(attachFeatureQuantitySchema), z2.undefined()]).optional(),
  version: z2.union([z2.number(), z2.undefined()]).optional(),
  customize: z2.union([attachCustomizeSchema, z2.undefined()]).optional(),
  invoiceMode: z2.union([attachInvoiceModeSchema, z2.undefined()]).optional(),
  prorationBehavior: z2.union([attachProrationBehaviorSchema, z2.undefined()]).optional(),
  redirectMode: z2.union([attachRedirectModeSchema, z2.undefined()]).optional(),
  subscriptionId: z2.union([z2.string(), z2.undefined()]).optional(),
  discounts: z2.union([z2.array(attachAttachDiscountSchema), z2.undefined()]).optional(),
  successUrl: z2.union([z2.string(), z2.undefined()]).optional(),
  newBillingSubscription: z2.union([z2.boolean(), z2.undefined()]).optional(),
  planSchedule: z2.union([attachPlanScheduleSchema, z2.undefined()]).optional(),
  checkoutSessionParams: z2.union([z2.record(z2.string(), z2.any()), z2.undefined()]).optional(),
  customLineItems: z2.union([z2.array(attachCustomLineItemSchema), z2.undefined()]).optional(),
  processorSubscriptionId: z2.union([z2.string(), z2.undefined()]).optional(),
  carryOverBalances: z2.union([attachCarryOverBalancesSchema, z2.undefined()]).optional(),
  carryOverUsages: z2.union([attachCarryOverUsagesSchema, z2.undefined()]).optional(),
  metadata: z2.union([z2.record(z2.string(), z2.string()), z2.undefined()]).optional()
});
var attachCodeSchema = openEnumSchema;
var attachRequiredActionSchema = z2.object({
  code: attachCodeSchema,
  reason: z2.string()
});
var attachResponseSchema = z2.object({
  customerId: z2.string(),
  entityId: z2.union([z2.string(), z2.undefined()]).optional(),
  invoice: z2.union([attachInvoiceSchema, z2.undefined()]).optional(),
  paymentUrl: z2.string().nullable(),
  requiredAction: z2.union([attachRequiredActionSchema, z2.undefined()]).optional()
});

// src/generated/createReferralCodeSchemas.ts
import { z as z3 } from "zod/v4";
var createReferralCodeGlobalsSchema = z3.object({
  xApiVersion: z3.union([z3.string(), z3.undefined()]).optional()
});
var createReferralCodeParamsSchema = z3.object({
  customerId: z3.string(),
  programId: z3.string()
});
var createReferralCodeResponseSchema = z3.object({
  code: z3.string(),
  customerId: z3.string(),
  createdAt: z3.number()
});
var createReferralCodeParamsOutboundSchema = z3.object({
  customer_id: z3.string(),
  program_id: z3.string()
});

// src/generated/getOrCreateCustomerSchemas.ts
import { z as z4 } from "zod/v4";
var getOrCreateCustomerGlobalsSchema = z4.object({
  xApiVersion: z4.union([z4.string(), z4.undefined()]).optional()
});
var getOrCreateCustomerSpendLimitSchema = z4.object({
  featureId: z4.union([z4.string(), z4.undefined()]).optional(),
  enabled: z4.union([z4.boolean(), z4.undefined()]).optional(),
  overageLimit: z4.union([z4.number(), z4.undefined()]).optional()
});
var getOrCreateCustomerPurchaseLimitOutboundSchema = z4.object({
  interval: z4.string(),
  interval_count: z4.number(),
  limit: z4.number()
});
var getOrCreateCustomerAutoTopupOutboundSchema = z4.object({
  feature_id: z4.string(),
  enabled: z4.boolean(),
  threshold: z4.number(),
  quantity: z4.number(),
  purchase_limit: z4.union([getOrCreateCustomerPurchaseLimitOutboundSchema, z4.undefined()]).optional()
});
var getOrCreateCustomerSpendLimitOutboundSchema = z4.object({
  feature_id: z4.union([z4.string(), z4.undefined()]).optional(),
  enabled: z4.boolean(),
  overage_limit: z4.union([z4.number(), z4.undefined()]).optional()
});
var getOrCreateCustomerUsageAlertOutboundSchema = z4.object({
  feature_id: z4.union([z4.string(), z4.undefined()]).optional(),
  enabled: z4.boolean(),
  threshold: z4.number(),
  threshold_type: z4.string(),
  name: z4.union([z4.string(), z4.undefined()]).optional()
});
var getOrCreateCustomerBillingControlsOutboundSchema = z4.object({
  auto_topups: z4.union([z4.array(getOrCreateCustomerAutoTopupOutboundSchema), z4.undefined()]).optional(),
  spend_limits: z4.union([
    z4.array(getOrCreateCustomerSpendLimitOutboundSchema),
    z4.undefined()
  ]).optional(),
  usage_alerts: z4.union([
    z4.array(getOrCreateCustomerUsageAlertOutboundSchema),
    z4.undefined()
  ]).optional()
});
var getOrCreateCustomerParamsOutboundSchema = z4.object({
  customer_id: z4.string().nullable(),
  name: z4.union([z4.string(), z4.undefined()]).optional().nullable(),
  email: z4.union([z4.string(), z4.undefined()]).optional().nullable(),
  fingerprint: z4.union([z4.string(), z4.undefined()]).optional().nullable(),
  metadata: z4.union([z4.record(z4.string(), z4.any()), z4.undefined()]).optional().nullable(),
  stripe_id: z4.union([z4.string(), z4.undefined()]).optional().nullable(),
  create_in_stripe: z4.union([z4.boolean(), z4.undefined()]).optional(),
  auto_enable_plan_id: z4.union([z4.string(), z4.undefined()]).optional(),
  send_email_receipts: z4.union([z4.boolean(), z4.undefined()]).optional(),
  billing_controls: z4.union([getOrCreateCustomerBillingControlsOutboundSchema, z4.undefined()]).optional(),
  expand: z4.union([z4.array(z4.string()), z4.undefined()]).optional()
});
var closedEnumSchema3 = z4.any();
var getOrCreateCustomerIntervalSchema = closedEnumSchema3;
var getOrCreateCustomerPurchaseLimitSchema = z4.object({
  interval: getOrCreateCustomerIntervalSchema,
  intervalCount: z4.union([z4.number(), z4.undefined()]).optional(),
  limit: z4.number()
});
var getOrCreateCustomerAutoTopupSchema = z4.object({
  featureId: z4.string(),
  enabled: z4.union([z4.boolean(), z4.undefined()]).optional(),
  threshold: z4.number(),
  quantity: z4.number(),
  purchaseLimit: z4.union([getOrCreateCustomerPurchaseLimitSchema, z4.undefined()]).optional()
});
var getOrCreateCustomerThresholdTypeSchema = closedEnumSchema3;
var getOrCreateCustomerUsageAlertSchema = z4.object({
  featureId: z4.union([z4.string(), z4.undefined()]).optional(),
  enabled: z4.union([z4.boolean(), z4.undefined()]).optional(),
  threshold: z4.number(),
  thresholdType: getOrCreateCustomerThresholdTypeSchema,
  name: z4.union([z4.string(), z4.undefined()]).optional()
});
var getOrCreateCustomerBillingControlsSchema = z4.object({
  autoTopups: z4.union([z4.array(getOrCreateCustomerAutoTopupSchema), z4.undefined()]).optional(),
  spendLimits: z4.union([z4.array(getOrCreateCustomerSpendLimitSchema), z4.undefined()]).optional(),
  usageAlerts: z4.union([z4.array(getOrCreateCustomerUsageAlertSchema), z4.undefined()]).optional()
});
var getOrCreateCustomerParamsSchema = z4.object({
  customerId: z4.string().nullable(),
  name: z4.union([z4.string(), z4.undefined()]).optional().nullable(),
  email: z4.union([z4.string(), z4.undefined()]).optional().nullable(),
  fingerprint: z4.union([z4.string(), z4.undefined()]).optional().nullable(),
  metadata: z4.union([z4.record(z4.string(), z4.any()), z4.undefined()]).optional().nullable(),
  stripeId: z4.union([z4.string(), z4.undefined()]).optional().nullable(),
  createInStripe: z4.union([z4.boolean(), z4.undefined()]).optional(),
  autoEnablePlanId: z4.union([z4.string(), z4.undefined()]).optional(),
  sendEmailReceipts: z4.union([z4.boolean(), z4.undefined()]).optional(),
  billingControls: z4.union([getOrCreateCustomerBillingControlsSchema, z4.undefined()]).optional(),
  expand: z4.union([z4.array(z4.string()), z4.undefined()]).optional()
});

// src/generated/listEventsSchemas.ts
import { z as z5 } from "zod/v4";
var listEventsGlobalsSchema = z5.object({
  xApiVersion: z5.union([z5.string(), z5.undefined()]).optional()
});
var listEventsFeatureIdSchema = z5.union([
  z5.string(),
  z5.array(z5.string())
]);
var listEventsCustomRangeSchema = z5.object({
  start: z5.union([z5.number(), z5.undefined()]).optional(),
  end: z5.union([z5.number(), z5.undefined()]).optional()
});
var eventsListParamsSchema = z5.object({
  offset: z5.union([z5.number(), z5.undefined()]).optional(),
  limit: z5.union([z5.number(), z5.undefined()]).optional(),
  customerId: z5.union([z5.string(), z5.undefined()]).optional(),
  entityId: z5.union([z5.string(), z5.undefined()]).optional(),
  featureId: z5.union([z5.string(), z5.array(z5.string()), z5.undefined()]).optional(),
  customRange: z5.union([listEventsCustomRangeSchema, z5.undefined()]).optional()
});
var listEventsPropertiesSchema = z5.object({});
var listEventsListSchema = z5.object({
  id: z5.string(),
  timestamp: z5.number(),
  featureId: z5.string(),
  customerId: z5.string(),
  value: z5.number(),
  properties: listEventsPropertiesSchema
});
var listEventsResponseSchema = z5.object({
  list: z5.array(listEventsListSchema),
  hasMore: z5.boolean(),
  offset: z5.number(),
  limit: z5.number(),
  total: z5.number()
});
var listEventsFeatureIdOutboundSchema = z5.union([
  z5.string(),
  z5.array(z5.string())
]);
var listEventsCustomRangeOutboundSchema = z5.object({
  start: z5.union([z5.number(), z5.undefined()]).optional(),
  end: z5.union([z5.number(), z5.undefined()]).optional()
});
var eventsListParamsOutboundSchema = z5.object({
  offset: z5.number(),
  limit: z5.number(),
  customer_id: z5.union([z5.string(), z5.undefined()]).optional(),
  entity_id: z5.union([z5.string(), z5.undefined()]).optional(),
  feature_id: z5.union([z5.string(), z5.array(z5.string()), z5.undefined()]).optional(),
  custom_range: z5.union([listEventsCustomRangeOutboundSchema, z5.undefined()]).optional()
});

// src/generated/listPlansSchemas.ts
import { z as z6 } from "zod/v4";
var listPlansGlobalsSchema = z6.object({
  xApiVersion: z6.union([z6.string(), z6.undefined()]).optional()
});
var listPlansParamsSchema = z6.object({
  customerId: z6.union([z6.string(), z6.undefined()]).optional(),
  entityId: z6.union([z6.string(), z6.undefined()]).optional(),
  includeArchived: z6.union([z6.boolean(), z6.undefined()]).optional()
});
var listPlansPriceDisplaySchema = z6.object({
  primaryText: z6.string(),
  secondaryText: z6.union([z6.string(), z6.undefined()]).optional()
});
var listPlansFeatureDisplaySchema = z6.object({
  singular: z6.string(),
  plural: z6.string()
});
var listPlansCreditSchemaSchema = z6.object({
  meteredFeatureId: z6.string(),
  creditCost: z6.number()
});
var listPlansItemDisplaySchema = z6.object({
  primaryText: z6.string(),
  secondaryText: z6.union([z6.string(), z6.undefined()]).optional()
});
var listPlansParamsOutboundSchema = z6.object({
  customer_id: z6.union([z6.string(), z6.undefined()]).optional(),
  entity_id: z6.union([z6.string(), z6.undefined()]).optional(),
  include_archived: z6.union([z6.boolean(), z6.undefined()]).optional()
});
var openEnumSchema2 = z6.any();
var listPlansPriceIntervalSchema = openEnumSchema2;
var listPlansPriceSchema = z6.object({
  amount: z6.number(),
  interval: listPlansPriceIntervalSchema,
  intervalCount: z6.union([z6.number(), z6.undefined()]).optional(),
  display: z6.union([listPlansPriceDisplaySchema, z6.undefined()]).optional()
});
var listPlansTypeSchema = openEnumSchema2;
var listPlansFeatureSchema = z6.object({
  id: z6.string(),
  name: z6.union([z6.string(), z6.undefined()]).optional().nullable(),
  type: listPlansTypeSchema,
  display: z6.union([listPlansFeatureDisplaySchema, z6.undefined()]).optional().nullable(),
  creditSchema: z6.union([z6.array(listPlansCreditSchemaSchema), z6.undefined()]).optional().nullable(),
  archived: z6.union([z6.boolean(), z6.undefined()]).optional().nullable()
});
var listPlansResetIntervalSchema = openEnumSchema2;
var listPlansResetSchema = z6.object({
  interval: listPlansResetIntervalSchema,
  intervalCount: z6.union([z6.number(), z6.undefined()]).optional()
});
var listPlansTierBehaviorSchema = openEnumSchema2;
var listPlansPriceItemIntervalSchema = openEnumSchema2;
var listPlansBillingMethodSchema = openEnumSchema2;
var listPlansItemPriceSchema = z6.object({
  amount: z6.union([z6.number(), z6.undefined()]).optional(),
  tiers: z6.union([z6.array(z6.any().nullable()), z6.undefined()]).optional(),
  tierBehavior: z6.union([listPlansTierBehaviorSchema, z6.undefined()]).optional(),
  interval: listPlansPriceItemIntervalSchema,
  intervalCount: z6.union([z6.number(), z6.undefined()]).optional(),
  billingUnits: z6.number(),
  billingMethod: listPlansBillingMethodSchema,
  maxPurchase: z6.number().nullable()
});
var listPlansExpiryDurationTypeSchema = openEnumSchema2;
var listPlansRolloverSchema = z6.object({
  max: z6.number().nullable(),
  expiryDurationType: listPlansExpiryDurationTypeSchema,
  expiryDurationLength: z6.union([z6.number(), z6.undefined()]).optional()
});
var listPlansItemSchema = z6.object({
  featureId: z6.string(),
  feature: z6.union([listPlansFeatureSchema, z6.undefined()]).optional(),
  included: z6.number(),
  unlimited: z6.boolean(),
  reset: listPlansResetSchema.nullable(),
  price: listPlansItemPriceSchema.nullable(),
  display: z6.union([listPlansItemDisplaySchema, z6.undefined()]).optional(),
  rollover: z6.union([listPlansRolloverSchema, z6.undefined()]).optional()
});
var listPlansDurationTypeSchema = openEnumSchema2;
var listPlansFreeTrialSchema = z6.object({
  durationLength: z6.number(),
  durationType: listPlansDurationTypeSchema,
  cardRequired: z6.boolean()
});
var listPlansEnvSchema = openEnumSchema2;
var listPlansStatusSchema = openEnumSchema2;
var listPlansAttachActionSchema = openEnumSchema2;
var listPlansCustomerEligibilitySchema = z6.object({
  trialAvailable: z6.union([z6.boolean(), z6.undefined()]).optional(),
  status: z6.union([listPlansStatusSchema, z6.undefined()]).optional(),
  canceling: z6.union([z6.boolean(), z6.undefined()]).optional(),
  trialing: z6.union([z6.boolean(), z6.undefined()]).optional(),
  attachAction: listPlansAttachActionSchema
});
var listPlansListSchema = z6.object({
  id: z6.string(),
  name: z6.string(),
  description: z6.string().nullable(),
  group: z6.string().nullable(),
  version: z6.number(),
  addOn: z6.boolean(),
  autoEnable: z6.boolean(),
  price: listPlansPriceSchema.nullable(),
  items: z6.array(listPlansItemSchema),
  freeTrial: z6.union([listPlansFreeTrialSchema, z6.undefined()]).optional(),
  createdAt: z6.number(),
  env: listPlansEnvSchema,
  archived: z6.boolean(),
  baseVariantId: z6.string().nullable(),
  customerEligibility: z6.union([listPlansCustomerEligibilitySchema, z6.undefined()]).optional()
});
var listPlansResponseSchema = z6.object({
  list: z6.array(listPlansListSchema)
});

// src/generated/multiAttachSchemas.ts
import { z as z7 } from "zod/v4";
var multiAttachGlobalsSchema = z7.object({
  xApiVersion: z7.union([z7.string(), z7.undefined()]).optional()
});
var multiAttachToSchema = z7.union([z7.number(), z7.string()]);
var multiAttachTierSchema = z7.object({
  to: z7.union([z7.number(), z7.string()]),
  amount: z7.union([z7.number(), z7.undefined()]).optional(),
  flatAmount: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachFeatureQuantitySchema = z7.object({
  featureId: z7.string(),
  quantity: z7.union([z7.number(), z7.undefined()]).optional(),
  adjustable: z7.union([z7.boolean(), z7.undefined()]).optional()
});
var multiAttachInvoiceModeSchema = z7.object({
  enabled: z7.boolean(),
  enablePlanImmediately: z7.union([z7.boolean(), z7.undefined()]).optional(),
  finalize: z7.union([z7.boolean(), z7.undefined()]).optional()
});
var multiAttachAttachDiscountSchema = z7.object({
  rewardId: z7.union([z7.string(), z7.undefined()]).optional(),
  promotionCode: z7.union([z7.string(), z7.undefined()]).optional()
});
var multiAttachSpendLimitSchema = z7.object({
  featureId: z7.union([z7.string(), z7.undefined()]).optional(),
  enabled: z7.union([z7.boolean(), z7.undefined()]).optional(),
  overageLimit: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachInvoiceSchema = z7.object({
  status: z7.string().nullable(),
  stripeId: z7.string(),
  total: z7.number(),
  currency: z7.string(),
  hostedInvoiceUrl: z7.string().nullable()
});
var multiAttachBasePriceOutboundSchema = z7.object({
  amount: z7.number(),
  interval: z7.string(),
  interval_count: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachResetOutboundSchema = z7.object({
  interval: z7.string(),
  interval_count: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachToOutboundSchema = z7.union([z7.number(), z7.string()]);
var multiAttachTierOutboundSchema = z7.object({
  to: z7.union([z7.number(), z7.string()]),
  amount: z7.union([z7.number(), z7.undefined()]).optional(),
  flat_amount: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachPriceOutboundSchema = z7.object({
  amount: z7.union([z7.number(), z7.undefined()]).optional(),
  tiers: z7.union([z7.array(multiAttachTierOutboundSchema), z7.undefined()]).optional(),
  tier_behavior: z7.union([z7.string(), z7.undefined()]).optional(),
  interval: z7.string(),
  interval_count: z7.number(),
  billing_units: z7.number(),
  billing_method: z7.string(),
  max_purchase: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachProrationOutboundSchema = z7.object({
  on_increase: z7.string(),
  on_decrease: z7.string()
});
var multiAttachRolloverOutboundSchema = z7.object({
  max: z7.union([z7.number(), z7.undefined()]).optional(),
  expiry_duration_type: z7.string(),
  expiry_duration_length: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachPlanItemOutboundSchema = z7.object({
  feature_id: z7.string(),
  included: z7.union([z7.number(), z7.undefined()]).optional(),
  unlimited: z7.union([z7.boolean(), z7.undefined()]).optional(),
  reset: z7.union([multiAttachResetOutboundSchema, z7.undefined()]).optional(),
  price: z7.union([multiAttachPriceOutboundSchema, z7.undefined()]).optional(),
  proration: z7.union([multiAttachProrationOutboundSchema, z7.undefined()]).optional(),
  rollover: z7.union([multiAttachRolloverOutboundSchema, z7.undefined()]).optional()
});
var multiAttachCustomizeOutboundSchema = z7.object({
  price: z7.union([multiAttachBasePriceOutboundSchema, z7.undefined()]).optional().nullable(),
  items: z7.union([z7.array(multiAttachPlanItemOutboundSchema), z7.undefined()]).optional()
});
var multiAttachFeatureQuantityOutboundSchema = z7.object({
  feature_id: z7.string(),
  quantity: z7.union([z7.number(), z7.undefined()]).optional(),
  adjustable: z7.union([z7.boolean(), z7.undefined()]).optional()
});
var multiAttachPlanOutboundSchema = z7.object({
  plan_id: z7.string(),
  customize: z7.union([multiAttachCustomizeOutboundSchema, z7.undefined()]).optional(),
  feature_quantities: z7.union([z7.array(multiAttachFeatureQuantityOutboundSchema), z7.undefined()]).optional(),
  version: z7.union([z7.number(), z7.undefined()]).optional(),
  subscription_id: z7.union([z7.string(), z7.undefined()]).optional()
});
var multiAttachFreeTrialParamsOutboundSchema = z7.object({
  duration_length: z7.number(),
  duration_type: z7.string(),
  card_required: z7.boolean()
});
var multiAttachInvoiceModeOutboundSchema = z7.object({
  enabled: z7.boolean(),
  enable_plan_immediately: z7.boolean(),
  finalize: z7.boolean()
});
var multiAttachAttachDiscountOutboundSchema = z7.object({
  reward_id: z7.union([z7.string(), z7.undefined()]).optional(),
  promotion_code: z7.union([z7.string(), z7.undefined()]).optional()
});
var multiAttachSpendLimitOutboundSchema = z7.object({
  feature_id: z7.union([z7.string(), z7.undefined()]).optional(),
  enabled: z7.boolean(),
  overage_limit: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachUsageAlertOutboundSchema = z7.object({
  feature_id: z7.union([z7.string(), z7.undefined()]).optional(),
  enabled: z7.boolean(),
  threshold: z7.number(),
  threshold_type: z7.string(),
  name: z7.union([z7.string(), z7.undefined()]).optional()
});
var multiAttachBillingControlsOutboundSchema = z7.object({
  spend_limits: z7.union([z7.array(multiAttachSpendLimitOutboundSchema), z7.undefined()]).optional(),
  usage_alerts: z7.union([z7.array(multiAttachUsageAlertOutboundSchema), z7.undefined()]).optional()
});
var multiAttachEntityDataOutboundSchema = z7.object({
  feature_id: z7.string(),
  name: z7.union([z7.string(), z7.undefined()]).optional(),
  billing_controls: z7.union([multiAttachBillingControlsOutboundSchema, z7.undefined()]).optional()
});
var closedEnumSchema4 = z7.any();
var customerDataSchema = z7.any();
var openEnumSchema3 = z7.any();
var customerDataOutboundSchema = z7.any();
var multiAttachPriceIntervalSchema = closedEnumSchema4;
var multiAttachBasePriceSchema = z7.object({
  amount: z7.number(),
  interval: multiAttachPriceIntervalSchema,
  intervalCount: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachResetIntervalSchema = closedEnumSchema4;
var multiAttachResetSchema = z7.object({
  interval: multiAttachResetIntervalSchema,
  intervalCount: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachTierBehaviorSchema = closedEnumSchema4;
var multiAttachItemPriceIntervalSchema = closedEnumSchema4;
var multiAttachBillingMethodSchema = closedEnumSchema4;
var multiAttachPriceSchema = z7.object({
  amount: z7.union([z7.number(), z7.undefined()]).optional(),
  tiers: z7.union([z7.array(multiAttachTierSchema), z7.undefined()]).optional(),
  tierBehavior: z7.union([multiAttachTierBehaviorSchema, z7.undefined()]).optional(),
  interval: multiAttachItemPriceIntervalSchema,
  intervalCount: z7.union([z7.number(), z7.undefined()]).optional(),
  billingUnits: z7.union([z7.number(), z7.undefined()]).optional(),
  billingMethod: multiAttachBillingMethodSchema,
  maxPurchase: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachOnIncreaseSchema = closedEnumSchema4;
var multiAttachOnDecreaseSchema = closedEnumSchema4;
var multiAttachProrationSchema = z7.object({
  onIncrease: multiAttachOnIncreaseSchema,
  onDecrease: multiAttachOnDecreaseSchema
});
var multiAttachExpiryDurationTypeSchema = closedEnumSchema4;
var multiAttachRolloverSchema = z7.object({
  max: z7.union([z7.number(), z7.undefined()]).optional(),
  expiryDurationType: multiAttachExpiryDurationTypeSchema,
  expiryDurationLength: z7.union([z7.number(), z7.undefined()]).optional()
});
var multiAttachPlanItemSchema = z7.object({
  featureId: z7.string(),
  included: z7.union([z7.number(), z7.undefined()]).optional(),
  unlimited: z7.union([z7.boolean(), z7.undefined()]).optional(),
  reset: z7.union([multiAttachResetSchema, z7.undefined()]).optional(),
  price: z7.union([multiAttachPriceSchema, z7.undefined()]).optional(),
  proration: z7.union([multiAttachProrationSchema, z7.undefined()]).optional(),
  rollover: z7.union([multiAttachRolloverSchema, z7.undefined()]).optional()
});
var multiAttachCustomizeSchema = z7.object({
  price: z7.union([multiAttachBasePriceSchema, z7.undefined()]).optional().nullable(),
  items: z7.union([z7.array(multiAttachPlanItemSchema), z7.undefined()]).optional()
});
var multiAttachPlanSchema = z7.object({
  planId: z7.string(),
  customize: z7.union([multiAttachCustomizeSchema, z7.undefined()]).optional(),
  featureQuantities: z7.union([z7.array(multiAttachFeatureQuantitySchema), z7.undefined()]).optional(),
  version: z7.union([z7.number(), z7.undefined()]).optional(),
  subscriptionId: z7.union([z7.string(), z7.undefined()]).optional()
});
var multiAttachDurationTypeSchema = closedEnumSchema4;
var multiAttachFreeTrialParamsSchema = z7.object({
  durationLength: z7.number(),
  durationType: z7.union([multiAttachDurationTypeSchema, z7.undefined()]).optional(),
  cardRequired: z7.union([z7.boolean(), z7.undefined()]).optional()
});
var multiAttachRedirectModeSchema = closedEnumSchema4;
var multiAttachThresholdTypeSchema = closedEnumSchema4;
var multiAttachUsageAlertSchema = z7.object({
  featureId: z7.union([z7.string(), z7.undefined()]).optional(),
  enabled: z7.union([z7.boolean(), z7.undefined()]).optional(),
  threshold: z7.number(),
  thresholdType: multiAttachThresholdTypeSchema,
  name: z7.union([z7.string(), z7.undefined()]).optional()
});
var multiAttachBillingControlsSchema = z7.object({
  spendLimits: z7.union([z7.array(multiAttachSpendLimitSchema), z7.undefined()]).optional(),
  usageAlerts: z7.union([z7.array(multiAttachUsageAlertSchema), z7.undefined()]).optional()
});
var multiAttachEntityDataSchema = z7.object({
  featureId: z7.string(),
  name: z7.union([z7.string(), z7.undefined()]).optional(),
  billingControls: z7.union([multiAttachBillingControlsSchema, z7.undefined()]).optional()
});
var multiAttachParamsSchema = z7.object({
  customerId: z7.string(),
  entityId: z7.union([z7.string(), z7.undefined()]).optional(),
  plans: z7.array(multiAttachPlanSchema),
  freeTrial: z7.union([multiAttachFreeTrialParamsSchema, z7.undefined()]).optional().nullable(),
  invoiceMode: z7.union([multiAttachInvoiceModeSchema, z7.undefined()]).optional(),
  discounts: z7.union([z7.array(multiAttachAttachDiscountSchema), z7.undefined()]).optional(),
  successUrl: z7.union([z7.string(), z7.undefined()]).optional(),
  checkoutSessionParams: z7.union([z7.record(z7.string(), z7.any()), z7.undefined()]).optional(),
  redirectMode: z7.union([multiAttachRedirectModeSchema, z7.undefined()]).optional(),
  newBillingSubscription: z7.union([z7.boolean(), z7.undefined()]).optional(),
  customerData: z7.union([customerDataSchema, z7.undefined()]).optional(),
  entityData: z7.union([multiAttachEntityDataSchema, z7.undefined()]).optional()
});
var multiAttachCodeSchema = openEnumSchema3;
var multiAttachRequiredActionSchema = z7.object({
  code: multiAttachCodeSchema,
  reason: z7.string()
});
var multiAttachResponseSchema = z7.object({
  customerId: z7.string(),
  entityId: z7.union([z7.string(), z7.undefined()]).optional(),
  invoice: z7.union([multiAttachInvoiceSchema, z7.undefined()]).optional(),
  paymentUrl: z7.string().nullable(),
  requiredAction: z7.union([multiAttachRequiredActionSchema, z7.undefined()]).optional()
});
var multiAttachParamsOutboundSchema = z7.object({
  customer_id: z7.string(),
  entity_id: z7.union([z7.string(), z7.undefined()]).optional(),
  plans: z7.array(multiAttachPlanOutboundSchema),
  free_trial: z7.union([multiAttachFreeTrialParamsOutboundSchema, z7.undefined()]).optional().nullable(),
  invoice_mode: z7.union([multiAttachInvoiceModeOutboundSchema, z7.undefined()]).optional(),
  discounts: z7.union([z7.array(multiAttachAttachDiscountOutboundSchema), z7.undefined()]).optional(),
  success_url: z7.union([z7.string(), z7.undefined()]).optional(),
  checkout_session_params: z7.union([z7.record(z7.string(), z7.any()), z7.undefined()]).optional(),
  redirect_mode: z7.string(),
  new_billing_subscription: z7.union([z7.boolean(), z7.undefined()]).optional(),
  customer_data: z7.union([customerDataOutboundSchema, z7.undefined()]).optional(),
  entity_data: z7.union([multiAttachEntityDataOutboundSchema, z7.undefined()]).optional()
});

// src/generated/openCustomerPortalSchemas.ts
import { z as z8 } from "zod/v4";
var openCustomerPortalGlobalsSchema = z8.object({
  xApiVersion: z8.union([z8.string(), z8.undefined()]).optional()
});
var openCustomerPortalParamsSchema = z8.object({
  customerId: z8.string(),
  configurationId: z8.union([z8.string(), z8.undefined()]).optional(),
  returnUrl: z8.union([z8.string(), z8.undefined()]).optional()
});
var openCustomerPortalResponseSchema = z8.object({
  customerId: z8.string(),
  url: z8.string()
});
var openCustomerPortalParamsOutboundSchema = z8.object({
  customer_id: z8.string(),
  configuration_id: z8.union([z8.string(), z8.undefined()]).optional(),
  return_url: z8.union([z8.string(), z8.undefined()]).optional()
});

// src/generated/previewAttachSchemas.ts
import { z as z9 } from "zod/v4";
var previewAttachGlobalsSchema = z9.object({
  xApiVersion: z9.union([z9.string(), z9.undefined()]).optional()
});
var previewAttachFeatureQuantityRequestSchema = z9.object({
  featureId: z9.string(),
  quantity: z9.union([z9.number(), z9.undefined()]).optional(),
  adjustable: z9.union([z9.boolean(), z9.undefined()]).optional()
});
var previewAttachToSchema = z9.union([z9.number(), z9.string()]);
var previewAttachTierSchema = z9.object({
  to: z9.union([z9.number(), z9.string()]),
  amount: z9.union([z9.number(), z9.undefined()]).optional(),
  flatAmount: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachInvoiceModeSchema = z9.object({
  enabled: z9.boolean(),
  enablePlanImmediately: z9.union([z9.boolean(), z9.undefined()]).optional(),
  finalize: z9.union([z9.boolean(), z9.undefined()]).optional()
});
var previewAttachAttachDiscountSchema = z9.object({
  rewardId: z9.union([z9.string(), z9.undefined()]).optional(),
  promotionCode: z9.union([z9.string(), z9.undefined()]).optional()
});
var previewAttachCustomLineItemSchema = z9.object({
  amount: z9.number(),
  description: z9.string()
});
var previewAttachCarryOverBalancesSchema = z9.object({
  enabled: z9.boolean(),
  featureIds: z9.union([z9.array(z9.string()), z9.undefined()]).optional()
});
var previewAttachCarryOverUsagesSchema = z9.object({
  enabled: z9.boolean(),
  featureIds: z9.union([z9.array(z9.string()), z9.undefined()]).optional()
});
var previewAttachDiscountSchema = z9.object({
  amountOff: z9.number(),
  percentOff: z9.union([z9.number(), z9.undefined()]).optional(),
  rewardId: z9.union([z9.string(), z9.undefined()]).optional(),
  rewardName: z9.union([z9.string(), z9.undefined()]).optional()
});
var previewAttachLineItemPeriodSchema = z9.object({
  start: z9.number(),
  end: z9.number()
});
var previewAttachLineItemSchema = z9.object({
  displayName: z9.string(),
  description: z9.string(),
  subtotal: z9.number(),
  total: z9.number(),
  discounts: z9.union([z9.array(previewAttachDiscountSchema), z9.undefined()]).optional(),
  planId: z9.string(),
  featureId: z9.string().nullable(),
  period: z9.union([previewAttachLineItemPeriodSchema, z9.undefined()]).optional(),
  quantity: z9.number()
});
var previewAttachNextCycleDiscountSchema = z9.object({
  amountOff: z9.number(),
  percentOff: z9.union([z9.number(), z9.undefined()]).optional(),
  rewardId: z9.union([z9.string(), z9.undefined()]).optional(),
  rewardName: z9.union([z9.string(), z9.undefined()]).optional()
});
var previewAttachNextCycleLineItemPeriodSchema = z9.object({
  start: z9.number(),
  end: z9.number()
});
var previewAttachNextCycleLineItemSchema = z9.object({
  displayName: z9.string(),
  description: z9.string(),
  subtotal: z9.number(),
  total: z9.number(),
  discounts: z9.union([z9.array(previewAttachNextCycleDiscountSchema), z9.undefined()]).optional(),
  planId: z9.string(),
  featureId: z9.string().nullable(),
  period: z9.union([previewAttachNextCycleLineItemPeriodSchema, z9.undefined()]).optional(),
  quantity: z9.number()
});
var previewAttachUsageLineItemPeriodSchema = z9.object({
  start: z9.number(),
  end: z9.number()
});
var previewAttachUsageLineItemSchema = z9.object({
  displayName: z9.string(),
  planId: z9.string(),
  featureId: z9.string().nullable(),
  period: z9.union([previewAttachUsageLineItemPeriodSchema, z9.undefined()]).optional()
});
var previewAttachNextCycleSchema = z9.object({
  startsAt: z9.number(),
  subtotal: z9.number(),
  total: z9.number(),
  lineItems: z9.array(previewAttachNextCycleLineItemSchema),
  usageLineItems: z9.array(previewAttachUsageLineItemSchema)
});
var previewAttachIncomingFeatureQuantitySchema = z9.object({
  featureId: z9.string(),
  quantity: z9.number()
});
var previewAttachOutgoingFeatureQuantitySchema = z9.object({
  featureId: z9.string(),
  quantity: z9.number()
});
var previewAttachFeatureQuantityRequestOutboundSchema = z9.object({
  feature_id: z9.string(),
  quantity: z9.union([z9.number(), z9.undefined()]).optional(),
  adjustable: z9.union([z9.boolean(), z9.undefined()]).optional()
});
var previewAttachBasePriceOutboundSchema = z9.object({
  amount: z9.number(),
  interval: z9.string(),
  interval_count: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachResetOutboundSchema = z9.object({
  interval: z9.string(),
  interval_count: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachToOutboundSchema = z9.union([z9.number(), z9.string()]);
var previewAttachTierOutboundSchema = z9.object({
  to: z9.union([z9.number(), z9.string()]),
  amount: z9.union([z9.number(), z9.undefined()]).optional(),
  flat_amount: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachPriceOutboundSchema = z9.object({
  amount: z9.union([z9.number(), z9.undefined()]).optional(),
  tiers: z9.union([z9.array(previewAttachTierOutboundSchema), z9.undefined()]).optional(),
  tier_behavior: z9.union([z9.string(), z9.undefined()]).optional(),
  interval: z9.string(),
  interval_count: z9.number(),
  billing_units: z9.number(),
  billing_method: z9.string(),
  max_purchase: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachProrationOutboundSchema = z9.object({
  on_increase: z9.string(),
  on_decrease: z9.string()
});
var previewAttachRolloverOutboundSchema = z9.object({
  max: z9.union([z9.number(), z9.undefined()]).optional(),
  expiry_duration_type: z9.string(),
  expiry_duration_length: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachPlanItemOutboundSchema = z9.object({
  feature_id: z9.string(),
  included: z9.union([z9.number(), z9.undefined()]).optional(),
  unlimited: z9.union([z9.boolean(), z9.undefined()]).optional(),
  reset: z9.union([previewAttachResetOutboundSchema, z9.undefined()]).optional(),
  price: z9.union([previewAttachPriceOutboundSchema, z9.undefined()]).optional(),
  proration: z9.union([previewAttachProrationOutboundSchema, z9.undefined()]).optional(),
  rollover: z9.union([previewAttachRolloverOutboundSchema, z9.undefined()]).optional()
});
var previewAttachFreeTrialParamsOutboundSchema = z9.object({
  duration_length: z9.number(),
  duration_type: z9.string(),
  card_required: z9.boolean()
});
var previewAttachCustomizeOutboundSchema = z9.object({
  price: z9.union([previewAttachBasePriceOutboundSchema, z9.undefined()]).optional().nullable(),
  items: z9.union([z9.array(previewAttachPlanItemOutboundSchema), z9.undefined()]).optional(),
  free_trial: z9.union([previewAttachFreeTrialParamsOutboundSchema, z9.undefined()]).optional().nullable()
});
var previewAttachInvoiceModeOutboundSchema = z9.object({
  enabled: z9.boolean(),
  enable_plan_immediately: z9.boolean(),
  finalize: z9.boolean()
});
var previewAttachAttachDiscountOutboundSchema = z9.object({
  reward_id: z9.union([z9.string(), z9.undefined()]).optional(),
  promotion_code: z9.union([z9.string(), z9.undefined()]).optional()
});
var previewAttachCustomLineItemOutboundSchema = z9.object({
  amount: z9.number(),
  description: z9.string()
});
var previewAttachCarryOverBalancesOutboundSchema = z9.object({
  enabled: z9.boolean(),
  feature_ids: z9.union([z9.array(z9.string()), z9.undefined()]).optional()
});
var previewAttachCarryOverUsagesOutboundSchema = z9.object({
  enabled: z9.boolean(),
  feature_ids: z9.union([z9.array(z9.string()), z9.undefined()]).optional()
});
var previewAttachParamsOutboundSchema = z9.object({
  customer_id: z9.string(),
  entity_id: z9.union([z9.string(), z9.undefined()]).optional(),
  plan_id: z9.string(),
  feature_quantities: z9.union([
    z9.array(previewAttachFeatureQuantityRequestOutboundSchema),
    z9.undefined()
  ]).optional(),
  version: z9.union([z9.number(), z9.undefined()]).optional(),
  customize: z9.union([previewAttachCustomizeOutboundSchema, z9.undefined()]).optional(),
  invoice_mode: z9.union([previewAttachInvoiceModeOutboundSchema, z9.undefined()]).optional(),
  proration_behavior: z9.union([z9.string(), z9.undefined()]).optional(),
  redirect_mode: z9.string(),
  subscription_id: z9.union([z9.string(), z9.undefined()]).optional(),
  discounts: z9.union([z9.array(previewAttachAttachDiscountOutboundSchema), z9.undefined()]).optional(),
  success_url: z9.union([z9.string(), z9.undefined()]).optional(),
  new_billing_subscription: z9.union([z9.boolean(), z9.undefined()]).optional(),
  plan_schedule: z9.union([z9.string(), z9.undefined()]).optional(),
  checkout_session_params: z9.union([z9.record(z9.string(), z9.any()), z9.undefined()]).optional(),
  custom_line_items: z9.union([z9.array(previewAttachCustomLineItemOutboundSchema), z9.undefined()]).optional(),
  processor_subscription_id: z9.union([z9.string(), z9.undefined()]).optional(),
  carry_over_balances: z9.union([previewAttachCarryOverBalancesOutboundSchema, z9.undefined()]).optional(),
  carry_over_usages: z9.union([previewAttachCarryOverUsagesOutboundSchema, z9.undefined()]).optional(),
  metadata: z9.union([z9.record(z9.string(), z9.string()), z9.undefined()]).optional()
});
var closedEnumSchema5 = z9.any();
var planSchema = z9.any();
var openEnumSchema4 = z9.any();
var previewAttachPriceIntervalSchema = closedEnumSchema5;
var previewAttachBasePriceSchema = z9.object({
  amount: z9.number(),
  interval: previewAttachPriceIntervalSchema,
  intervalCount: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachResetIntervalSchema = closedEnumSchema5;
var previewAttachResetSchema = z9.object({
  interval: previewAttachResetIntervalSchema,
  intervalCount: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachTierBehaviorSchema = closedEnumSchema5;
var previewAttachItemPriceIntervalSchema = closedEnumSchema5;
var previewAttachBillingMethodSchema = closedEnumSchema5;
var previewAttachPriceSchema = z9.object({
  amount: z9.union([z9.number(), z9.undefined()]).optional(),
  tiers: z9.union([z9.array(previewAttachTierSchema), z9.undefined()]).optional(),
  tierBehavior: z9.union([previewAttachTierBehaviorSchema, z9.undefined()]).optional(),
  interval: previewAttachItemPriceIntervalSchema,
  intervalCount: z9.union([z9.number(), z9.undefined()]).optional(),
  billingUnits: z9.union([z9.number(), z9.undefined()]).optional(),
  billingMethod: previewAttachBillingMethodSchema,
  maxPurchase: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachOnIncreaseSchema = closedEnumSchema5;
var previewAttachOnDecreaseSchema = closedEnumSchema5;
var previewAttachProrationSchema = z9.object({
  onIncrease: previewAttachOnIncreaseSchema,
  onDecrease: previewAttachOnDecreaseSchema
});
var previewAttachExpiryDurationTypeSchema = closedEnumSchema5;
var previewAttachRolloverSchema = z9.object({
  max: z9.union([z9.number(), z9.undefined()]).optional(),
  expiryDurationType: previewAttachExpiryDurationTypeSchema,
  expiryDurationLength: z9.union([z9.number(), z9.undefined()]).optional()
});
var previewAttachPlanItemSchema = z9.object({
  featureId: z9.string(),
  included: z9.union([z9.number(), z9.undefined()]).optional(),
  unlimited: z9.union([z9.boolean(), z9.undefined()]).optional(),
  reset: z9.union([previewAttachResetSchema, z9.undefined()]).optional(),
  price: z9.union([previewAttachPriceSchema, z9.undefined()]).optional(),
  proration: z9.union([previewAttachProrationSchema, z9.undefined()]).optional(),
  rollover: z9.union([previewAttachRolloverSchema, z9.undefined()]).optional()
});
var previewAttachDurationTypeSchema = closedEnumSchema5;
var previewAttachFreeTrialParamsSchema = z9.object({
  durationLength: z9.number(),
  durationType: z9.union([previewAttachDurationTypeSchema, z9.undefined()]).optional(),
  cardRequired: z9.union([z9.boolean(), z9.undefined()]).optional()
});
var previewAttachCustomizeSchema = z9.object({
  price: z9.union([previewAttachBasePriceSchema, z9.undefined()]).optional().nullable(),
  items: z9.union([z9.array(previewAttachPlanItemSchema), z9.undefined()]).optional(),
  freeTrial: z9.union([previewAttachFreeTrialParamsSchema, z9.undefined()]).optional().nullable()
});
var previewAttachProrationBehaviorSchema = closedEnumSchema5;
var previewAttachRedirectModeSchema = closedEnumSchema5;
var previewAttachPlanScheduleSchema = closedEnumSchema5;
var previewAttachParamsSchema = z9.object({
  customerId: z9.string(),
  entityId: z9.union([z9.string(), z9.undefined()]).optional(),
  planId: z9.string(),
  featureQuantities: z9.union([z9.array(previewAttachFeatureQuantityRequestSchema), z9.undefined()]).optional(),
  version: z9.union([z9.number(), z9.undefined()]).optional(),
  customize: z9.union([previewAttachCustomizeSchema, z9.undefined()]).optional(),
  invoiceMode: z9.union([previewAttachInvoiceModeSchema, z9.undefined()]).optional(),
  prorationBehavior: z9.union([previewAttachProrationBehaviorSchema, z9.undefined()]).optional(),
  redirectMode: z9.union([previewAttachRedirectModeSchema, z9.undefined()]).optional(),
  subscriptionId: z9.union([z9.string(), z9.undefined()]).optional(),
  discounts: z9.union([z9.array(previewAttachAttachDiscountSchema), z9.undefined()]).optional(),
  successUrl: z9.union([z9.string(), z9.undefined()]).optional(),
  newBillingSubscription: z9.union([z9.boolean(), z9.undefined()]).optional(),
  planSchedule: z9.union([previewAttachPlanScheduleSchema, z9.undefined()]).optional(),
  checkoutSessionParams: z9.union([z9.record(z9.string(), z9.any()), z9.undefined()]).optional(),
  customLineItems: z9.union([z9.array(previewAttachCustomLineItemSchema), z9.undefined()]).optional(),
  processorSubscriptionId: z9.union([z9.string(), z9.undefined()]).optional(),
  carryOverBalances: z9.union([previewAttachCarryOverBalancesSchema, z9.undefined()]).optional(),
  carryOverUsages: z9.union([previewAttachCarryOverUsagesSchema, z9.undefined()]).optional(),
  metadata: z9.union([z9.record(z9.string(), z9.string()), z9.undefined()]).optional()
});
var previewAttachIncomingSchema = z9.object({
  planId: z9.string(),
  plan: z9.union([planSchema, z9.undefined()]).optional(),
  featureQuantities: z9.array(previewAttachIncomingFeatureQuantitySchema),
  effectiveAt: z9.number().nullable(),
  canceledAt: z9.number().nullable(),
  expiresAt: z9.number().nullable()
});
var previewAttachOutgoingSchema = z9.object({
  planId: z9.string(),
  plan: z9.union([planSchema, z9.undefined()]).optional(),
  featureQuantities: z9.array(previewAttachOutgoingFeatureQuantitySchema),
  effectiveAt: z9.number().nullable(),
  canceledAt: z9.number().nullable(),
  expiresAt: z9.number().nullable()
});
var previewAttachCheckoutTypeSchema = openEnumSchema4;
var previewAttachResponseSchema = z9.object({
  customerId: z9.string(),
  lineItems: z9.array(previewAttachLineItemSchema),
  subtotal: z9.number(),
  total: z9.number(),
  currency: z9.string(),
  nextCycle: z9.union([previewAttachNextCycleSchema, z9.undefined()]).optional(),
  expand: z9.union([z9.array(z9.string()), z9.undefined()]).optional(),
  incoming: z9.array(previewAttachIncomingSchema),
  outgoing: z9.array(previewAttachOutgoingSchema),
  redirectToCheckout: z9.boolean(),
  checkoutType: previewAttachCheckoutTypeSchema.nullable()
});

// src/generated/previewMultiAttachSchemas.ts
import { z as z10 } from "zod/v4";
var previewMultiAttachGlobalsSchema = z10.object({
  xApiVersion: z10.union([z10.string(), z10.undefined()]).optional()
});
var previewMultiAttachToSchema = z10.union([z10.number(), z10.string()]);
var previewMultiAttachTierSchema = z10.object({
  to: z10.union([z10.number(), z10.string()]),
  amount: z10.union([z10.number(), z10.undefined()]).optional(),
  flatAmount: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachPlanFeatureQuantitySchema = z10.object({
  featureId: z10.string(),
  quantity: z10.union([z10.number(), z10.undefined()]).optional(),
  adjustable: z10.union([z10.boolean(), z10.undefined()]).optional()
});
var previewMultiAttachInvoiceModeSchema = z10.object({
  enabled: z10.boolean(),
  enablePlanImmediately: z10.union([z10.boolean(), z10.undefined()]).optional(),
  finalize: z10.union([z10.boolean(), z10.undefined()]).optional()
});
var previewMultiAttachAttachDiscountSchema = z10.object({
  rewardId: z10.union([z10.string(), z10.undefined()]).optional(),
  promotionCode: z10.union([z10.string(), z10.undefined()]).optional()
});
var previewMultiAttachSpendLimitSchema = z10.object({
  featureId: z10.union([z10.string(), z10.undefined()]).optional(),
  enabled: z10.union([z10.boolean(), z10.undefined()]).optional(),
  overageLimit: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachDiscountSchema = z10.object({
  amountOff: z10.number(),
  percentOff: z10.union([z10.number(), z10.undefined()]).optional(),
  rewardId: z10.union([z10.string(), z10.undefined()]).optional(),
  rewardName: z10.union([z10.string(), z10.undefined()]).optional()
});
var previewMultiAttachLineItemPeriodSchema = z10.object({
  start: z10.number(),
  end: z10.number()
});
var previewMultiAttachLineItemSchema = z10.object({
  displayName: z10.string(),
  description: z10.string(),
  subtotal: z10.number(),
  total: z10.number(),
  discounts: z10.union([z10.array(previewMultiAttachDiscountSchema), z10.undefined()]).optional(),
  planId: z10.string(),
  featureId: z10.string().nullable(),
  period: z10.union([previewMultiAttachLineItemPeriodSchema, z10.undefined()]).optional(),
  quantity: z10.number()
});
var previewMultiAttachNextCycleDiscountSchema = z10.object({
  amountOff: z10.number(),
  percentOff: z10.union([z10.number(), z10.undefined()]).optional(),
  rewardId: z10.union([z10.string(), z10.undefined()]).optional(),
  rewardName: z10.union([z10.string(), z10.undefined()]).optional()
});
var previewMultiAttachNextCycleLineItemPeriodSchema = z10.object({
  start: z10.number(),
  end: z10.number()
});
var previewMultiAttachNextCycleLineItemSchema = z10.object({
  displayName: z10.string(),
  description: z10.string(),
  subtotal: z10.number(),
  total: z10.number(),
  discounts: z10.union([z10.array(previewMultiAttachNextCycleDiscountSchema), z10.undefined()]).optional(),
  planId: z10.string(),
  featureId: z10.string().nullable(),
  period: z10.union([previewMultiAttachNextCycleLineItemPeriodSchema, z10.undefined()]).optional(),
  quantity: z10.number()
});
var previewMultiAttachUsageLineItemPeriodSchema = z10.object({
  start: z10.number(),
  end: z10.number()
});
var previewMultiAttachUsageLineItemSchema = z10.object({
  displayName: z10.string(),
  planId: z10.string(),
  featureId: z10.string().nullable(),
  period: z10.union([previewMultiAttachUsageLineItemPeriodSchema, z10.undefined()]).optional()
});
var previewMultiAttachNextCycleSchema = z10.object({
  startsAt: z10.number(),
  subtotal: z10.number(),
  total: z10.number(),
  lineItems: z10.array(previewMultiAttachNextCycleLineItemSchema),
  usageLineItems: z10.array(previewMultiAttachUsageLineItemSchema)
});
var previewMultiAttachIncomingFeatureQuantitySchema = z10.object({
  featureId: z10.string(),
  quantity: z10.number()
});
var previewMultiAttachOutgoingFeatureQuantitySchema = z10.object({
  featureId: z10.string(),
  quantity: z10.number()
});
var previewMultiAttachBasePriceOutboundSchema = z10.object({
  amount: z10.number(),
  interval: z10.string(),
  interval_count: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachResetOutboundSchema = z10.object({
  interval: z10.string(),
  interval_count: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachToOutboundSchema = z10.union([
  z10.number(),
  z10.string()
]);
var previewMultiAttachTierOutboundSchema = z10.object({
  to: z10.union([z10.number(), z10.string()]),
  amount: z10.union([z10.number(), z10.undefined()]).optional(),
  flat_amount: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachPriceOutboundSchema = z10.object({
  amount: z10.union([z10.number(), z10.undefined()]).optional(),
  tiers: z10.union([z10.array(previewMultiAttachTierOutboundSchema), z10.undefined()]).optional(),
  tier_behavior: z10.union([z10.string(), z10.undefined()]).optional(),
  interval: z10.string(),
  interval_count: z10.number(),
  billing_units: z10.number(),
  billing_method: z10.string(),
  max_purchase: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachProrationOutboundSchema = z10.object({
  on_increase: z10.string(),
  on_decrease: z10.string()
});
var previewMultiAttachRolloverOutboundSchema = z10.object({
  max: z10.union([z10.number(), z10.undefined()]).optional(),
  expiry_duration_type: z10.string(),
  expiry_duration_length: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachPlanItemOutboundSchema = z10.object({
  feature_id: z10.string(),
  included: z10.union([z10.number(), z10.undefined()]).optional(),
  unlimited: z10.union([z10.boolean(), z10.undefined()]).optional(),
  reset: z10.union([previewMultiAttachResetOutboundSchema, z10.undefined()]).optional(),
  price: z10.union([previewMultiAttachPriceOutboundSchema, z10.undefined()]).optional(),
  proration: z10.union([previewMultiAttachProrationOutboundSchema, z10.undefined()]).optional(),
  rollover: z10.union([previewMultiAttachRolloverOutboundSchema, z10.undefined()]).optional()
});
var previewMultiAttachCustomizeOutboundSchema = z10.object({
  price: z10.union([previewMultiAttachBasePriceOutboundSchema, z10.undefined()]).optional().nullable(),
  items: z10.union([z10.array(previewMultiAttachPlanItemOutboundSchema), z10.undefined()]).optional()
});
var previewMultiAttachPlanFeatureQuantityOutboundSchema = z10.object({
  feature_id: z10.string(),
  quantity: z10.union([z10.number(), z10.undefined()]).optional(),
  adjustable: z10.union([z10.boolean(), z10.undefined()]).optional()
});
var previewMultiAttachPlanOutboundSchema = z10.object({
  plan_id: z10.string(),
  customize: z10.union([previewMultiAttachCustomizeOutboundSchema, z10.undefined()]).optional(),
  feature_quantities: z10.union([
    z10.array(previewMultiAttachPlanFeatureQuantityOutboundSchema),
    z10.undefined()
  ]).optional(),
  version: z10.union([z10.number(), z10.undefined()]).optional(),
  subscription_id: z10.union([z10.string(), z10.undefined()]).optional()
});
var previewMultiAttachFreeTrialParamsOutboundSchema = z10.object({
  duration_length: z10.number(),
  duration_type: z10.string(),
  card_required: z10.boolean()
});
var previewMultiAttachInvoiceModeOutboundSchema = z10.object({
  enabled: z10.boolean(),
  enable_plan_immediately: z10.boolean(),
  finalize: z10.boolean()
});
var previewMultiAttachAttachDiscountOutboundSchema = z10.object({
  reward_id: z10.union([z10.string(), z10.undefined()]).optional(),
  promotion_code: z10.union([z10.string(), z10.undefined()]).optional()
});
var previewMultiAttachSpendLimitOutboundSchema = z10.object({
  feature_id: z10.union([z10.string(), z10.undefined()]).optional(),
  enabled: z10.boolean(),
  overage_limit: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachUsageAlertOutboundSchema = z10.object({
  feature_id: z10.union([z10.string(), z10.undefined()]).optional(),
  enabled: z10.boolean(),
  threshold: z10.number(),
  threshold_type: z10.string(),
  name: z10.union([z10.string(), z10.undefined()]).optional()
});
var previewMultiAttachBillingControlsOutboundSchema = z10.object({
  spend_limits: z10.union([z10.array(previewMultiAttachSpendLimitOutboundSchema), z10.undefined()]).optional(),
  usage_alerts: z10.union([z10.array(previewMultiAttachUsageAlertOutboundSchema), z10.undefined()]).optional()
});
var previewMultiAttachEntityDataOutboundSchema = z10.object({
  feature_id: z10.string(),
  name: z10.union([z10.string(), z10.undefined()]).optional(),
  billing_controls: z10.union([previewMultiAttachBillingControlsOutboundSchema, z10.undefined()]).optional()
});
var closedEnumSchema6 = z10.any();
var customerDataSchema2 = z10.any();
var planSchema2 = z10.any();
var openEnumSchema5 = z10.any();
var customerDataOutboundSchema2 = z10.any();
var previewMultiAttachPriceIntervalSchema = closedEnumSchema6;
var previewMultiAttachBasePriceSchema = z10.object({
  amount: z10.number(),
  interval: previewMultiAttachPriceIntervalSchema,
  intervalCount: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachResetIntervalSchema = closedEnumSchema6;
var previewMultiAttachResetSchema = z10.object({
  interval: previewMultiAttachResetIntervalSchema,
  intervalCount: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachTierBehaviorSchema = closedEnumSchema6;
var previewMultiAttachItemPriceIntervalSchema = closedEnumSchema6;
var previewMultiAttachBillingMethodSchema = closedEnumSchema6;
var previewMultiAttachPriceSchema = z10.object({
  amount: z10.union([z10.number(), z10.undefined()]).optional(),
  tiers: z10.union([z10.array(previewMultiAttachTierSchema), z10.undefined()]).optional(),
  tierBehavior: z10.union([previewMultiAttachTierBehaviorSchema, z10.undefined()]).optional(),
  interval: previewMultiAttachItemPriceIntervalSchema,
  intervalCount: z10.union([z10.number(), z10.undefined()]).optional(),
  billingUnits: z10.union([z10.number(), z10.undefined()]).optional(),
  billingMethod: previewMultiAttachBillingMethodSchema,
  maxPurchase: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachOnIncreaseSchema = closedEnumSchema6;
var previewMultiAttachOnDecreaseSchema = closedEnumSchema6;
var previewMultiAttachProrationSchema = z10.object({
  onIncrease: previewMultiAttachOnIncreaseSchema,
  onDecrease: previewMultiAttachOnDecreaseSchema
});
var previewMultiAttachExpiryDurationTypeSchema = closedEnumSchema6;
var previewMultiAttachRolloverSchema = z10.object({
  max: z10.union([z10.number(), z10.undefined()]).optional(),
  expiryDurationType: previewMultiAttachExpiryDurationTypeSchema,
  expiryDurationLength: z10.union([z10.number(), z10.undefined()]).optional()
});
var previewMultiAttachPlanItemSchema = z10.object({
  featureId: z10.string(),
  included: z10.union([z10.number(), z10.undefined()]).optional(),
  unlimited: z10.union([z10.boolean(), z10.undefined()]).optional(),
  reset: z10.union([previewMultiAttachResetSchema, z10.undefined()]).optional(),
  price: z10.union([previewMultiAttachPriceSchema, z10.undefined()]).optional(),
  proration: z10.union([previewMultiAttachProrationSchema, z10.undefined()]).optional(),
  rollover: z10.union([previewMultiAttachRolloverSchema, z10.undefined()]).optional()
});
var previewMultiAttachCustomizeSchema = z10.object({
  price: z10.union([previewMultiAttachBasePriceSchema, z10.undefined()]).optional().nullable(),
  items: z10.union([z10.array(previewMultiAttachPlanItemSchema), z10.undefined()]).optional()
});
var previewMultiAttachPlanSchema = z10.object({
  planId: z10.string(),
  customize: z10.union([previewMultiAttachCustomizeSchema, z10.undefined()]).optional(),
  featureQuantities: z10.union([
    z10.array(previewMultiAttachPlanFeatureQuantitySchema),
    z10.undefined()
  ]).optional(),
  version: z10.union([z10.number(), z10.undefined()]).optional(),
  subscriptionId: z10.union([z10.string(), z10.undefined()]).optional()
});
var previewMultiAttachDurationTypeSchema = closedEnumSchema6;
var previewMultiAttachFreeTrialParamsSchema = z10.object({
  durationLength: z10.number(),
  durationType: z10.union([previewMultiAttachDurationTypeSchema, z10.undefined()]).optional(),
  cardRequired: z10.union([z10.boolean(), z10.undefined()]).optional()
});
var previewMultiAttachRedirectModeSchema = closedEnumSchema6;
var previewMultiAttachThresholdTypeSchema = closedEnumSchema6;
var previewMultiAttachUsageAlertSchema = z10.object({
  featureId: z10.union([z10.string(), z10.undefined()]).optional(),
  enabled: z10.union([z10.boolean(), z10.undefined()]).optional(),
  threshold: z10.number(),
  thresholdType: previewMultiAttachThresholdTypeSchema,
  name: z10.union([z10.string(), z10.undefined()]).optional()
});
var previewMultiAttachBillingControlsSchema = z10.object({
  spendLimits: z10.union([z10.array(previewMultiAttachSpendLimitSchema), z10.undefined()]).optional(),
  usageAlerts: z10.union([z10.array(previewMultiAttachUsageAlertSchema), z10.undefined()]).optional()
});
var previewMultiAttachEntityDataSchema = z10.object({
  featureId: z10.string(),
  name: z10.union([z10.string(), z10.undefined()]).optional(),
  billingControls: z10.union([previewMultiAttachBillingControlsSchema, z10.undefined()]).optional()
});
var previewMultiAttachParamsSchema = z10.object({
  customerId: z10.string(),
  entityId: z10.union([z10.string(), z10.undefined()]).optional(),
  plans: z10.array(previewMultiAttachPlanSchema),
  freeTrial: z10.union([previewMultiAttachFreeTrialParamsSchema, z10.undefined()]).optional().nullable(),
  invoiceMode: z10.union([previewMultiAttachInvoiceModeSchema, z10.undefined()]).optional(),
  discounts: z10.union([z10.array(previewMultiAttachAttachDiscountSchema), z10.undefined()]).optional(),
  successUrl: z10.union([z10.string(), z10.undefined()]).optional(),
  checkoutSessionParams: z10.union([z10.record(z10.string(), z10.any()), z10.undefined()]).optional(),
  redirectMode: z10.union([previewMultiAttachRedirectModeSchema, z10.undefined()]).optional(),
  newBillingSubscription: z10.union([z10.boolean(), z10.undefined()]).optional(),
  customerData: z10.union([customerDataSchema2, z10.undefined()]).optional(),
  entityData: z10.union([previewMultiAttachEntityDataSchema, z10.undefined()]).optional()
});
var previewMultiAttachIncomingSchema = z10.object({
  planId: z10.string(),
  plan: z10.union([planSchema2, z10.undefined()]).optional(),
  featureQuantities: z10.array(previewMultiAttachIncomingFeatureQuantitySchema),
  effectiveAt: z10.number().nullable(),
  canceledAt: z10.number().nullable(),
  expiresAt: z10.number().nullable()
});
var previewMultiAttachOutgoingSchema = z10.object({
  planId: z10.string(),
  plan: z10.union([planSchema2, z10.undefined()]).optional(),
  featureQuantities: z10.array(previewMultiAttachOutgoingFeatureQuantitySchema),
  effectiveAt: z10.number().nullable(),
  canceledAt: z10.number().nullable(),
  expiresAt: z10.number().nullable()
});
var previewMultiAttachCheckoutTypeSchema = openEnumSchema5;
var previewMultiAttachResponseSchema = z10.object({
  customerId: z10.string(),
  lineItems: z10.array(previewMultiAttachLineItemSchema),
  subtotal: z10.number(),
  total: z10.number(),
  currency: z10.string(),
  nextCycle: z10.union([previewMultiAttachNextCycleSchema, z10.undefined()]).optional(),
  expand: z10.union([z10.array(z10.string()), z10.undefined()]).optional(),
  incoming: z10.array(previewMultiAttachIncomingSchema),
  outgoing: z10.array(previewMultiAttachOutgoingSchema),
  redirectToCheckout: z10.boolean(),
  checkoutType: previewMultiAttachCheckoutTypeSchema.nullable()
});
var previewMultiAttachParamsOutboundSchema = z10.object({
  customer_id: z10.string(),
  entity_id: z10.union([z10.string(), z10.undefined()]).optional(),
  plans: z10.array(previewMultiAttachPlanOutboundSchema),
  free_trial: z10.union([previewMultiAttachFreeTrialParamsOutboundSchema, z10.undefined()]).optional().nullable(),
  invoice_mode: z10.union([previewMultiAttachInvoiceModeOutboundSchema, z10.undefined()]).optional(),
  discounts: z10.union([
    z10.array(previewMultiAttachAttachDiscountOutboundSchema),
    z10.undefined()
  ]).optional(),
  success_url: z10.union([z10.string(), z10.undefined()]).optional(),
  checkout_session_params: z10.union([z10.record(z10.string(), z10.any()), z10.undefined()]).optional(),
  redirect_mode: z10.string(),
  new_billing_subscription: z10.union([z10.boolean(), z10.undefined()]).optional(),
  customer_data: z10.union([customerDataOutboundSchema2, z10.undefined()]).optional(),
  entity_data: z10.union([previewMultiAttachEntityDataOutboundSchema, z10.undefined()]).optional()
});

// src/generated/previewUpdateSubscriptionSchemas.ts
import { z as z11 } from "zod/v4";
var previewUpdateGlobalsSchema = z11.object({
  xApiVersion: z11.union([z11.string(), z11.undefined()]).optional()
});
var previewUpdateFeatureQuantityRequestSchema = z11.object({
  featureId: z11.string(),
  quantity: z11.union([z11.number(), z11.undefined()]).optional(),
  adjustable: z11.union([z11.boolean(), z11.undefined()]).optional()
});
var previewUpdateToSchema = z11.union([z11.number(), z11.string()]);
var previewUpdateTierSchema = z11.object({
  to: z11.union([z11.number(), z11.string()]),
  amount: z11.union([z11.number(), z11.undefined()]).optional(),
  flatAmount: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdateInvoiceModeSchema = z11.object({
  enabled: z11.boolean(),
  enablePlanImmediately: z11.union([z11.boolean(), z11.undefined()]).optional(),
  finalize: z11.union([z11.boolean(), z11.undefined()]).optional()
});
var previewUpdateRecalculateBalancesSchema = z11.object({
  enabled: z11.boolean()
});
var previewUpdateDiscountSchema = z11.object({
  amountOff: z11.number(),
  percentOff: z11.union([z11.number(), z11.undefined()]).optional(),
  rewardId: z11.union([z11.string(), z11.undefined()]).optional(),
  rewardName: z11.union([z11.string(), z11.undefined()]).optional()
});
var previewUpdateLineItemPeriodSchema = z11.object({
  start: z11.number(),
  end: z11.number()
});
var previewUpdateLineItemSchema = z11.object({
  displayName: z11.string(),
  description: z11.string(),
  subtotal: z11.number(),
  total: z11.number(),
  discounts: z11.union([z11.array(previewUpdateDiscountSchema), z11.undefined()]).optional(),
  planId: z11.string(),
  featureId: z11.string().nullable(),
  period: z11.union([previewUpdateLineItemPeriodSchema, z11.undefined()]).optional(),
  quantity: z11.number()
});
var previewUpdateNextCycleDiscountSchema = z11.object({
  amountOff: z11.number(),
  percentOff: z11.union([z11.number(), z11.undefined()]).optional(),
  rewardId: z11.union([z11.string(), z11.undefined()]).optional(),
  rewardName: z11.union([z11.string(), z11.undefined()]).optional()
});
var previewUpdateNextCycleLineItemPeriodSchema = z11.object({
  start: z11.number(),
  end: z11.number()
});
var previewUpdateNextCycleLineItemSchema = z11.object({
  displayName: z11.string(),
  description: z11.string(),
  subtotal: z11.number(),
  total: z11.number(),
  discounts: z11.union([z11.array(previewUpdateNextCycleDiscountSchema), z11.undefined()]).optional(),
  planId: z11.string(),
  featureId: z11.string().nullable(),
  period: z11.union([previewUpdateNextCycleLineItemPeriodSchema, z11.undefined()]).optional(),
  quantity: z11.number()
});
var previewUpdateUsageLineItemPeriodSchema = z11.object({
  start: z11.number(),
  end: z11.number()
});
var previewUpdateUsageLineItemSchema = z11.object({
  displayName: z11.string(),
  planId: z11.string(),
  featureId: z11.string().nullable(),
  period: z11.union([previewUpdateUsageLineItemPeriodSchema, z11.undefined()]).optional()
});
var previewUpdateNextCycleSchema = z11.object({
  startsAt: z11.number(),
  subtotal: z11.number(),
  total: z11.number(),
  lineItems: z11.array(previewUpdateNextCycleLineItemSchema),
  usageLineItems: z11.array(previewUpdateUsageLineItemSchema)
});
var previewUpdateIncomingFeatureQuantitySchema = z11.object({
  featureId: z11.string(),
  quantity: z11.number()
});
var previewUpdateOutgoingFeatureQuantitySchema = z11.object({
  featureId: z11.string(),
  quantity: z11.number()
});
var previewUpdateFeatureQuantityRequestOutboundSchema = z11.object({
  feature_id: z11.string(),
  quantity: z11.union([z11.number(), z11.undefined()]).optional(),
  adjustable: z11.union([z11.boolean(), z11.undefined()]).optional()
});
var previewUpdateBasePriceOutboundSchema = z11.object({
  amount: z11.number(),
  interval: z11.string(),
  interval_count: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdateResetOutboundSchema = z11.object({
  interval: z11.string(),
  interval_count: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdateToOutboundSchema = z11.union([z11.number(), z11.string()]);
var previewUpdateTierOutboundSchema = z11.object({
  to: z11.union([z11.number(), z11.string()]),
  amount: z11.union([z11.number(), z11.undefined()]).optional(),
  flat_amount: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdatePriceOutboundSchema = z11.object({
  amount: z11.union([z11.number(), z11.undefined()]).optional(),
  tiers: z11.union([z11.array(previewUpdateTierOutboundSchema), z11.undefined()]).optional(),
  tier_behavior: z11.union([z11.string(), z11.undefined()]).optional(),
  interval: z11.string(),
  interval_count: z11.number(),
  billing_units: z11.number(),
  billing_method: z11.string(),
  max_purchase: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdateProrationOutboundSchema = z11.object({
  on_increase: z11.string(),
  on_decrease: z11.string()
});
var previewUpdateRolloverOutboundSchema = z11.object({
  max: z11.union([z11.number(), z11.undefined()]).optional(),
  expiry_duration_type: z11.string(),
  expiry_duration_length: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdatePlanItemOutboundSchema = z11.object({
  feature_id: z11.string(),
  included: z11.union([z11.number(), z11.undefined()]).optional(),
  unlimited: z11.union([z11.boolean(), z11.undefined()]).optional(),
  reset: z11.union([previewUpdateResetOutboundSchema, z11.undefined()]).optional(),
  price: z11.union([previewUpdatePriceOutboundSchema, z11.undefined()]).optional(),
  proration: z11.union([previewUpdateProrationOutboundSchema, z11.undefined()]).optional(),
  rollover: z11.union([previewUpdateRolloverOutboundSchema, z11.undefined()]).optional()
});
var previewUpdateFreeTrialParamsOutboundSchema = z11.object({
  duration_length: z11.number(),
  duration_type: z11.string(),
  card_required: z11.boolean()
});
var previewUpdateCustomizeOutboundSchema = z11.object({
  price: z11.union([previewUpdateBasePriceOutboundSchema, z11.undefined()]).optional().nullable(),
  items: z11.union([z11.array(previewUpdatePlanItemOutboundSchema), z11.undefined()]).optional(),
  free_trial: z11.union([previewUpdateFreeTrialParamsOutboundSchema, z11.undefined()]).optional().nullable()
});
var previewUpdateInvoiceModeOutboundSchema = z11.object({
  enabled: z11.boolean(),
  enable_plan_immediately: z11.boolean(),
  finalize: z11.boolean()
});
var previewUpdateRecalculateBalancesOutboundSchema = z11.object({
  enabled: z11.boolean()
});
var previewUpdateParamsOutboundSchema = z11.object({
  customer_id: z11.string(),
  entity_id: z11.union([z11.string(), z11.undefined()]).optional(),
  plan_id: z11.union([z11.string(), z11.undefined()]).optional(),
  feature_quantities: z11.union([
    z11.array(previewUpdateFeatureQuantityRequestOutboundSchema),
    z11.undefined()
  ]).optional(),
  version: z11.union([z11.number(), z11.undefined()]).optional(),
  customize: z11.union([previewUpdateCustomizeOutboundSchema, z11.undefined()]).optional(),
  invoice_mode: z11.union([previewUpdateInvoiceModeOutboundSchema, z11.undefined()]).optional(),
  proration_behavior: z11.union([z11.string(), z11.undefined()]).optional(),
  redirect_mode: z11.string(),
  subscription_id: z11.union([z11.string(), z11.undefined()]).optional(),
  cancel_action: z11.union([z11.string(), z11.undefined()]).optional(),
  no_billing_changes: z11.union([z11.boolean(), z11.undefined()]).optional(),
  recalculate_balances: z11.union([previewUpdateRecalculateBalancesOutboundSchema, z11.undefined()]).optional()
});
var closedEnumSchema7 = z11.any();
var planSchema3 = z11.any();
var openEnumSchema6 = z11.any();
var previewUpdatePriceIntervalSchema = closedEnumSchema7;
var previewUpdateBasePriceSchema = z11.object({
  amount: z11.number(),
  interval: previewUpdatePriceIntervalSchema,
  intervalCount: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdateResetIntervalSchema = closedEnumSchema7;
var previewUpdateResetSchema = z11.object({
  interval: previewUpdateResetIntervalSchema,
  intervalCount: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdateTierBehaviorSchema = closedEnumSchema7;
var previewUpdateItemPriceIntervalSchema = closedEnumSchema7;
var previewUpdateBillingMethodSchema = closedEnumSchema7;
var previewUpdatePriceSchema = z11.object({
  amount: z11.union([z11.number(), z11.undefined()]).optional(),
  tiers: z11.union([z11.array(previewUpdateTierSchema), z11.undefined()]).optional(),
  tierBehavior: z11.union([previewUpdateTierBehaviorSchema, z11.undefined()]).optional(),
  interval: previewUpdateItemPriceIntervalSchema,
  intervalCount: z11.union([z11.number(), z11.undefined()]).optional(),
  billingUnits: z11.union([z11.number(), z11.undefined()]).optional(),
  billingMethod: previewUpdateBillingMethodSchema,
  maxPurchase: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdateOnIncreaseSchema = closedEnumSchema7;
var previewUpdateOnDecreaseSchema = closedEnumSchema7;
var previewUpdateProrationSchema = z11.object({
  onIncrease: previewUpdateOnIncreaseSchema,
  onDecrease: previewUpdateOnDecreaseSchema
});
var previewUpdateExpiryDurationTypeSchema = closedEnumSchema7;
var previewUpdateRolloverSchema = z11.object({
  max: z11.union([z11.number(), z11.undefined()]).optional(),
  expiryDurationType: previewUpdateExpiryDurationTypeSchema,
  expiryDurationLength: z11.union([z11.number(), z11.undefined()]).optional()
});
var previewUpdatePlanItemSchema = z11.object({
  featureId: z11.string(),
  included: z11.union([z11.number(), z11.undefined()]).optional(),
  unlimited: z11.union([z11.boolean(), z11.undefined()]).optional(),
  reset: z11.union([previewUpdateResetSchema, z11.undefined()]).optional(),
  price: z11.union([previewUpdatePriceSchema, z11.undefined()]).optional(),
  proration: z11.union([previewUpdateProrationSchema, z11.undefined()]).optional(),
  rollover: z11.union([previewUpdateRolloverSchema, z11.undefined()]).optional()
});
var previewUpdateDurationTypeSchema = closedEnumSchema7;
var previewUpdateFreeTrialParamsSchema = z11.object({
  durationLength: z11.number(),
  durationType: z11.union([previewUpdateDurationTypeSchema, z11.undefined()]).optional(),
  cardRequired: z11.union([z11.boolean(), z11.undefined()]).optional()
});
var previewUpdateCustomizeSchema = z11.object({
  price: z11.union([previewUpdateBasePriceSchema, z11.undefined()]).optional().nullable(),
  items: z11.union([z11.array(previewUpdatePlanItemSchema), z11.undefined()]).optional(),
  freeTrial: z11.union([previewUpdateFreeTrialParamsSchema, z11.undefined()]).optional().nullable()
});
var previewUpdateProrationBehaviorSchema = closedEnumSchema7;
var previewUpdateRedirectModeSchema = closedEnumSchema7;
var previewUpdateCancelActionSchema = closedEnumSchema7;
var previewUpdateParamsSchema = z11.object({
  customerId: z11.string(),
  entityId: z11.union([z11.string(), z11.undefined()]).optional(),
  planId: z11.union([z11.string(), z11.undefined()]).optional(),
  featureQuantities: z11.union([z11.array(previewUpdateFeatureQuantityRequestSchema), z11.undefined()]).optional(),
  version: z11.union([z11.number(), z11.undefined()]).optional(),
  customize: z11.union([previewUpdateCustomizeSchema, z11.undefined()]).optional(),
  invoiceMode: z11.union([previewUpdateInvoiceModeSchema, z11.undefined()]).optional(),
  prorationBehavior: z11.union([previewUpdateProrationBehaviorSchema, z11.undefined()]).optional(),
  redirectMode: z11.union([previewUpdateRedirectModeSchema, z11.undefined()]).optional(),
  subscriptionId: z11.union([z11.string(), z11.undefined()]).optional(),
  cancelAction: z11.union([previewUpdateCancelActionSchema, z11.undefined()]).optional(),
  noBillingChanges: z11.union([z11.boolean(), z11.undefined()]).optional(),
  recalculateBalances: z11.union([previewUpdateRecalculateBalancesSchema, z11.undefined()]).optional()
});
var previewUpdateIncomingSchema = z11.object({
  planId: z11.string(),
  plan: z11.union([planSchema3, z11.undefined()]).optional(),
  featureQuantities: z11.array(previewUpdateIncomingFeatureQuantitySchema),
  effectiveAt: z11.number().nullable(),
  canceledAt: z11.number().nullable(),
  expiresAt: z11.number().nullable()
});
var previewUpdateOutgoingSchema = z11.object({
  planId: z11.string(),
  plan: z11.union([planSchema3, z11.undefined()]).optional(),
  featureQuantities: z11.array(previewUpdateOutgoingFeatureQuantitySchema),
  effectiveAt: z11.number().nullable(),
  canceledAt: z11.number().nullable(),
  expiresAt: z11.number().nullable()
});
var intentSchema = openEnumSchema6;
var previewUpdateResponseSchema = z11.object({
  customerId: z11.string(),
  lineItems: z11.array(previewUpdateLineItemSchema),
  subtotal: z11.number(),
  total: z11.number(),
  currency: z11.string(),
  nextCycle: z11.union([previewUpdateNextCycleSchema, z11.undefined()]).optional(),
  expand: z11.union([z11.array(z11.string()), z11.undefined()]).optional(),
  incoming: z11.array(previewUpdateIncomingSchema),
  outgoing: z11.array(previewUpdateOutgoingSchema),
  intent: intentSchema
});

// src/generated/redeemReferralCodeSchemas.ts
import { z as z12 } from "zod/v4";
var redeemReferralCodeGlobalsSchema = z12.object({
  xApiVersion: z12.union([z12.string(), z12.undefined()]).optional()
});
var redeemReferralCodeParamsSchema = z12.object({
  code: z12.string(),
  customerId: z12.string()
});
var redeemReferralCodeResponseSchema = z12.object({
  id: z12.string(),
  customerId: z12.string(),
  rewardId: z12.string()
});
var redeemReferralCodeParamsOutboundSchema = z12.object({
  code: z12.string(),
  customer_id: z12.string()
});

// src/generated/setupPaymentSchemas.ts
import { z as z13 } from "zod/v4";
var setupPaymentGlobalsSchema = z13.object({
  xApiVersion: z13.union([z13.string(), z13.undefined()]).optional()
});
var setupPaymentFeatureQuantitySchema = z13.object({
  featureId: z13.string(),
  quantity: z13.union([z13.number(), z13.undefined()]).optional(),
  adjustable: z13.union([z13.boolean(), z13.undefined()]).optional()
});
var setupPaymentToSchema = z13.union([z13.number(), z13.string()]);
var setupPaymentTierSchema = z13.object({
  to: z13.union([z13.number(), z13.string()]),
  amount: z13.union([z13.number(), z13.undefined()]).optional(),
  flatAmount: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentAttachDiscountSchema = z13.object({
  rewardId: z13.union([z13.string(), z13.undefined()]).optional(),
  promotionCode: z13.union([z13.string(), z13.undefined()]).optional()
});
var setupPaymentCustomLineItemSchema = z13.object({
  amount: z13.number(),
  description: z13.string()
});
var setupPaymentCarryOverBalancesSchema = z13.object({
  enabled: z13.boolean(),
  featureIds: z13.union([z13.array(z13.string()), z13.undefined()]).optional()
});
var setupPaymentCarryOverUsagesSchema = z13.object({
  enabled: z13.boolean(),
  featureIds: z13.union([z13.array(z13.string()), z13.undefined()]).optional()
});
var setupPaymentResponseSchema = z13.object({
  customerId: z13.string(),
  entityId: z13.union([z13.string(), z13.undefined()]).optional(),
  url: z13.string()
});
var setupPaymentFeatureQuantityOutboundSchema = z13.object({
  feature_id: z13.string(),
  quantity: z13.union([z13.number(), z13.undefined()]).optional(),
  adjustable: z13.union([z13.boolean(), z13.undefined()]).optional()
});
var setupPaymentBasePriceOutboundSchema = z13.object({
  amount: z13.number(),
  interval: z13.string(),
  interval_count: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentResetOutboundSchema = z13.object({
  interval: z13.string(),
  interval_count: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentToOutboundSchema = z13.union([z13.number(), z13.string()]);
var setupPaymentTierOutboundSchema = z13.object({
  to: z13.union([z13.number(), z13.string()]),
  amount: z13.union([z13.number(), z13.undefined()]).optional(),
  flat_amount: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentPriceOutboundSchema = z13.object({
  amount: z13.union([z13.number(), z13.undefined()]).optional(),
  tiers: z13.union([z13.array(setupPaymentTierOutboundSchema), z13.undefined()]).optional(),
  tier_behavior: z13.union([z13.string(), z13.undefined()]).optional(),
  interval: z13.string(),
  interval_count: z13.number(),
  billing_units: z13.number(),
  billing_method: z13.string(),
  max_purchase: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentProrationOutboundSchema = z13.object({
  on_increase: z13.string(),
  on_decrease: z13.string()
});
var setupPaymentRolloverOutboundSchema = z13.object({
  max: z13.union([z13.number(), z13.undefined()]).optional(),
  expiry_duration_type: z13.string(),
  expiry_duration_length: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentPlanItemOutboundSchema = z13.object({
  feature_id: z13.string(),
  included: z13.union([z13.number(), z13.undefined()]).optional(),
  unlimited: z13.union([z13.boolean(), z13.undefined()]).optional(),
  reset: z13.union([setupPaymentResetOutboundSchema, z13.undefined()]).optional(),
  price: z13.union([setupPaymentPriceOutboundSchema, z13.undefined()]).optional(),
  proration: z13.union([setupPaymentProrationOutboundSchema, z13.undefined()]).optional(),
  rollover: z13.union([setupPaymentRolloverOutboundSchema, z13.undefined()]).optional()
});
var setupPaymentFreeTrialParamsOutboundSchema = z13.object({
  duration_length: z13.number(),
  duration_type: z13.string(),
  card_required: z13.boolean()
});
var setupPaymentCustomizeOutboundSchema = z13.object({
  price: z13.union([setupPaymentBasePriceOutboundSchema, z13.undefined()]).optional().nullable(),
  items: z13.union([z13.array(setupPaymentPlanItemOutboundSchema), z13.undefined()]).optional(),
  free_trial: z13.union([setupPaymentFreeTrialParamsOutboundSchema, z13.undefined()]).optional().nullable()
});
var setupPaymentAttachDiscountOutboundSchema = z13.object({
  reward_id: z13.union([z13.string(), z13.undefined()]).optional(),
  promotion_code: z13.union([z13.string(), z13.undefined()]).optional()
});
var setupPaymentCustomLineItemOutboundSchema = z13.object({
  amount: z13.number(),
  description: z13.string()
});
var setupPaymentCarryOverBalancesOutboundSchema = z13.object({
  enabled: z13.boolean(),
  feature_ids: z13.union([z13.array(z13.string()), z13.undefined()]).optional()
});
var setupPaymentCarryOverUsagesOutboundSchema = z13.object({
  enabled: z13.boolean(),
  feature_ids: z13.union([z13.array(z13.string()), z13.undefined()]).optional()
});
var setupPaymentParamsOutboundSchema = z13.object({
  customer_id: z13.string(),
  entity_id: z13.union([z13.string(), z13.undefined()]).optional(),
  plan_id: z13.union([z13.string(), z13.undefined()]).optional(),
  feature_quantities: z13.union([z13.array(setupPaymentFeatureQuantityOutboundSchema), z13.undefined()]).optional(),
  version: z13.union([z13.number(), z13.undefined()]).optional(),
  customize: z13.union([setupPaymentCustomizeOutboundSchema, z13.undefined()]).optional(),
  proration_behavior: z13.union([z13.string(), z13.undefined()]).optional(),
  subscription_id: z13.union([z13.string(), z13.undefined()]).optional(),
  discounts: z13.union([z13.array(setupPaymentAttachDiscountOutboundSchema), z13.undefined()]).optional(),
  success_url: z13.union([z13.string(), z13.undefined()]).optional(),
  checkout_session_params: z13.union([z13.record(z13.string(), z13.any()), z13.undefined()]).optional(),
  custom_line_items: z13.union([z13.array(setupPaymentCustomLineItemOutboundSchema), z13.undefined()]).optional(),
  processor_subscription_id: z13.union([z13.string(), z13.undefined()]).optional(),
  carry_over_balances: z13.union([setupPaymentCarryOverBalancesOutboundSchema, z13.undefined()]).optional(),
  carry_over_usages: z13.union([setupPaymentCarryOverUsagesOutboundSchema, z13.undefined()]).optional(),
  metadata: z13.union([z13.record(z13.string(), z13.string()), z13.undefined()]).optional()
});
var closedEnumSchema8 = z13.any();
var setupPaymentPriceIntervalSchema = closedEnumSchema8;
var setupPaymentBasePriceSchema = z13.object({
  amount: z13.number(),
  interval: setupPaymentPriceIntervalSchema,
  intervalCount: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentResetIntervalSchema = closedEnumSchema8;
var setupPaymentResetSchema = z13.object({
  interval: setupPaymentResetIntervalSchema,
  intervalCount: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentTierBehaviorSchema = closedEnumSchema8;
var setupPaymentItemPriceIntervalSchema = closedEnumSchema8;
var setupPaymentBillingMethodSchema = closedEnumSchema8;
var setupPaymentPriceSchema = z13.object({
  amount: z13.union([z13.number(), z13.undefined()]).optional(),
  tiers: z13.union([z13.array(setupPaymentTierSchema), z13.undefined()]).optional(),
  tierBehavior: z13.union([setupPaymentTierBehaviorSchema, z13.undefined()]).optional(),
  interval: setupPaymentItemPriceIntervalSchema,
  intervalCount: z13.union([z13.number(), z13.undefined()]).optional(),
  billingUnits: z13.union([z13.number(), z13.undefined()]).optional(),
  billingMethod: setupPaymentBillingMethodSchema,
  maxPurchase: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentOnIncreaseSchema = closedEnumSchema8;
var setupPaymentOnDecreaseSchema = closedEnumSchema8;
var setupPaymentProrationSchema = z13.object({
  onIncrease: setupPaymentOnIncreaseSchema,
  onDecrease: setupPaymentOnDecreaseSchema
});
var setupPaymentExpiryDurationTypeSchema = closedEnumSchema8;
var setupPaymentRolloverSchema = z13.object({
  max: z13.union([z13.number(), z13.undefined()]).optional(),
  expiryDurationType: setupPaymentExpiryDurationTypeSchema,
  expiryDurationLength: z13.union([z13.number(), z13.undefined()]).optional()
});
var setupPaymentPlanItemSchema = z13.object({
  featureId: z13.string(),
  included: z13.union([z13.number(), z13.undefined()]).optional(),
  unlimited: z13.union([z13.boolean(), z13.undefined()]).optional(),
  reset: z13.union([setupPaymentResetSchema, z13.undefined()]).optional(),
  price: z13.union([setupPaymentPriceSchema, z13.undefined()]).optional(),
  proration: z13.union([setupPaymentProrationSchema, z13.undefined()]).optional(),
  rollover: z13.union([setupPaymentRolloverSchema, z13.undefined()]).optional()
});
var setupPaymentDurationTypeSchema = closedEnumSchema8;
var setupPaymentFreeTrialParamsSchema = z13.object({
  durationLength: z13.number(),
  durationType: z13.union([setupPaymentDurationTypeSchema, z13.undefined()]).optional(),
  cardRequired: z13.union([z13.boolean(), z13.undefined()]).optional()
});
var setupPaymentCustomizeSchema = z13.object({
  price: z13.union([setupPaymentBasePriceSchema, z13.undefined()]).optional().nullable(),
  items: z13.union([z13.array(setupPaymentPlanItemSchema), z13.undefined()]).optional(),
  freeTrial: z13.union([setupPaymentFreeTrialParamsSchema, z13.undefined()]).optional().nullable()
});
var setupPaymentProrationBehaviorSchema = closedEnumSchema8;
var setupPaymentParamsSchema = z13.object({
  customerId: z13.string(),
  entityId: z13.union([z13.string(), z13.undefined()]).optional(),
  planId: z13.union([z13.string(), z13.undefined()]).optional(),
  featureQuantities: z13.union([z13.array(setupPaymentFeatureQuantitySchema), z13.undefined()]).optional(),
  version: z13.union([z13.number(), z13.undefined()]).optional(),
  customize: z13.union([setupPaymentCustomizeSchema, z13.undefined()]).optional(),
  prorationBehavior: z13.union([setupPaymentProrationBehaviorSchema, z13.undefined()]).optional(),
  subscriptionId: z13.union([z13.string(), z13.undefined()]).optional(),
  discounts: z13.union([z13.array(setupPaymentAttachDiscountSchema), z13.undefined()]).optional(),
  successUrl: z13.union([z13.string(), z13.undefined()]).optional(),
  checkoutSessionParams: z13.union([z13.record(z13.string(), z13.any()), z13.undefined()]).optional(),
  customLineItems: z13.union([z13.array(setupPaymentCustomLineItemSchema), z13.undefined()]).optional(),
  processorSubscriptionId: z13.union([z13.string(), z13.undefined()]).optional(),
  carryOverBalances: z13.union([setupPaymentCarryOverBalancesSchema, z13.undefined()]).optional(),
  carryOverUsages: z13.union([setupPaymentCarryOverUsagesSchema, z13.undefined()]).optional(),
  metadata: z13.union([z13.record(z13.string(), z13.string()), z13.undefined()]).optional()
});

// src/generated/updateSubscriptionSchemas.ts
import { z as z14 } from "zod/v4";
var billingUpdateGlobalsSchema = z14.object({
  xApiVersion: z14.union([z14.string(), z14.undefined()]).optional()
});
var billingUpdateFeatureQuantitySchema = z14.object({
  featureId: z14.string(),
  quantity: z14.union([z14.number(), z14.undefined()]).optional(),
  adjustable: z14.union([z14.boolean(), z14.undefined()]).optional()
});
var billingUpdateToSchema = z14.union([z14.number(), z14.string()]);
var billingUpdateTierSchema = z14.object({
  to: z14.union([z14.number(), z14.string()]),
  amount: z14.union([z14.number(), z14.undefined()]).optional(),
  flatAmount: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdateInvoiceModeSchema = z14.object({
  enabled: z14.boolean(),
  enablePlanImmediately: z14.union([z14.boolean(), z14.undefined()]).optional(),
  finalize: z14.union([z14.boolean(), z14.undefined()]).optional()
});
var billingUpdateRecalculateBalancesSchema = z14.object({
  enabled: z14.boolean()
});
var billingUpdateInvoiceSchema = z14.object({
  status: z14.string().nullable(),
  stripeId: z14.string(),
  total: z14.number(),
  currency: z14.string(),
  hostedInvoiceUrl: z14.string().nullable()
});
var billingUpdateFeatureQuantityOutboundSchema = z14.object({
  feature_id: z14.string(),
  quantity: z14.union([z14.number(), z14.undefined()]).optional(),
  adjustable: z14.union([z14.boolean(), z14.undefined()]).optional()
});
var billingUpdateBasePriceOutboundSchema = z14.object({
  amount: z14.number(),
  interval: z14.string(),
  interval_count: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdateResetOutboundSchema = z14.object({
  interval: z14.string(),
  interval_count: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdateToOutboundSchema = z14.union([z14.number(), z14.string()]);
var billingUpdateTierOutboundSchema = z14.object({
  to: z14.union([z14.number(), z14.string()]),
  amount: z14.union([z14.number(), z14.undefined()]).optional(),
  flat_amount: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdatePriceOutboundSchema = z14.object({
  amount: z14.union([z14.number(), z14.undefined()]).optional(),
  tiers: z14.union([z14.array(billingUpdateTierOutboundSchema), z14.undefined()]).optional(),
  tier_behavior: z14.union([z14.string(), z14.undefined()]).optional(),
  interval: z14.string(),
  interval_count: z14.number(),
  billing_units: z14.number(),
  billing_method: z14.string(),
  max_purchase: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdateProrationOutboundSchema = z14.object({
  on_increase: z14.string(),
  on_decrease: z14.string()
});
var billingUpdateRolloverOutboundSchema = z14.object({
  max: z14.union([z14.number(), z14.undefined()]).optional(),
  expiry_duration_type: z14.string(),
  expiry_duration_length: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdatePlanItemOutboundSchema = z14.object({
  feature_id: z14.string(),
  included: z14.union([z14.number(), z14.undefined()]).optional(),
  unlimited: z14.union([z14.boolean(), z14.undefined()]).optional(),
  reset: z14.union([billingUpdateResetOutboundSchema, z14.undefined()]).optional(),
  price: z14.union([billingUpdatePriceOutboundSchema, z14.undefined()]).optional(),
  proration: z14.union([billingUpdateProrationOutboundSchema, z14.undefined()]).optional(),
  rollover: z14.union([billingUpdateRolloverOutboundSchema, z14.undefined()]).optional()
});
var billingUpdateFreeTrialParamsOutboundSchema = z14.object({
  duration_length: z14.number(),
  duration_type: z14.string(),
  card_required: z14.boolean()
});
var billingUpdateCustomizeOutboundSchema = z14.object({
  price: z14.union([billingUpdateBasePriceOutboundSchema, z14.undefined()]).optional().nullable(),
  items: z14.union([z14.array(billingUpdatePlanItemOutboundSchema), z14.undefined()]).optional(),
  free_trial: z14.union([billingUpdateFreeTrialParamsOutboundSchema, z14.undefined()]).optional().nullable()
});
var billingUpdateInvoiceModeOutboundSchema = z14.object({
  enabled: z14.boolean(),
  enable_plan_immediately: z14.boolean(),
  finalize: z14.boolean()
});
var billingUpdateRecalculateBalancesOutboundSchema = z14.object({
  enabled: z14.boolean()
});
var updateSubscriptionParamsOutboundSchema = z14.object({
  customer_id: z14.string(),
  entity_id: z14.union([z14.string(), z14.undefined()]).optional(),
  plan_id: z14.union([z14.string(), z14.undefined()]).optional(),
  feature_quantities: z14.union([z14.array(billingUpdateFeatureQuantityOutboundSchema), z14.undefined()]).optional(),
  version: z14.union([z14.number(), z14.undefined()]).optional(),
  customize: z14.union([billingUpdateCustomizeOutboundSchema, z14.undefined()]).optional(),
  invoice_mode: z14.union([billingUpdateInvoiceModeOutboundSchema, z14.undefined()]).optional(),
  proration_behavior: z14.union([z14.string(), z14.undefined()]).optional(),
  redirect_mode: z14.string(),
  subscription_id: z14.union([z14.string(), z14.undefined()]).optional(),
  cancel_action: z14.union([z14.string(), z14.undefined()]).optional(),
  no_billing_changes: z14.union([z14.boolean(), z14.undefined()]).optional(),
  recalculate_balances: z14.union([billingUpdateRecalculateBalancesOutboundSchema, z14.undefined()]).optional()
});
var closedEnumSchema9 = z14.any();
var openEnumSchema7 = z14.any();
var billingUpdatePriceIntervalSchema = closedEnumSchema9;
var billingUpdateBasePriceSchema = z14.object({
  amount: z14.number(),
  interval: billingUpdatePriceIntervalSchema,
  intervalCount: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdateResetIntervalSchema = closedEnumSchema9;
var billingUpdateResetSchema = z14.object({
  interval: billingUpdateResetIntervalSchema,
  intervalCount: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdateTierBehaviorSchema = closedEnumSchema9;
var billingUpdateItemPriceIntervalSchema = closedEnumSchema9;
var billingUpdateBillingMethodSchema = closedEnumSchema9;
var billingUpdatePriceSchema = z14.object({
  amount: z14.union([z14.number(), z14.undefined()]).optional(),
  tiers: z14.union([z14.array(billingUpdateTierSchema), z14.undefined()]).optional(),
  tierBehavior: z14.union([billingUpdateTierBehaviorSchema, z14.undefined()]).optional(),
  interval: billingUpdateItemPriceIntervalSchema,
  intervalCount: z14.union([z14.number(), z14.undefined()]).optional(),
  billingUnits: z14.union([z14.number(), z14.undefined()]).optional(),
  billingMethod: billingUpdateBillingMethodSchema,
  maxPurchase: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdateOnIncreaseSchema = closedEnumSchema9;
var billingUpdateOnDecreaseSchema = closedEnumSchema9;
var billingUpdateProrationSchema = z14.object({
  onIncrease: billingUpdateOnIncreaseSchema,
  onDecrease: billingUpdateOnDecreaseSchema
});
var billingUpdateExpiryDurationTypeSchema = closedEnumSchema9;
var billingUpdateRolloverSchema = z14.object({
  max: z14.union([z14.number(), z14.undefined()]).optional(),
  expiryDurationType: billingUpdateExpiryDurationTypeSchema,
  expiryDurationLength: z14.union([z14.number(), z14.undefined()]).optional()
});
var billingUpdatePlanItemSchema = z14.object({
  featureId: z14.string(),
  included: z14.union([z14.number(), z14.undefined()]).optional(),
  unlimited: z14.union([z14.boolean(), z14.undefined()]).optional(),
  reset: z14.union([billingUpdateResetSchema, z14.undefined()]).optional(),
  price: z14.union([billingUpdatePriceSchema, z14.undefined()]).optional(),
  proration: z14.union([billingUpdateProrationSchema, z14.undefined()]).optional(),
  rollover: z14.union([billingUpdateRolloverSchema, z14.undefined()]).optional()
});
var billingUpdateDurationTypeSchema = closedEnumSchema9;
var billingUpdateFreeTrialParamsSchema = z14.object({
  durationLength: z14.number(),
  durationType: z14.union([billingUpdateDurationTypeSchema, z14.undefined()]).optional(),
  cardRequired: z14.union([z14.boolean(), z14.undefined()]).optional()
});
var billingUpdateCustomizeSchema = z14.object({
  price: z14.union([billingUpdateBasePriceSchema, z14.undefined()]).optional().nullable(),
  items: z14.union([z14.array(billingUpdatePlanItemSchema), z14.undefined()]).optional(),
  freeTrial: z14.union([billingUpdateFreeTrialParamsSchema, z14.undefined()]).optional().nullable()
});
var billingUpdateProrationBehaviorSchema = closedEnumSchema9;
var billingUpdateRedirectModeSchema = closedEnumSchema9;
var billingUpdateCancelActionSchema = closedEnumSchema9;
var updateSubscriptionParamsSchema = z14.object({
  customerId: z14.string(),
  entityId: z14.union([z14.string(), z14.undefined()]).optional(),
  planId: z14.union([z14.string(), z14.undefined()]).optional(),
  featureQuantities: z14.union([z14.array(billingUpdateFeatureQuantitySchema), z14.undefined()]).optional(),
  version: z14.union([z14.number(), z14.undefined()]).optional(),
  customize: z14.union([billingUpdateCustomizeSchema, z14.undefined()]).optional(),
  invoiceMode: z14.union([billingUpdateInvoiceModeSchema, z14.undefined()]).optional(),
  prorationBehavior: z14.union([billingUpdateProrationBehaviorSchema, z14.undefined()]).optional(),
  redirectMode: z14.union([billingUpdateRedirectModeSchema, z14.undefined()]).optional(),
  subscriptionId: z14.union([z14.string(), z14.undefined()]).optional(),
  cancelAction: z14.union([billingUpdateCancelActionSchema, z14.undefined()]).optional(),
  noBillingChanges: z14.union([z14.boolean(), z14.undefined()]).optional(),
  recalculateBalances: z14.union([billingUpdateRecalculateBalancesSchema, z14.undefined()]).optional()
});
var billingUpdateCodeSchema = openEnumSchema7;
var billingUpdateRequiredActionSchema = z14.object({
  code: billingUpdateCodeSchema,
  reason: z14.string()
});
var billingUpdateResponseSchema = z14.object({
  customerId: z14.string(),
  entityId: z14.union([z14.string(), z14.undefined()]).optional(),
  invoice: z14.union([billingUpdateInvoiceSchema, z14.undefined()]).optional(),
  paymentUrl: z14.string().nullable(),
  requiredAction: z14.union([billingUpdateRequiredActionSchema, z14.undefined()]).optional()
});

// src/backend/core/routes/routeConfigs.ts
var getEntityBodySchema = z15.object({
  entityId: z15.string()
});
var routeConfigs = [
  {
    route: "getOrCreateCustomer",
    sdkMethod: (autumn, args) => autumn.customers.getOrCreate(args),
    requireCustomer: false,
    // customHandler handles auth logic for errorOnNotFound
    bodySchema: z15.object({
      errorOnNotFound: z15.boolean().optional().default(true),
      // expand: z.array(z.enum(CustomerExpand)).optional(),
      expand: z15.array(z15.string()).optional()
    }),
    customHandler: async ({ autumn, identity, body }) => {
      const sanitizedBody = sanitizeBody(body);
      if (!identity?.customerId && sanitizedBody.errorOnNotFound === false) {
        return backendSuccess({ statusCode: 204, body: null });
      }
      if (!identity?.customerId) {
        return backendError({
          message: "customerId not found",
          code: "no_customer_id",
          statusCode: 401
        });
      }
      const existingExpand = Array.isArray(sanitizedBody.expand) ? sanitizedBody.expand : [];
      const args = {
        customerId: identity.customerId,
        ...identity.customerData,
        ...sanitizedBody,
        expand: [...existingExpand, "balances.feature"]
      };
      return autumn.customers.getOrCreate(args);
    }
  },
  {
    route: "getEntity",
    sdkMethod: (autumn, args) => autumn.entities.get(args),
    bodySchema: getEntityBodySchema
  },
  {
    route: "attach",
    sdkMethod: (autumn, args) => autumn.billing.attach(args),
    bodySchema: attachParamsSchema
  },
  {
    route: "previewAttach",
    sdkMethod: (autumn, args) => autumn.billing.previewAttach(args),
    bodySchema: previewAttachParamsSchema
  },
  {
    route: "updateSubscription",
    sdkMethod: (autumn, args) => autumn.billing.update(args),
    bodySchema: updateSubscriptionParamsSchema
  },
  {
    route: "previewUpdateSubscription",
    sdkMethod: (autumn, args) => autumn.billing.previewUpdate(args),
    bodySchema: previewUpdateParamsSchema
  },
  {
    route: "openCustomerPortal",
    sdkMethod: (autumn, args) => autumn.billing.openCustomerPortal(args),
    bodySchema: openCustomerPortalParamsSchema
  },
  {
    route: "createReferralCode",
    sdkMethod: (autumn, args) => autumn.referrals.createCode(args),
    bodySchema: createReferralCodeParamsSchema
  },
  {
    route: "redeemReferralCode",
    sdkMethod: (autumn, args) => autumn.referrals.redeemCode(args),
    bodySchema: redeemReferralCodeParamsSchema
  },
  {
    route: "multiAttach",
    sdkMethod: (autumn, args) => autumn.billing.multiAttach(args),
    bodySchema: multiAttachParamsSchema
  },
  {
    route: "previewMultiAttach",
    sdkMethod: (autumn, args) => autumn.billing.previewMultiAttach(args),
    bodySchema: previewMultiAttachParamsSchema
  },
  {
    route: "setupPayment",
    sdkMethod: (autumn, args) => autumn.billing.setupPayment(args),
    bodySchema: setupPaymentParamsSchema
  },
  {
    route: "listPlans",
    sdkMethod: (autumn, args) => autumn.plans.list(args),
    requireCustomer: false,
    bodySchema: listPlansParamsSchema.optional()
  },
  {
    route: "listEvents",
    sdkMethod: (autumn, args) => autumn.events.list(args),
    bodySchema: eventsListParamsSchema.optional()
  },
  {
    route: "aggregateEvents",
    sdkMethod: (autumn, args) => autumn.events.aggregate(args),
    bodySchema: eventsAggregateParamsSchema
  }
];

export {
  routeConfigs
};
