// autumn-js/dist/backend/index.mjs
import "./chunk-WYDGEDEU.mjs";
import "./chunk-VXND2UDT.mjs";
import "./chunk-JDL37I7Y.mjs";
import {
  createCoreHandler
} from "./chunk-VR35K2UZ.mjs";
import "./chunk-MJBKNDWD.mjs";
import "./chunk-55C2WWXQ.mjs";
import "./chunk-TBQ27LIK.mjs";
import "./chunk-J44WR3QZ.mjs";
import "./chunk-TSF4YCUO.mjs";
import {
  buildRouter
} from "./chunk-UAR3TMWI.mjs";
import {
  routeConfigs
} from "./chunk-BSY23UM3.mjs";
import "./chunk-OTXV5FY7.mjs";
import {
  secretKeyCheck
} from "./chunk-VZIMT7AO.mjs";
import {
  sanitizeBody
} from "./chunk-Z6QP3YEA.mjs";
import {
  backendError,
  backendSuccess,
  isBackendResult
} from "./chunk-N7NFYY54.mjs";

// src/backend/index.ts
async function autumnHandler(options) {
  const {
    request,
    customerId,
    customerData,
    clientOptions,
    pathPrefix,
    routes
  } = options;
  const handler = createCoreHandler({
    identify: async () => ({
      customerId: customerId ?? null,
      customerData
    }),
    secretKey: clientOptions?.secretKey,
    autumnURL: clientOptions?.baseURL,
    pathPrefix,
    routes
  });
  let url;
  if (!request.url.includes("http")) {
    url = new URL(request.url, "http://localhost:3000");
  } else {
    url = new URL(request.url);
  }
  const result = await handler({
    method: request.method,
    path: url.pathname,
    body: request.body,
    raw: request
  });
  return {
    statusCode: result.status,
    response: result.status === 204 ? null : result.body
  };
}
export {
  autumnHandler,
  backendError,
  backendSuccess,
  buildRouter,
  createCoreHandler,
  isBackendResult,
  routeConfigs,
  sanitizeBody,
  secretKeyCheck
};
