// autumn-js/dist/backend/core/utils/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/backend/core/utils/index.ts
var utils_exports = {};
__export(utils_exports, {
  backendError: () => backendError,
  backendSuccess: () => backendSuccess,
  isBackendResult: () => isBackendResult,
  sanitizeBody: () => sanitizeBody,
  secretKeyCheck: () => secretKeyCheck
});
module.exports = __toCommonJS(utils_exports);

// src/backend/core/utils/secretKeyCheck.ts
var secretKeyCheck = (secretKey) => {
  if (!secretKey && !process.env.AUTUMN_SECRET_KEY) {
    return {
      found: false,
      error: {
        statusCode: 500,
        message: "Autumn secret key not found in ENV variables or passed into autumnHandler",
        code: "no_secret_key"
      }
    };
  }
  return { found: true, error: null };
};

// src/backend/core/utils/backendRes.ts
var backendSuccess = ({
  body,
  statusCode = 200
}) => {
  return { statusCode, body };
};
var backendError = ({
  message,
  code = "internal_server_error",
  statusCode = 500,
  details
}) => {
  return {
    statusCode,
    body: {
      message: message || "Internal server error",
      code,
      statusCode,
      ...details !== void 0 ? { details } : {}
    }
  };
};
var isBackendResult = (value) => {
  return !!value && typeof value === "object" && "statusCode" in value && "body" in value;
};

// src/backend/core/utils/sanitizeBody.ts
var PROTECTED_FIELDS = [
  "customerId",
  "name",
  "email",
  "metadata",
  "stripeId"
];
var sanitizeBody = (body) => {
  const rawBody = body || {};
  const sanitized = {};
  for (const [key, value] of Object.entries(rawBody)) {
    if (!PROTECTED_FIELDS.includes(key)) {
      sanitized[key] = value;
    }
  }
  return sanitized;
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  backendError,
  backendSuccess,
  isBackendResult,
  sanitizeBody,
  secretKeyCheck
});
