// autumn-js/dist/better-auth/index.mjs
import "./chunk-HIWOK3G6.mjs";
import {
  createAutumnEndpoint
} from "./chunk-TDX4NREA.mjs";
import {
  createHandleBetterAuthRoute
} from "./chunk-I4LI2AOX.mjs";
import "./chunk-J2WYVGHS.mjs";
import "./chunk-LHA6NFFS.mjs";
import "./chunk-BT3EAYVW.mjs";
import "./chunk-GJAMWZNZ.mjs";

// src/better-auth/index.ts
function autumn(options = {}) {
  const {
    secretKey,
    baseURL,
    autumnURL,
    customerScope = "user",
    identify
  } = options;
  const handleRoute = createHandleBetterAuthRoute({
    secretKey,
    baseURL: autumnURL ?? baseURL,
    customerScope,
    identify
  });
  const endpoints = {
    getOrCreateCustomer: createAutumnEndpoint(
      "getOrCreateCustomer",
      handleRoute
    ),
    getEntity: createAutumnEndpoint("getEntity", handleRoute),
    attach: createAutumnEndpoint("attach", handleRoute),
    previewAttach: createAutumnEndpoint("previewAttach", handleRoute),
    updateSubscription: createAutumnEndpoint("updateSubscription", handleRoute),
    previewUpdateSubscription: createAutumnEndpoint(
      "previewUpdateSubscription",
      handleRoute
    ),
    openCustomerPortal: createAutumnEndpoint("openCustomerPortal", handleRoute),
    createReferralCode: createAutumnEndpoint("createReferralCode", handleRoute),
    redeemReferralCode: createAutumnEndpoint("redeemReferralCode", handleRoute),
    listPlans: createAutumnEndpoint("listPlans", handleRoute),
    listEvents: createAutumnEndpoint("listEvents", handleRoute),
    aggregateEvents: createAutumnEndpoint("aggregateEvents", handleRoute),
    multiAttach: createAutumnEndpoint("multiAttach", handleRoute),
    previewMultiAttach: createAutumnEndpoint("previewMultiAttach", handleRoute),
    setupPayment: createAutumnEndpoint("setupPayment", handleRoute)
  };
  return {
    id: "autumn",
    endpoints
  };
}
export {
  autumn
};
